<!DOCTYPE html>
<html>

<head>

<meta charset="utf-8">

<title>short_ Achab catch</title>


<script type="text/javascript" language="javascript" src='https://code.jquery.com/jquery-3.5.1.js'></script>

<script type="text/javascript" language="javascript" src='https://cdn.datatables.net/fixedheader/3.1.7/js/dataTables.fixedHeader.min.js'></script>

<script type="text/javascript" language="javascript" src='https://cdn.datatables.net/1.10.22/js/jquery.dataTables.min.js'></script>

<script>

var filter;

$(document).ready(function () {


	$('#table thead tr').clone(true).appendTo( '#table thead' );
            $('#table thead tr:eq(1) th').each( function (i) {
             var title = $(this).text();
              $(this).html( '<input type="text" placeholder="Search" />' );
                                                 
              $( 'input', this ).on( 'keyup change', function () {
                  if ( table.column(i).search() !== this.value ) {
                        table
                       .column(i)
                       .search( this.value,true,false )
                       .draw();
                  }
            } );
        } );
	



	var table = $('#table').DataTable(        {"dom": 'ift', "order": [] ,"lengthMenu":[ [ -1 ],[ "All" ]], "fixedHeader": true, "orderCellsTop": true, "oLanguage": {"sInfo": "TOTAL: _TOTAL_ lines" } } );


filter = function  (cat) {

	var metadata = document.getElementsByClassName("META");
	
	if (cat === "META"){
		
		document.getElementById("ALL").style.visibility = "collapse";

		for (i = 0; i < metadata.length; i++) {
			metadata[i].style.display = "block";
		}

	}else{
		
		for (i = 0; i < metadata.length; i++) {
			metadata[i].style.display = "none";
		}
		
		document.getElementById("ALL").style.visibility = "visible";
	
		var rowunselected = document.getElementsByTagName("TR");
		for (i = 0; i < rowunselected.length; i++) {
			rowunselected[i].style.visibility = "collapse";
		}
		var rowselect = document.getElementById('table').getElementsByClassName(cat);
		for (i = 0; i < rowselect.length; i++) {
			rowselect[i].style.visibility = "visible";
		}
	     
		var rowhead = document.getElementById('table').getElementsByClassName('head');
		for (i = 0; i < rowhead.length; i++) {
			rowhead[i].style.visibility = "visible";
		}


	}
}




});


</script>

<link rel="stylesheet" type="text/css" href='https://cdn.datatables.net/1.10.22/css/jquery.dataTables.min.css'>

<link rel="stylesheet" type="text/css" href='https://cdn.datatables.net/fixedheader/3.1.7/css/fixedHeader.dataTables.min.css'>


<style>



/* Style the tab */
        .tab {
                overflow: hidden;
                border: 1px solid #ccc;
                background-color: #f1f1f1;
                left: 0;
                bottom: 0;
                position: fixed;
                position: -webkit-sticky;
                position: sticky;
                }

/* Style the buttons inside the tab */
        .tab input {
                background-color: inherit;
                float: left;
                border: none;
                outline: none;
                cursor: pointer;
                padding: 14px 16px;
                transition: 0.3s;
                font-size: 17px;
                }
/* Change background color of buttons on hover */
        .tab input:hover {
                background-color: #ddd;
                }

/* Create an active/current tablink class */
        .tab input.active {
                background-color: #ccc;
                }
        .tab input:focus {
                background-color: #ccc;
                }

        thead input {
                width: 100%;
                }



/* Style METADATA */
		.META {
			display: none;
			white-space: pre-line;
		}

/* Style the tab content */
        .tabcontent {
                visibility: visible;
                padding: 6px 12px;
                /*border: 1px solid #ccc;*/
                border-top: none;
                }

/* Style the tab content FULL */

        .tooltip {
                position: relative;
                display: inline-block;
                border-bottom: 1px dotted black;
                max-width: 300px;
                word-wrap: break-word;
                }
        
        .tooltip .tooltiptext {
                visibility: hidden;
                width: auto;
                min-width: 300px;
                max-width: 600px;
                height: auto;
                background-color: #555;
                color: #fff;
                text-align: left;
                border-radius: 6px;
                padding: 5px 0;
                position: absolute;
                z-index: 1;
                top: 100%;
                left: 20%;
                margin-left: -20px;
                opacity: 0;
                transition: opacity 0.3s;
                overflow: hidden;
                text-overflow: ellipsis;
                white-space: pre-line;
                overflow-wrap: break-word;
                }

        .tooltip:hover .tooltiptext {
                visibility: visible;
                opacity: 1;
                }




</style>




</head>

	<body><div class="META"><b>Arguments:</b><br>--vcf   INPUT_hg19_multianno_MPA.vcf   --candidates   candidats.txt   --trio   --case   B00GMSH   --dad   B00GMSI   --mum   B00GMSJ   --outPrefix   short   <br><br><br><b>VCF Header:</b><br>##fileformat=VCFv4.2
##GATKCommandLine.HaplotypeCaller=ID=HaplotypeCaller,Version=3.8-0-ge9d806836,Date="Wed Aug 01 04:38:56 CEST 2018",Epoch=1533091136036,CommandLineOptions="analysis_type=HaplotypeCaller input_file=[/lustre/guignardt/output/28658/AJI_Fam8/MERGED_SAMPLES/MERGED_SAMPLES.bam] showFullBamList=false read_buffer_size=null read_filter=[] disable_read_filter=[] intervals=[/lustre/guignardt/output/28658/AJI_Fam8/MERGED_SAMPLES/559/DIR_DRMAA/HaplotypeCaller-1-sg/temp_01_of_28/scatter.intervals] excludeIntervals=null interval_set_rule=UNION interval_merging=ALL interval_padding=0 reference_sequence=/home/guignardt/work-mobidic/nenufaar/refData/genome/hg19/hg19.fa nonDeterministicRandomSeed=false disableDithering=false maxRuntime=-1 maxRuntimeUnits=MINUTES downsampling_type=NONE downsample_to_fraction=null downsample_to_coverage=null baq=OFF baqGapOpenPenalty=40.0 refactor_NDN_cigar_string=false fix_misencoded_quality_scores=false allow_potentially_misencoded_quality_scores=false useOriginalQualities=false defaultBaseQualities=-1 performanceLog=null BQSR=null quantize_quals=0 static_quantized_quals=null round_down_quantized=false disable_indel_quals=false emit_original_quals=false preserve_qscores_less_than=6 globalQScorePrior=-1.0 secondsBetweenProgressUpdates=10 validation_strictness=SILENT remove_program_records=false keep_program_records=false sample_rename_mapping_file=null unsafe=null use_jdk_deflater=false use_jdk_inflater=false disable_auto_index_creation_and_locking_when_reading_rods=false no_cmdline_in_header=false sites_only=false never_trim_vcf_format_field=false bcf=false bam_compression=null simplifyBAM=false disable_bam_indexing=false generate_md5=false num_threads=1 num_cpu_threads_per_data_thread=1 num_io_threads=0 monitorThreadEfficiency=false num_bam_file_handles=null read_group_black_list=null pedigree=[] pedigreeString=[] pedigreeValidationType=STRICT allow_intervals_with_unindexed_bam=false generateShadowBCF=false variant_index_type=DYNAMIC_SEEK variant_index_parameter=-1 reference_window_stop=0 phone_home= gatk_key=null tag=NA logging_level=INFO log_to_file=null help=false version=false out=/lustre/guignardt/output/28658/AJI_Fam8/MERGED_SAMPLES/559/DIR_DRMAA/HaplotypeCaller-1-sg/temp_01_of_28/MERGED_SAMPLES.raw.vcf likelihoodCalculationEngine=PairHMM heterogeneousKmerSizeResolution=COMBO_MIN dbsnp=(RodBinding name=dbsnp source=/home/guignardt/work-mobidic/nenufaar/refData/dbSNP/138/CORRECT_dbsnp_138.hg19.vcf.gz) dontTrimActiveRegions=false maxDiscARExtension=25 maxGGAARExtension=300 paddingAroundIndels=150 paddingAroundSNPs=20 comp=[] annotation=[ReadPosRankSumTest] excludeAnnotation=[] group=[StandardAnnotation, StandardHCAnnotation] debug=false useFilteredReadsForAnnotations=false emitRefConfidence=NONE bamOutput=null bamWriterType=CALLED_HAPLOTYPES emitDroppedReads=false disableOptimizations=false annotateNDA=false useNewAFCalculator=false heterozygosity=0.001 indel_heterozygosity=1.25E-4 heterozygosity_stdev=0.01 standard_min_confidence_threshold_for_calling=30.0 standard_min_confidence_threshold_for_emitting=30.0 max_alternate_alleles=10 max_genotype_count=1024 max_num_PL_values=100 input_prior=[] sample_ploidy=2 genotyping_mode=DISCOVERY alleles=(RodBinding name= source=UNBOUND) contamination_fraction_to_filter=0.0 contamination_fraction_per_sample_file=null p_nonref_model=null exactcallslog=null output_mode=EMIT_VARIANTS_ONLY allSitePLs=false gcpHMM=10 pair_hmm_implementation=FASTEST_AVAILABLE phredScaledGlobalReadMismappingRate=45 noFpga=false nativePairHmmThreads=1 useDoublePrecision=false sample_name=null kmerSize=[10, 25] dontIncreaseKmerSizesForCycles=false allowNonUniqueKmersInRef=false numPruningSamples=1 recoverDanglingHeads=false doNotRecoverDanglingBranches=false minDanglingBranchLength=4 consensus=false maxNumHaplotypesInPopulation=128 errorCorrectKmers=false minPruning=1 debugGraphTransformations=false allowCyclesInKmerGraphToGeneratePaths=false graphOutput=null kmerLengthForReadErrorCorrection=25 minObservationsForKmerToBeSolid=20 GVCFGQBands=[1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 58, 59, 60, 70, 80, 90, 99] indelSizeToEliminateInRefModel=10 min_base_quality_score=10 includeUmappedReads=false useAllelesTrigger=false doNotRunPhysicalPhasing=true keepRG=null justDetermineActiveRegions=false dontGenotype=false dontUseSoftClippedBases=false captureAssemblyFailureBAM=false errorCorrectReads=false pcr_indel_model=CONSERVATIVE maxReadsInRegionPerSample=10000 minReadsPerAlignmentStart=10 mergeVariantsViaLD=false activityProfileOut=null activeRegionOut=null activeRegionIn=null activeRegionExtension=null forceActive=false activeRegionMaxSize=null bandPassSigma=null maxReadsInMemoryPerSample=30000 maxTotalReadsInMemory=10000000 maxProbPropagationDistance=50 activeProbabilityThreshold=0.002 min_mapping_quality_score=20 filter_reads_with_N_cigar=false filter_mismatching_base_and_quals=false filter_bases_not_stored=false"
##GATKCommandLine.VariantFiltration=ID=VariantFiltration,Version=3.8-0-ge9d806836,Date="Wed Aug 01 05:16:32 CEST 2018",Epoch=1533093392176,CommandLineOptions="analysis_type=VariantFiltration input_file=[] showFullBamList=false read_buffer_size=null read_filter=[] disable_read_filter=[] intervals=[/lustre/guignardt/output/28658/AJI_Fam8/MERGED_SAMPLES/559/DIR_DRMAA/VariantFiltration-1-sg/temp_01_of_28/scatter.intervals] excludeIntervals=null interval_set_rule=UNION interval_merging=ALL interval_padding=0 reference_sequence=/home/guignardt/work-mobidic/nenufaar/refData/genome/hg19/hg19.fa nonDeterministicRandomSeed=false disableDithering=false maxRuntime=-1 maxRuntimeUnits=MINUTES downsampling_type=BY_SAMPLE downsample_to_fraction=null downsample_to_coverage=1000 baq=OFF baqGapOpenPenalty=40.0 refactor_NDN_cigar_string=false fix_misencoded_quality_scores=false allow_potentially_misencoded_quality_scores=false useOriginalQualities=false defaultBaseQualities=-1 performanceLog=null BQSR=null quantize_quals=0 static_quantized_quals=null round_down_quantized=false disable_indel_quals=false emit_original_quals=false preserve_qscores_less_than=6 globalQScorePrior=-1.0 secondsBetweenProgressUpdates=10 validation_strictness=SILENT remove_program_records=false keep_program_records=false sample_rename_mapping_file=null unsafe=null use_jdk_deflater=false use_jdk_inflater=false disable_auto_index_creation_and_locking_when_reading_rods=false no_cmdline_in_header=false sites_only=false never_trim_vcf_format_field=false bcf=false bam_compression=null simplifyBAM=false disable_bam_indexing=false generate_md5=false num_threads=1 num_cpu_threads_per_data_thread=1 num_io_threads=0 monitorThreadEfficiency=false num_bam_file_handles=null read_group_black_list=null pedigree=[] pedigreeString=[] pedigreeValidationType=STRICT allow_intervals_with_unindexed_bam=false generateShadowBCF=false variant_index_type=DYNAMIC_SEEK variant_index_parameter=-1 reference_window_stop=0 phone_home= gatk_key=null tag=NA logging_level=INFO log_to_file=null help=false version=false variant=(RodBinding name=variant source=/lustre/guignardt/output/28658/AJI_Fam8/MERGED_SAMPLES/559/MERGED_SAMPLES.raw.polyx.vcf) mask=(RodBinding name= source=UNBOUND) out=/lustre/guignardt/output/28658/AJI_Fam8/MERGED_SAMPLES/559/DIR_DRMAA/VariantFiltration-1-sg/temp_01_of_28/MERGED_SAMPLES.raw.polyx.gatk.vcf filterExpression=[DP  20, QUAL  30.0, QD  1.5, FS  60.000, MQ  10.00, POLYX  7] filterName=[LowCoverage, LowQual, LowQD, StrandBias, LowMappingQuality, R8] genotypeFilterExpression=[] genotypeFilterName=[] clusterSize=3 clusterWindowSize=0 maskExtension=0 maskName=Mask filterNotInMask=false missingValuesInExpressionsShouldEvaluateAsFailing=false invalidatePreviousFilters=false invertFilterExpression=false invertGenotypeFilterExpression=false setFilteredGtToNocall=false filter_reads_with_N_cigar=false filter_mismatching_base_and_quals=false filter_bases_not_stored=false"
##VCFPolyXCmdLine=-R /home/guignardt/work-mobidic/nenufaar/refData/genome/hg19/hg19.fa -o output/28658/AJI_Fam8/MERGED_SAMPLES/559/MERGED_SAMPLES.raw.polyx.vcf output/28658/AJI_Fam8/MERGED_SAMPLES/559/MERGED_SAMPLES.raw.vcf
##VCFPolyXHtsJdkHome=lib/com/github/samtools/htsjdk/2.6.1/htsjdk-2.6.1.jar
##VCFPolyXHtsJdkVersion=2.6.1
##VCFPolyXVersion=ff2326c651be81dd8b99173a69d6005b274ca4c6
##bcftools_normCommand=norm -O v -m - -o /mnt/data/MobiDL/captainAchab/Done/B00GMSH/CaptainAchab/B00GMSH.vcf /home/adminngs/autoAchab/cromwell-executions/captainAchab/125ed6dd-2cf6-417a-8456-32f4c307d094/call-bcftoolsNorm/inputs/-793930549/B00GMSH_leftalign.vcf; Date=Fri Jul  3 15:39:50 2020
##bcftools_normVersion=1.9+htslib-1.9
##reference=file:///home/guignardt/work-mobidic/nenufaar/refData/genome/hg19/hg19.fa
##INFO=ID=AC,Number=A,Type=Integer,Description="Allele count in genotypes, for each ALT allele, in the same order as listed"
##INFO=ID=AF,Number=A,Type=Float,Description="Allele Frequency, for each ALT allele, in the same order as listed"
##INFO=ID=AN,Number=1,Type=Integer,Description="Total number of alleles in called genotypes"
##INFO=ID=BaseQRankSum,Number=1,Type=Float,Description="Z-score from Wilcoxon rank sum test of Alt Vs. Ref base qualities"
##INFO=ID=ClippingRankSum,Number=1,Type=Float,Description="Z-score From Wilcoxon rank sum test of Alt vs. Ref number of hard clipped bases"
##INFO=ID=DB,Number=0,Type=Flag,Description="dbSNP Membership"
##INFO=ID=DP,Number=1,Type=Integer,Description="Approximate read depth; some reads may have been filtered"
##INFO=ID=DS,Number=0,Type=Flag,Description="Were any of the samples downsampled?"
##INFO=ID=ExcessHet,Number=1,Type=Float,Description="Phred-scaled p-value for exact test of excess heterozygosity"
##INFO=ID=FS,Number=1,Type=Float,Description="Phred-scaled p-value using Fisher's exact test to detect strand bias"
##INFO=ID=HaplotypeScore,Number=1,Type=Float,Description="Consistency of the site with at most two segregating haplotypes"
##INFO=ID=InbreedingCoeff,Number=1,Type=Float,Description="Inbreeding coefficient as estimated from the genotype likelihoods per-sample when compared against the Hardy-Weinberg expectation"
##INFO=ID=MLEAC,Number=A,Type=Integer,Description="Maximum likelihood expectation (MLE) for the allele counts (not necessarily the same as the AC), for each ALT allele, in the same order as listed"
##INFO=ID=MLEAF,Number=A,Type=Float,Description="Maximum likelihood expectation (MLE) for the allele frequency (not necessarily the same as the AF), for each ALT allele, in the same order as listed"
##INFO=ID=MQ,Number=1,Type=Float,Description="RMS Mapping Quality"
##INFO=ID=MQRankSum,Number=1,Type=Float,Description="Z-score From Wilcoxon rank sum test of Alt vs. Ref read mapping qualities"
##INFO=ID=POLYX,Number=1,Type=Integer,Description="Number of repeated bases around REF"
##INFO=ID=QD,Number=1,Type=Float,Description="Variant Confidence/Quality by Depth"
##INFO=ID=ReadPosRankSum,Number=1,Type=Float,Description="Z-score from Wilcoxon rank sum test of Alt vs. Ref read position bias"
##INFO=ID=SOR,Number=1,Type=Float,Description="Symmetric Odds Ratio of 2x2 contingency table to detect strand bias"
##INFO=ID=ANNOVAR_DATE,Number=1,Type=String,Description="Flag the start of ANNOVAR annotation for one alternative allele"
##INFO=ID=Func.refGene,Number=.,Type=String,Description="Func.refGene annotation provided by ANNOVAR"
##INFO=ID=Gene.refGene,Number=.,Type=String,Description="Gene.refGene annotation provided by ANNOVAR"
##INFO=ID=GeneDetail.refGene,Number=.,Type=String,Description="GeneDetail.refGene annotation provided by ANNOVAR"
##INFO=ID=ExonicFunc.refGene,Number=.,Type=String,Description="ExonicFunc.refGene annotation provided by ANNOVAR"
##INFO=ID=AAChange.refGene,Number=.,Type=String,Description="AAChange.refGene annotation provided by ANNOVAR"
##INFO=ID=Approved_name.refGene,Number=.,Type=String,Description="Approved_name.refGene annotation provided by ANNOVAR"
##INFO=ID=Phenotypes.refGene,Number=.,Type=String,Description="Phenotypes.refGene annotation provided by ANNOVAR"
##INFO=ID=oe_lof_upper_rank.refGene,Number=.,Type=String,Description="oe_lof_upper_rank.refGene annotation provided by ANNOVAR"
##INFO=ID=oe_lof_upper_bin.refGene,Number=.,Type=String,Description="oe_lof_upper_bin.refGene annotation provided by ANNOVAR"
##INFO=ID=oe_lof.refGene,Number=.,Type=String,Description="oe_lof.refGene annotation provided by ANNOVAR"
##INFO=ID=oe_lof_lower.refGene,Number=.,Type=String,Description="oe_lof_lower.refGene annotation provided by ANNOVAR"
##INFO=ID=oe_lof_upper.refGene,Number=.,Type=String,Description="oe_lof_upper.refGene annotation provided by ANNOVAR"
##INFO=ID=oe_mis.refGene,Number=.,Type=String,Description="oe_mis.refGene annotation provided by ANNOVAR"
##INFO=ID=oe_mis_lower.refGene,Number=.,Type=String,Description="oe_mis_lower.refGene annotation provided by ANNOVAR"
##INFO=ID=oe_mis_upper.refGene,Number=.,Type=String,Description="oe_mis_upper.refGene annotation provided by ANNOVAR"
##INFO=ID=oe_syn.refGene,Number=.,Type=String,Description="oe_syn.refGene annotation provided by ANNOVAR"
##INFO=ID=oe_syn_lower.refGene,Number=.,Type=String,Description="oe_syn_lower.refGene annotation provided by ANNOVAR"
##INFO=ID=oe_syn_upper.refGene,Number=.,Type=String,Description="oe_syn_upper.refGene annotation provided by ANNOVAR"
##INFO=ID=Function_description.refGene,Number=.,Type=String,Description="Function_description.refGene annotation provided by ANNOVAR"
##INFO=ID=Disease_description.refGene,Number=.,Type=String,Description="Disease_description.refGene annotation provided by ANNOVAR"
##INFO=ID=Tissue_specificity(Uniprot).refGene,Number=.,Type=String,Description="Tissue_specificity(Uniprot).refGene annotation provided by ANNOVAR"
##INFO=ID=CLNALLELEID,Number=.,Type=String,Description="CLNALLELEID annotation provided by ANNOVAR"
##INFO=ID=CLNDN,Number=.,Type=String,Description="CLNDN annotation provided by ANNOVAR"
##INFO=ID=CLNDISDB,Number=.,Type=String,Description="CLNDISDB annotation provided by ANNOVAR"
##INFO=ID=CLNREVSTAT,Number=.,Type=String,Description="CLNREVSTAT annotation provided by ANNOVAR"
##INFO=ID=CLNSIG,Number=.,Type=String,Description="CLNSIG annotation provided by ANNOVAR"
##INFO=ID=SIFT_score,Number=.,Type=String,Description="SIFT_score annotation provided by ANNOVAR"
##INFO=ID=SIFT_converted_rankscore,Number=.,Type=String,Description="SIFT_converted_rankscore annotation provided by ANNOVAR"
##INFO=ID=SIFT_pred,Number=.,Type=String,Description="SIFT_pred annotation provided by ANNOVAR"
##INFO=ID=Polyphen2_HDIV_score,Number=.,Type=String,Description="Polyphen2_HDIV_score annotation provided by ANNOVAR"
##INFO=ID=Polyphen2_HDIV_rankscore,Number=.,Type=String,Description="Polyphen2_HDIV_rankscore annotation provided by ANNOVAR"
##INFO=ID=Polyphen2_HDIV_pred,Number=.,Type=String,Description="Polyphen2_HDIV_pred annotation provided by ANNOVAR"
##INFO=ID=Polyphen2_HVAR_score,Number=.,Type=String,Description="Polyphen2_HVAR_score annotation provided by ANNOVAR"
##INFO=ID=Polyphen2_HVAR_rankscore,Number=.,Type=String,Description="Polyphen2_HVAR_rankscore annotation provided by ANNOVAR"
##INFO=ID=Polyphen2_HVAR_pred,Number=.,Type=String,Description="Polyphen2_HVAR_pred annotation provided by ANNOVAR"
##INFO=ID=LRT_score,Number=.,Type=String,Description="LRT_score annotation provided by ANNOVAR"
##INFO=ID=LRT_converted_rankscore,Number=.,Type=String,Description="LRT_converted_rankscore annotation provided by ANNOVAR"
##INFO=ID=LRT_pred,Number=.,Type=String,Description="LRT_pred annotation provided by ANNOVAR"
##INFO=ID=MutationTaster_score,Number=.,Type=String,Description="MutationTaster_score annotation provided by ANNOVAR"
##INFO=ID=MutationTaster_converted_rankscore,Number=.,Type=String,Description="MutationTaster_converted_rankscore annotation provided by ANNOVAR"
##INFO=ID=MutationTaster_pred,Number=.,Type=String,Description="MutationTaster_pred annotation provided by ANNOVAR"
##INFO=ID=MutationAssessor_score,Number=.,Type=String,Description="MutationAssessor_score annotation provided by ANNOVAR"
##INFO=ID=MutationAssessor_score_rankscore,Number=.,Type=String,Description="MutationAssessor_score_rankscore annotation provided by ANNOVAR"
##INFO=ID=MutationAssessor_pred,Number=.,Type=String,Description="MutationAssessor_pred annotation provided by ANNOVAR"
##INFO=ID=FATHMM_score,Number=.,Type=String,Description="FATHMM_score annotation provided by ANNOVAR"
##INFO=ID=FATHMM_converted_rankscore,Number=.,Type=String,Description="FATHMM_converted_rankscore annotation provided by ANNOVAR"
##INFO=ID=FATHMM_pred,Number=.,Type=String,Description="FATHMM_pred annotation provided by ANNOVAR"
##INFO=ID=PROVEAN_score,Number=.,Type=String,Description="PROVEAN_score annotation provided by ANNOVAR"
##INFO=ID=PROVEAN_converted_rankscore,Number=.,Type=String,Description="PROVEAN_converted_rankscore annotation provided by ANNOVAR"
##INFO=ID=PROVEAN_pred,Number=.,Type=String,Description="PROVEAN_pred annotation provided by ANNOVAR"
##INFO=ID=VEST3_score,Number=.,Type=String,Description="VEST3_score annotation provided by ANNOVAR"
##INFO=ID=VEST3_rankscore,Number=.,Type=String,Description="VEST3_rankscore annotation provided by ANNOVAR"
##INFO=ID=MetaSVM_score,Number=.,Type=String,Description="MetaSVM_score annotation provided by ANNOVAR"
##INFO=ID=MetaSVM_rankscore,Number=.,Type=String,Description="MetaSVM_rankscore annotation provided by ANNOVAR"
##INFO=ID=MetaSVM_pred,Number=.,Type=String,Description="MetaSVM_pred annotation provided by ANNOVAR"
##INFO=ID=MetaLR_score,Number=.,Type=String,Description="MetaLR_score annotation provided by ANNOVAR"
##INFO=ID=MetaLR_rankscore,Number=.,Type=String,Description="MetaLR_rankscore annotation provided by ANNOVAR"
##INFO=ID=MetaLR_pred,Number=.,Type=String,Description="MetaLR_pred annotation provided by ANNOVAR"
##INFO=ID=M-CAP_score,Number=.,Type=String,Description="M-CAP_score annotation provided by ANNOVAR"
##INFO=ID=M-CAP_rankscore,Number=.,Type=String,Description="M-CAP_rankscore annotation provided by ANNOVAR"
##INFO=ID=M-CAP_pred,Number=.,Type=String,Description="M-CAP_pred annotation provided by ANNOVAR"
##INFO=ID=REVEL_score,Number=.,Type=String,Description="REVEL_score annotation provided by ANNOVAR"
##INFO=ID=REVEL_rankscore,Number=.,Type=String,Description="REVEL_rankscore annotation provided by ANNOVAR"
##INFO=ID=MutPred_score,Number=.,Type=String,Description="MutPred_score annotation provided by ANNOVAR"
##INFO=ID=MutPred_rankscore,Number=.,Type=String,Description="MutPred_rankscore annotation provided by ANNOVAR"
##INFO=ID=CADD_raw,Number=.,Type=String,Description="CADD_raw annotation provided by ANNOVAR"
##INFO=ID=CADD_raw_rankscore,Number=.,Type=String,Description="CADD_raw_rankscore annotation provided by ANNOVAR"
##INFO=ID=CADD_phred,Number=.,Type=String,Description="CADD_phred annotation provided by ANNOVAR"
##INFO=ID=DANN_score,Number=.,Type=String,Description="DANN_score annotation provided by ANNOVAR"
##INFO=ID=DANN_rankscore,Number=.,Type=String,Description="DANN_rankscore annotation provided by ANNOVAR"
##INFO=ID=fathmm-MKL_coding_score,Number=.,Type=String,Description="fathmm-MKL_coding_score annotation provided by ANNOVAR"
##INFO=ID=fathmm-MKL_coding_rankscore,Number=.,Type=String,Description="fathmm-MKL_coding_rankscore annotation provided by ANNOVAR"
##INFO=ID=fathmm-MKL_coding_pred,Number=.,Type=String,Description="fathmm-MKL_coding_pred annotation provided by ANNOVAR"
##INFO=ID=Eigen_coding_or_noncoding,Number=.,Type=String,Description="Eigen_coding_or_noncoding annotation provided by ANNOVAR"
##INFO=ID=Eigen-raw,Number=.,Type=String,Description="Eigen-raw annotation provided by ANNOVAR"
##INFO=ID=Eigen-PC-raw,Number=.,Type=String,Description="Eigen-PC-raw annotation provided by ANNOVAR"
##INFO=ID=GenoCanyon_score,Number=.,Type=String,Description="GenoCanyon_score annotation provided by ANNOVAR"
##INFO=ID=GenoCanyon_score_rankscore,Number=.,Type=String,Description="GenoCanyon_score_rankscore annotation provided by ANNOVAR"
##INFO=ID=integrated_fitCons_score,Number=.,Type=String,Description="integrated_fitCons_score annotation provided by ANNOVAR"
##INFO=ID=integrated_fitCons_score_rankscore,Number=.,Type=String,Description="integrated_fitCons_score_rankscore annotation provided by ANNOVAR"
##INFO=ID=integrated_confidence_value,Number=.,Type=String,Description="integrated_confidence_value annotation provided by ANNOVAR"
##INFO=ID=GERP++_RS,Number=.,Type=String,Description="GERP++_RS annotation provided by ANNOVAR"
##INFO=ID=GERP++_RS_rankscore,Number=.,Type=String,Description="GERP++_RS_rankscore annotation provided by ANNOVAR"
##INFO=ID=phyloP100way_vertebrate,Number=.,Type=String,Description="phyloP100way_vertebrate annotation provided by ANNOVAR"
##INFO=ID=phyloP100way_vertebrate_rankscore,Number=.,Type=String,Description="phyloP100way_vertebrate_rankscore annotation provided by ANNOVAR"
##INFO=ID=phyloP20way_mammalian,Number=.,Type=String,Description="phyloP20way_mammalian annotation provided by ANNOVAR"
##INFO=ID=phyloP20way_mammalian_rankscore,Number=.,Type=String,Description="phyloP20way_mammalian_rankscore annotation provided by ANNOVAR"
##INFO=ID=phastCons100way_vertebrate,Number=.,Type=String,Description="phastCons100way_vertebrate annotation provided by ANNOVAR"
##INFO=ID=phastCons100way_vertebrate_rankscore,Number=.,Type=String,Description="phastCons100way_vertebrate_rankscore annotation provided by ANNOVAR"
##INFO=ID=phastCons20way_mammalian,Number=.,Type=String,Description="phastCons20way_mammalian annotation provided by ANNOVAR"
##INFO=ID=phastCons20way_mammalian_rankscore,Number=.,Type=String,Description="phastCons20way_mammalian_rankscore annotation provided by ANNOVAR"
##INFO=ID=SiPhy_29way_logOdds,Number=.,Type=String,Description="SiPhy_29way_logOdds annotation provided by ANNOVAR"
##INFO=ID=SiPhy_29way_logOdds_rankscore,Number=.,Type=String,Description="SiPhy_29way_logOdds_rankscore annotation provided by ANNOVAR"
##INFO=ID=Interpro_domain,Number=.,Type=String,Description="Interpro_domain annotation provided by ANNOVAR"
##INFO=ID=GTEx_V6p_gene,Number=.,Type=String,Description="GTEx_V6p_gene annotation provided by ANNOVAR"
##INFO=ID=GTEx_V6p_tissue,Number=.,Type=String,Description="GTEx_V6p_tissue annotation provided by ANNOVAR"
##INFO=ID=dbscSNV_ADA_SCORE,Number=.,Type=String,Description="dbscSNV_ADA_SCORE annotation provided by ANNOVAR"
##INFO=ID=dbscSNV_RF_SCORE,Number=.,Type=String,Description="dbscSNV_RF_SCORE annotation provided by ANNOVAR"
##INFO=ID=gnomAD_exome_ALL,Number=.,Type=String,Description="gnomAD_exome_ALL annotation provided by ANNOVAR"
##INFO=ID=gnomAD_exome_AFR,Number=.,Type=String,Description="gnomAD_exome_AFR annotation provided by ANNOVAR"
##INFO=ID=gnomAD_exome_AMR,Number=.,Type=String,Description="gnomAD_exome_AMR annotation provided by ANNOVAR"
##INFO=ID=gnomAD_exome_ASJ,Number=.,Type=String,Description="gnomAD_exome_ASJ annotation provided by ANNOVAR"
##INFO=ID=gnomAD_exome_EAS,Number=.,Type=String,Description="gnomAD_exome_EAS annotation provided by ANNOVAR"
##INFO=ID=gnomAD_exome_FIN,Number=.,Type=String,Description="gnomAD_exome_FIN annotation provided by ANNOVAR"
##INFO=ID=gnomAD_exome_NFE,Number=.,Type=String,Description="gnomAD_exome_NFE annotation provided by ANNOVAR"
##INFO=ID=gnomAD_exome_OTH,Number=.,Type=String,Description="gnomAD_exome_OTH annotation provided by ANNOVAR"
##INFO=ID=gnomAD_exome_SAS,Number=.,Type=String,Description="gnomAD_exome_SAS annotation provided by ANNOVAR"
##INFO=ID=gnomAD_genome_ALL,Number=.,Type=String,Description="gnomAD_genome_ALL annotation provided by ANNOVAR"
##INFO=ID=gnomAD_genome_AFR,Number=.,Type=String,Description="gnomAD_genome_AFR annotation provided by ANNOVAR"
##INFO=ID=gnomAD_genome_AMR,Number=.,Type=String,Description="gnomAD_genome_AMR annotation provided by ANNOVAR"
##INFO=ID=gnomAD_genome_ASJ,Number=.,Type=String,Description="gnomAD_genome_ASJ annotation provided by ANNOVAR"
##INFO=ID=gnomAD_genome_EAS,Number=.,Type=String,Description="gnomAD_genome_EAS annotation provided by ANNOVAR"
##INFO=ID=gnomAD_genome_FIN,Number=.,Type=String,Description="gnomAD_genome_FIN annotation provided by ANNOVAR"
##INFO=ID=gnomAD_genome_NFE,Number=.,Type=String,Description="gnomAD_genome_NFE annotation provided by ANNOVAR"
##INFO=ID=gnomAD_genome_OTH,Number=.,Type=String,Description="gnomAD_genome_OTH annotation provided by ANNOVAR"
##INFO=ID=InterVar_automated,Number=.,Type=String,Description="InterVar_automated annotation provided by ANNOVAR"
##INFO=ID=PVS1,Number=.,Type=String,Description="PVS1 annotation provided by ANNOVAR"
##INFO=ID=PS1,Number=.,Type=String,Description="PS1 annotation provided by ANNOVAR"
##INFO=ID=PS2,Number=.,Type=String,Description="PS2 annotation provided by ANNOVAR"
##INFO=ID=PS3,Number=.,Type=String,Description="PS3 annotation provided by ANNOVAR"
##INFO=ID=PS4,Number=.,Type=String,Description="PS4 annotation provided by ANNOVAR"
##INFO=ID=PM1,Number=.,Type=String,Description="PM1 annotation provided by ANNOVAR"
##INFO=ID=PM2,Number=.,Type=String,Description="PM2 annotation provided by ANNOVAR"
##INFO=ID=PM3,Number=.,Type=String,Description="PM3 annotation provided by ANNOVAR"
##INFO=ID=PM4,Number=.,Type=String,Description="PM4 annotation provided by ANNOVAR"
##INFO=ID=PM5,Number=.,Type=String,Description="PM5 annotation provided by ANNOVAR"
##INFO=ID=PM6,Number=.,Type=String,Description="PM6 annotation provided by ANNOVAR"
##INFO=ID=PP1,Number=.,Type=String,Description="PP1 annotation provided by ANNOVAR"
##INFO=ID=PP2,Number=.,Type=String,Description="PP2 annotation provided by ANNOVAR"
##INFO=ID=PP3,Number=.,Type=String,Description="PP3 annotation provided by ANNOVAR"
##INFO=ID=PP4,Number=.,Type=String,Description="PP4 annotation provided by ANNOVAR"
##INFO=ID=PP5,Number=.,Type=String,Description="PP5 annotation provided by ANNOVAR"
##INFO=ID=BA1,Number=.,Type=String,Description="BA1 annotation provided by ANNOVAR"
##INFO=ID=BS1,Number=.,Type=String,Description="BS1 annotation provided by ANNOVAR"
##INFO=ID=BS2,Number=.,Type=String,Description="BS2 annotation provided by ANNOVAR"
##INFO=ID=BS3,Number=.,Type=String,Description="BS3 annotation provided by ANNOVAR"
##INFO=ID=BS4,Number=.,Type=String,Description="BS4 annotation provided by ANNOVAR"
##INFO=ID=BP1,Number=.,Type=String,Description="BP1 annotation provided by ANNOVAR"
##INFO=ID=BP2,Number=.,Type=String,Description="BP2 annotation provided by ANNOVAR"
##INFO=ID=BP3,Number=.,Type=String,Description="BP3 annotation provided by ANNOVAR"
##INFO=ID=BP4,Number=.,Type=String,Description="BP4 annotation provided by ANNOVAR"
##INFO=ID=BP5,Number=.,Type=String,Description="BP5 annotation provided by ANNOVAR"
##INFO=ID=BP6,Number=.,Type=String,Description="BP6 annotation provided by ANNOVAR"
##INFO=ID=BP7,Number=.,Type=String,Description="BP7 annotation provided by ANNOVAR"
##INFO=ID=spliceai_filtered,Number=.,Type=String,Description="spliceai_filtered annotation provided by ANNOVAR"
##INFO=ID=dpsi_max_tissue,Number=.,Type=String,Description="dpsi_max_tissue annotation provided by ANNOVAR"
##INFO=ID=dpsi_zscore,Number=.,Type=String,Description="dpsi_zscore annotation provided by ANNOVAR"
##INFO=ID=PopFreqMax,Number=.,Type=String,Description="PopFreqMax annotation provided by ANNOVAR"
##INFO=ID=ALLELE_END,Number=0,Type=Flag,Description="Flag the end of ANNOVAR annotation for one alternative allele"
##INFO=ID=MPA_adjusted,Number=.,Type=String,Description="MPA_adjusted : normalize MPA missense score from 0 to 10"
##INFO=ID=MPA_available,Number=.,Type=String,Description="MPA_available : number of missense tools annotation available for this variant"
##INFO=ID=MPA_deleterious,Number=.,Type=String,Description="MPA_deleterious : number of missense tools that annotate this variant pathogenic"
##INFO=ID=MPA_final_score,Number=.,Type=String,Description="MPA_final_score : unique score that take into account curated database, biological assumptions, splicing predictions and the sum of various predictors for missense alterations. Annotations are made for exonic and splicing variants up to +300nt."
##INFO=ID=MPA_impact,Number=.,Type=String,Description="MPA_impact : pathogenic predictions (clinvar_pathogenicity, splice_impact, stop and frameshift_impact)"
##INFO=ID=MPA_ranking,Number=.,Type=String,Description="MPA_ranking : prioritize variants with ranks from 1 to 10"
##FORMAT=ID=AD,Number=R,Type=Integer,Description="Allelic depths for the ref and alt alleles in the order listed"
##FORMAT=ID=DP,Number=1,Type=Integer,Description="Approximate read depth (reads with MQ=255 or with bad mates are filtered)"
##FORMAT=ID=GQ,Number=1,Type=Integer,Description="Genotype Quality"
##FORMAT=ID=GT,Number=1,Type=String,Description="Genotype"
##FORMAT=ID=PL,Number=G,Type=Integer,Description="Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification"
##FILTER=ID=LowCoverage,Description="DP  20"
##FILTER=ID=LowMappingQuality,Description="MQ  10.00"
##FILTER=ID=LowQD,Description="QD  1.5"
##FILTER=ID=LowQual,Description="Low quality"
##FILTER=ID=PASS,Description="All filters passed"
##FILTER=ID=R8,Description="POLYX  7"
##FILTER=ID=StrandBias,Description="FS  60.000"
##contig=ID=chr1,length=249250621
##contig=ID=chr2,length=243199373
##contig=ID=chr3,length=198022430
##contig=ID=chr4,length=191154276
##contig=ID=chr5,length=180915260
##contig=ID=chr6,length=171115067
##contig=ID=chr7,length=159138663
##contig=ID=chr8,length=146364022
##contig=ID=chr9,length=141213431
##contig=ID=chr10,length=135534747
##contig=ID=chr11,length=135006516
##contig=ID=chr12,length=133851895
##contig=ID=chr13,length=115169878
##contig=ID=chr14,length=107349540
##contig=ID=chr15,length=102531392
##contig=ID=chr16,length=90354753
##contig=ID=chr17,length=81195210
##contig=ID=chr18,length=78077248
##contig=ID=chr19,length=59128983
##contig=ID=chr20,length=63025520
##contig=ID=chr21,length=48129895
##contig=ID=chr22,length=51304566
##contig=ID=chrX,length=155270560
##contig=ID=chrY,length=59373566
##contig=ID=chrM,length=16571
#CHROM	POS	ID	REF	ALT	QUAL	FILTER	INFO	FORMAT	B00GMSH	B00GMSI	B00GMSJ
</div><div id="ALL" class="tabcontent">
	<table id='table' class="display compact" >
		<thead><tr class="head">	<th >MPA_ranking	</th>
	<th >MPA_impact	</th>
	<th >Phenolyzer	</th>
	<th >Gene.refGene	</th>
	<th >Phenotypes.refGene	</th>
	<th >gnomAD_genome_ALL	</th>
	<th >gnomAD_exome_ALL	</th>
	<th >CLNSIG	</th>
	<th >InterVar_automated	</th>
	<th >SecondHit-CNV	</th>
	<th >Func.refGene	</th>
	<th >Genotype-B00GMSH	</th>
	<th >Genotype-B00GMSI	</th>
	<th >Genotype-B00GMSJ	</th>
	<th >#CHROM	</th>
	<th >POS	</th>
	<th >ID	</th>
	<th >REF	</th>
	<th >ALT	</th>
	<th >FILTER	</th>
</tr>
</thead>
<tbody>
<tr class="ALL" >
	<td ><div class="tooltip">1<span class="tooltiptext tooltip-bottom">MPA_ranking	= 3
MPA_final_score	= 10
MPA_impact	= splice_impact
MPA_adjusted	= 0
MPA_available	= 0
MPA_deleterious	= 0

--- SPLICE ---
dbscSNV_ADA_SCORE	= 0.9954
dbscSNV_RF_SCORE	= 0.75
spliceai_filtered	 DS_AG=0.0000
spliceai_filtered	 DS_AL=0.0000
spliceai_filtered	 DS_DG=0.0008
spliceai_filtered	 DS_DL=0.4060

--- MISSENSE ---
LRT_pred	= .
SIFT_pred	= .
FATHMM_pred	= .
MetaLR_pred	= .
MetaSVM_pred	= .
PROVEAN_pred	= .
Polyphen2_HDIV_pred	= .
Polyphen2_HVAR_pred	= .
MutationTaster_pred	= .
fathmm-MKL_coding_pred	= .
</span></div>	</td>
	<td><div class="tooltip">splice_impact</div>	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom"></span></div> 	</td>
	<td style="background-color:#7FFF00"><div class="tooltip">C1orf159<span class="tooltiptext tooltip-bottom">LOEUF = 1.302
LOEUF_decile = 7.0

oe_lof = 0.69466
oe_lof_lower = 0.394

Missense_upper = 1.016

</span></div>    	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom">Disease_description.refGene:
 .

</span></div> 	</td>
	<td ><div class="tooltip">0.0053<span class="tooltiptext tooltip-bottom">gnomAD_genome_ALL	= 0.0053
gnomAD_genome_AFR	= 0.0014
gnomAD_genome_AMR	= 0.0024
gnomAD_genome_ASJ	= 0
gnomAD_genome_EAS	= 0
gnomAD_genome_FIN	= 0.0052
gnomAD_genome_NFE	= 0.0084
gnomAD_genome_OTH	= 0.0051
</span></div> 	</td>
	<td ><div class="tooltip">0.0047<span class="tooltiptext tooltip-bottom">gnomAD_exome_ALL	= 0.0047
gnomAD_exome_AFR	= 0.0017
gnomAD_exome_AMR	= 0.0017
gnomAD_exome_ASJ	= 0.0041
gnomAD_exome_EAS	= 0
gnomAD_exome_FIN	= 0.0091
gnomAD_exome_NFE	= 0.0083
gnomAD_exome_OTH	= 0.0049
</span></div> 	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom">CLNREVSTAT:
 .

CLNDN:
 .

CLNALLELEID:
 .

CLNDISDB:
 .

</span></div> 	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom"></span></div> 	</td>
	<td><div class="tooltip">.</div>	</td>
	<td ><div class="tooltip">splicing;intronic<span class="tooltiptext tooltip-bottom">ExonicFunc.refGene:
 .

AAChange.refGene:
 .

GeneDetail.refGene:
 NM_017891:exon2:UTR5
 NM_001330306:exon3:UTR5

</span></div> 	</td>
	<td style="background-color:inherit"><div class="tooltip">0/1<span class="tooltiptext tooltip-bottom">CALLER = GATK	 QUALITY = 2523.16

Genotype-B00GMSH	 -	 0/1
DP = 96	 AD = 52,44	 AB = 0.458

Genotype-B00GMSI	 -	 0/0
DP = 110	 AD = 109,1	 AB = 0.009

Genotype-B00GMSJ	 -	 0/1
DP = 105	 AD = 61,44	 AB = 0.419

</span></div>	</td>
	<td style="background-color:inherit"><div class="tooltip">0/0</div>	</td>
	<td style="background-color:inherit"><div class="tooltip">0/1</div>	</td>
	<td><div class="tooltip">chr1</div>	</td>
	<td><div class="tooltip">1027366</div>	</td>
	<td><div class="tooltip">rs143221733</div>	</td>
	<td><div class="tooltip">C</div>	</td>
	<td><div class="tooltip">T</div>	</td>
	<td><div class="tooltip">PASS</div>	</td>
</tr>
<tr class="ALL" >
	<td ><div class="tooltip">2<span class="tooltiptext tooltip-bottom">MPA_ranking	= 5
MPA_final_score	= 10.0
MPA_impact	= missense_impact
MPA_adjusted	= 10.0
MPA_available	= 10
MPA_deleterious	= 10

--- SPLICE ---
dbscSNV_ADA_SCORE	= .
dbscSNV_RF_SCORE	= .
spliceai_filtered	= .

--- MISSENSE ---
LRT_pred	= D
SIFT_pred	= D
FATHMM_pred	= D
MetaLR_pred	= D
MetaSVM_pred	= D
PROVEAN_pred	= D
Polyphen2_HDIV_pred	= D
Polyphen2_HVAR_pred	= D
MutationTaster_pred	= D
fathmm-MKL_coding_pred	= D
</span></div>	</td>
	<td><div class="tooltip">10_missense_impact</div>	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom"></span></div> 	</td>
	<td style="background-color:#00FF00"><div class="tooltip">PLA2G2F<span class="tooltiptext tooltip-bottom">LOEUF = 1.817
LOEUF_decile = 9.0

oe_lof = 1.1229
oe_lof_lower = 0.651

Missense_upper = 1.237

Function Description:
FUNCTION:_May_play_a_role_in_lipid_mediator_production_in_inflammatory_conditions,_by_providing_arachidonic_acid_to_downstream_cyclooxygenases_and_lipoxygenases_(By_similarity)._Phospholipase_A2,_which_catalyzes_the_calcium_dependent_hydrolysis_of_the_2_acyl_groups_in_3_sn_phosphoglycerides_(PubMed:11112443)._Hydrolyzes_phosphatidylethanolamine_more_efficiently_than_phosphatidylcholine,_with_only_a_modest_preference_for_arachidonic_acid_versus_linoelic_acid_at_the_sn_2_position._Comparable_activity_toward_1_palmitoyl_2_oleoyl_phosphatidylserine_vesicles_to_that_toward_1_palmitoyl_2_oleoyl_phosphatidylglycerol_(By_similarity)._Hydrolyzes_phosphatidylglycerol_versus_phosphatidylcholine_with_a_15_fold_preference_(PubMed:11112443)._{ECO:0000250|UniProtKB:Q9QZT4,_ECO:0000269|PubMed:11112443}.

Tissue specificity:
TISSUE_SPECIFICITY:_Expressed_at_high_levels_in_placenta,_testis,_thymus_and_at_lower_levels_in_heart,_kidney,_liver_and_prostate_(PubMed:11112443)._Highly_expressed_in_rheumatoid_arthritic_tissues,_including_synovial_lining_cells_in_the_intima,_capillary_endothelial_cells_and_plasma_cells_(PubMed:11877435)._{ECO:0000269|PubMed:11112443,_ECO:0000269|PubMed:11877435}.
</span></div>    	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom">Disease_description.refGene:
 .

</span></div> 	</td>
	<td ><div class="tooltip">3.233e-05<span class="tooltiptext tooltip-bottom">gnomAD_genome_ALL	= 3.233e-05
gnomAD_genome_AFR	= 0
gnomAD_genome_AMR	= 0
gnomAD_genome_ASJ	= 0
gnomAD_genome_EAS	= 0
gnomAD_genome_FIN	= 0
gnomAD_genome_NFE	= 0
gnomAD_genome_OTH	= 0.0010
</span></div> 	</td>
	<td ><div class="tooltip">4.068e-06<span class="tooltiptext tooltip-bottom">gnomAD_exome_ALL	= 4.068e-06
gnomAD_exome_AFR	= 0
gnomAD_exome_AMR	= 0
gnomAD_exome_ASJ	= 0
gnomAD_exome_EAS	= 0
gnomAD_exome_FIN	= 0
gnomAD_exome_NFE	= 8.982e-06
gnomAD_exome_OTH	= 0
</span></div> 	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom">CLNREVSTAT:
 .

CLNDN:
 .

CLNALLELEID:
 .

CLNDISDB:
 .

</span></div> 	</td>
	<td ><div class="tooltip">Uncertain_significance<span class="tooltiptext tooltip-bottom">PM1	= 1
Located in a mutational hot spot and/or critical and well-established functional domain (e.g., active site of an enzyme) without benign variation

PP3	= 1
Multiple lines of computational evidence support a deleterious effect on the gene or gene product (conservation, evolutionary, splicing impact, etc.) sfit for conservation, GERP++_RS for evolutionary, splicing impact from dbNSFP

</span></div> 	</td>
	<td><div class="tooltip">.</div>	</td>
	<td ><div class="tooltip">exonic<span class="tooltiptext tooltip-bottom">ExonicFunc.refGene:
 nonsynonymous_SNV

AAChange.refGene:
 PLA2G2F:NM_022819:exon3:c.G262A:p.G88S
 PLA2G2F:NM_022819:exon3:c.262G>A:p.G88S

GeneDetail.refGene:
 .

</span></div> 	</td>
	<td style="background-color:inherit"><div class="tooltip">0/1<span class="tooltiptext tooltip-bottom">CALLER = GATK	 QUALITY = 2455.16

Genotype-B00GMSH	 -	 0/1
DP = 73	 AD = 31,42	 AB = 0.575

Genotype-B00GMSI	 -	 0/0
DP = 59	 AD = 59,0	 AB = 0

Genotype-B00GMSJ	 -	 0/1
DP = 82	 AD = 43,39	 AB = 0.475

</span></div>	</td>
	<td style="background-color:inherit"><div class="tooltip">0/0</div>	</td>
	<td style="background-color:inherit"><div class="tooltip">0/1</div>	</td>
	<td><div class="tooltip">chr1</div>	</td>
	<td><div class="tooltip">20470031</div>	</td>
	<td><div class="tooltip">.</div>	</td>
	<td><div class="tooltip">G</div>	</td>
	<td><div class="tooltip">A</div>	</td>
	<td><div class="tooltip">PASS</div>	</td>
</tr>
<tr class="ALL OMIMREC  disease2" >
	<td ><div class="tooltip">3<span class="tooltiptext tooltip-bottom">MPA_ranking	= 5
MPA_final_score	= 9.0
MPA_impact	= missense_impact
MPA_adjusted	= 9.0
MPA_available	= 10
MPA_deleterious	= 9

--- SPLICE ---
dbscSNV_ADA_SCORE	= .
dbscSNV_RF_SCORE	= .
spliceai_filtered	= .

--- MISSENSE ---
LRT_pred	= D
SIFT_pred	= D
FATHMM_pred	= T
MetaLR_pred	= D
MetaSVM_pred	= D
PROVEAN_pred	= D
Polyphen2_HDIV_pred	= D
Polyphen2_HVAR_pred	= D
MutationTaster_pred	= D
fathmm-MKL_coding_pred	= D
</span></div>	</td>
	<td><div class="tooltip">9_missense_impact</div>	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom"></span></div> 	</td>
	<td style="background-color:#FFFF00"><div class="tooltip">H6PD<span class="tooltiptext tooltip-bottom">LOEUF = 0.941
LOEUF_decile = 5.0

oe_lof = 0.6271399999999999
oe_lof_lower = 0.428

Missense_upper = 1.23

Function Description:
FUNCTION:_Bifunctional_enzyme_localized_in_the_lumen_of_the_endoplasmic_reticulum_that_catalyzes_the_first_two_steps_of_the_oxidative_branch_of_the_pentose_phosphate_pathway/shunt,_an_alternative_to_glycolysis_and_a_major_source_of_reducing_power_and_metabolic_intermediates_for_biosynthetic_processes_(By_similarity)._Has_a_hexose_6_phosphate_dehydrogenase_activity,_with_broad_substrate_specificity_compared_to_glucose_6_phosphate_1_dehydrogenase/G6PD,_and_catalyzes_the_first_step_of_the_pentose_phosphate_pathway_(PubMed:12858176,_PubMed:18628520,_PubMed:23132696)._In_addition,_acts_as_a_6_phosphogluconolactonase_and_catalyzes_the_second_step_of_the_pentose_phosphate_pathway_(By_similarity)._May_have_a_dehydrogenase_activity_for_alternative_substrates_including_glucosamine_6_phosphate_and_glucose_6_sulfate_(By_similarity)._The_main_function_of_this_enzyme_is_to_provide_reducing_equivalents_such_as_NADPH_to_maintain_the_adequate_levels_of_reductive_cofactors_in_the_oxidizing_environment_of_the_endoplasmic_reticulum_(PubMed:12858176,_PubMed:18628520,_PubMed:23132696)._By_producing_NADPH_that_is_needed_by_reductases_of_the_lumen_of_the_endoplasmic_reticulum_like_corticosteroid_11_beta_dehydrogenase_isozyme_1/HSD11B1,_indirectly_regulates_their_activity_(PubMed:18628520)._{ECO:0000250|UniProtKB:Q8CFX1,_ECO:0000269|PubMed:12858176,_ECO:0000269|PubMed:18628520,_ECO:0000269|PubMed:23132696}.

Tissue specificity:
TISSUE_SPECIFICITY:_Present_in_most_tissues_examined,_strongest_in_liver._{ECO:0000269|PubMed:10349511}.
</span></div>    	</td>
	<td ><div class="tooltip">Cortisone_reductase_deficiency_1,_604931__3_,_Autosomal_recessive<span class="tooltiptext tooltip-bottom">Disease_description.refGene:
 

DISEASE:_Cortisone_reductase_deficiency_1_(CORTRD1)_[MIM:604931]:_An_autosomal_recessive_error_of_cortisone_metabolism_characterized_by_a_failure_to_regenerate_cortisol_from_cortisone,_resulting_in_increased_cortisol_clearance,_activation_of_the_hypothalamic_pituitary_axis_and_ACTH_mediated_adrenal_androgen_excess._Clinical_features_include_hyperandrogenism_resulting_in_hirsutism,_oligo_amenorrhea,_and_infertility_in_females_and_premature_pseudopuberty_in_males._{ECO:0000269|PubMed:12858176,_ECO:0000269|PubMed:17974005,_ECO:0000269|PubMed:18628520,_ECO:0000269|PubMed:23132696}._Note=The_disease_is_caused_by_mutations_affecting_the_gene_represented_in_this_entry.

</span></div> 	</td>
	<td ><div class="tooltip">3.233e-05<span class="tooltiptext tooltip-bottom">gnomAD_genome_ALL	= 3.233e-05
gnomAD_genome_AFR	= 0.0001
gnomAD_genome_AMR	= 0
gnomAD_genome_ASJ	= 0
gnomAD_genome_EAS	= 0
gnomAD_genome_FIN	= 0
gnomAD_genome_NFE	= 0
gnomAD_genome_OTH	= 0
</span></div> 	</td>
	<td ><div class="tooltip">8.577e-06<span class="tooltiptext tooltip-bottom">gnomAD_exome_ALL	= 8.577e-06
gnomAD_exome_AFR	= 7.179e-05
gnomAD_exome_AMR	= 0
gnomAD_exome_ASJ	= 0
gnomAD_exome_EAS	= 0
gnomAD_exome_FIN	= 0
gnomAD_exome_NFE	= 9.54e-06
gnomAD_exome_OTH	= 0
</span></div> 	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom">CLNREVSTAT:
 .

CLNDN:
 .

CLNALLELEID:
 .

CLNDISDB:
 .

</span></div> 	</td>
	<td ><div class="tooltip">Uncertain_significance<span class="tooltiptext tooltip-bottom">PM1	= 1
Located in a mutational hot spot and/or critical and well-established functional domain (e.g., active site of an enzyme) without benign variation

PP3	= 1
Multiple lines of computational evidence support a deleterious effect on the gene or gene product (conservation, evolutionary, splicing impact, etc.) sfit for conservation, GERP++_RS for evolutionary, splicing impact from dbNSFP

PM2	= 1
Absent from controls (or at extremely low frequency if recessive) (Table 6) in Exome Sequencing Project, 1000 Genomes Project, or Exome Aggregation Consortium

</span></div> 	</td>
	<td><div class="tooltip">.</div>	</td>
	<td ><div class="tooltip">exonic<span class="tooltiptext tooltip-bottom">ExonicFunc.refGene:
 nonsynonymous_SNV

AAChange.refGene:
 H6PD:NM_001282587:exon5:c.C1802T:p.S601L
 H6PD:NM_004285:exon5:c.C1769T:p.S590L
 H6PD:NM_001282587:exon5:c.1802C>T:p.S601L
 H6PD:NM_004285:exon5:c.1769C>T:p.S590L

GeneDetail.refGene:
 .

</span></div> 	</td>
	<td style="background-color:inherit"><div class="tooltip">0/1<span class="tooltiptext tooltip-bottom">CALLER = GATK	 QUALITY = 3859.16

Genotype-B00GMSH	 -	 0/1
DP = 100	 AD = 48,52	 AB = 0.52

Genotype-B00GMSI	 -	 0/1
DP = 143	 AD = 71,72	 AB = 0.503

Genotype-B00GMSJ	 -	 0/0
DP = 137	 AD = 137,0	 AB = 0

</span></div>	</td>
	<td style="background-color:inherit"><div class="tooltip">0/1</div>	</td>
	<td style="background-color:inherit"><div class="tooltip">0/0</div>	</td>
	<td><div class="tooltip">chr1</div>	</td>
	<td><div class="tooltip">9324321</div>	</td>
	<td><div class="tooltip">.</div>	</td>
	<td><div class="tooltip">C</div>	</td>
	<td><div class="tooltip">T</div>	</td>
	<td><div class="tooltip">PASS</div>	</td>
</tr>
<tr class="ALL" >
	<td ><div class="tooltip">4<span class="tooltiptext tooltip-bottom">MPA_ranking	= 5
MPA_final_score	= 7.0
MPA_impact	= missense_impact
MPA_adjusted	= 7.0
MPA_available	= 10
MPA_deleterious	= 7

--- SPLICE ---
dbscSNV_ADA_SCORE	= .
dbscSNV_RF_SCORE	= .
spliceai_filtered	= .

--- MISSENSE ---
LRT_pred	= D
SIFT_pred	= D
FATHMM_pred	= T
MetaLR_pred	= T
MetaSVM_pred	= T
PROVEAN_pred	= D
Polyphen2_HDIV_pred	= D
Polyphen2_HVAR_pred	= D
MutationTaster_pred	= D
fathmm-MKL_coding_pred	= D
</span></div>	</td>
	<td><div class="tooltip">7_missense_impact</div>	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom"></span></div> 	</td>
	<td style="background-color:#FF9900"><div class="tooltip">EPHA8<span class="tooltiptext tooltip-bottom">LOEUF = 0.712
LOEUF_decile = 3.0

oe_lof = 0.506
oe_lof_lower = 0.365

Missense_upper = 0.987

Function Description:
FUNCTION:_Receptor_tyrosine_kinase_which_binds_promiscuously_GPI_anchored_ephrin_A_family_ligands_residing_on_adjacent_cells,_leading_to_contact_dependent_bidirectional_signaling_into_neighboring_cells._The_signaling_pathway_downstream_of_the_receptor_is_referred_to_as_forward_signaling_while_the_signaling_pathway_downstream_of_the_ephrin_ligand_is_referred_to_as_reverse_signaling._The_GPI_anchored_ephrin_A_EFNA2,_EFNA3,_and_EFNA5_are_able_to_activate_EPHA8_through_phosphorylation._With_EFNA5_may_regulate_integrin_mediated_cell_adhesion_and_migration_on_fibronectin_substrate_but_also_neurite_outgrowth._During_development_of_the_nervous_system_plays_also_a_role_in_axon_guidance._Downstream_effectors_of_the_EPHA8_signaling_pathway_include_FYN_which_promotes_cell_adhesion_upon_activation_by_EPHA8_and_the_MAP_kinases_in_the_stimulation_of_neurite_outgrowth_(By_similarity)._{ECO:0000250}.;FUNCTION:_Receptor_tyrosine_kinase_which_binds_promiscuously_GPI_anchored_ephrin_A_family_ligands_residing_on_adjacent_cells,_leading_to_contact_dependent_bidirectional_signaling_into_neighboring_cells._The_signaling_pathway_downstream_of_the_receptor_is_referred_to_as_forward_signaling_while_the_signaling_pathway_downstream_of_the_ephrin_ligand_is_referred_to_as_reverse_signaling._The_GPI_anchored_ephrin_A_EFNA2,_EFNA3,_and_EFNA5_are_able_to_activate_EPHA8_through_phosphorylation._With_EFNA5_may_regulate_integrin_mediated_cell_adhesion_and_migration_on_fibronectin_substrate_but_also_neurite_outgrowth._During_development_of_the_nervous_system_plays_also_a_role_in_axon_guidance._Downstream_effectors_of_the_EPHA8_signaling_pathway_include_FYN_which_promotes_cell_adhesion_upon_activation_by_EPHA8_and_the_MAP_kinases_in_the_stimulation_of_neurite_outgrowth_(By_similarity)._{ECO:0000250}.

</span></div>    	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom">Disease_description.refGene:
 .

</span></div> 	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom">gnomAD_genome_ALL	= .
gnomAD_genome_AFR	= .
gnomAD_genome_AMR	= .
gnomAD_genome_ASJ	= .
gnomAD_genome_EAS	= .
gnomAD_genome_FIN	= .
gnomAD_genome_NFE	= .
gnomAD_genome_OTH	= .
</span></div> 	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom">gnomAD_exome_ALL	= .
gnomAD_exome_AFR	= .
gnomAD_exome_AMR	= .
gnomAD_exome_ASJ	= .
gnomAD_exome_EAS	= .
gnomAD_exome_FIN	= .
gnomAD_exome_NFE	= .
gnomAD_exome_OTH	= .
</span></div> 	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom">CLNREVSTAT:
 .

CLNDN:
 .

CLNALLELEID:
 .

CLNDISDB:
 .

</span></div> 	</td>
	<td ><div class="tooltip">Uncertain_significance<span class="tooltiptext tooltip-bottom">PM1	= 1
Located in a mutational hot spot and/or critical and well-established functional domain (e.g., active site of an enzyme) without benign variation

PM2	= 1
Absent from controls (or at extremely low frequency if recessive) (Table 6) in Exome Sequencing Project, 1000 Genomes Project, or Exome Aggregation Consortium

</span></div> 	</td>
	<td><div class="tooltip">.</div>	</td>
	<td ><div class="tooltip">exonic<span class="tooltiptext tooltip-bottom">ExonicFunc.refGene:
 nonsynonymous_SNV

AAChange.refGene:
 EPHA8:NM_020526:exon10:c.A1892G:p.E631G
 EPHA8:NM_020526:exon10:c.1892A>G:p.E631G

GeneDetail.refGene:
 .

</span></div> 	</td>
	<td style="background-color:inherit"><div class="tooltip">0/1<span class="tooltiptext tooltip-bottom">CALLER = GATK	 QUALITY = 3067.16

Genotype-B00GMSH	 -	 0/1
DP = 105	 AD = 53,52	 AB = 0.495

Genotype-B00GMSI	 -	 0/0
DP = 138	 AD = 138,0	 AB = 0

Genotype-B00GMSJ	 -	 0/1
DP = 148	 AD = 81,67	 AB = 0.452

</span></div>	</td>
	<td style="background-color:inherit"><div class="tooltip">0/0</div>	</td>
	<td style="background-color:inherit"><div class="tooltip">0/1</div>	</td>
	<td><div class="tooltip">chr1</div>	</td>
	<td><div class="tooltip">22923931</div>	</td>
	<td><div class="tooltip">.</div>	</td>
	<td><div class="tooltip">A</div>	</td>
	<td><div class="tooltip">G</div>	</td>
	<td><div class="tooltip">PASS</div>	</td>
</tr>
<tr class="ALL" >
	<td ><div class="tooltip">5<span class="tooltiptext tooltip-bottom">MPA_ranking	= 5
MPA_final_score	= 7.0
MPA_impact	= missense_impact,splice_impact
MPA_adjusted	= 7.0
MPA_available	= 10
MPA_deleterious	= 7

--- SPLICE ---
dbscSNV_ADA_SCORE	= .
dbscSNV_RF_SCORE	= .
spliceai_filtered	 DS_AG=0.3887
spliceai_filtered	 DS_AL=0.0000
spliceai_filtered	 DS_DG=0.0000
spliceai_filtered	 DS_DL=0.0000

--- MISSENSE ---
LRT_pred	= D
SIFT_pred	= D
FATHMM_pred	= T
MetaLR_pred	= T
MetaSVM_pred	= T
PROVEAN_pred	= D
Polyphen2_HDIV_pred	= D
Polyphen2_HVAR_pred	= D
MutationTaster_pred	= D
fathmm-MKL_coding_pred	= D
</span></div>	</td>
	<td><div class="tooltip">7_missense_impact,splice_impact</div>	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom"></span></div> 	</td>
	<td style="background-color:#3FFF00"><div class="tooltip">AKR7A2<span class="tooltiptext tooltip-bottom">LOEUF = 1.522
LOEUF_decile = 8.0

oe_lof = 0.99174
oe_lof_lower = 0.662

Missense_upper = 1.215

Function Description:
FUNCTION:_Catalyzes_the_NADPH_dependent_reduction_of_succinic_semialdehyde_to_gamma_hydroxybutyrate._May_have_an_important_role_in_producing_the_neuromodulator_gamma_hydroxybutyrate_(GHB)._Has_broad_substrate_specificity._Has_NADPH_dependent_aldehyde_reductase_activity_towards_2_carboxybenzaldehyde,_2_nitrobenzaldehyde_and_pyridine_2_aldehyde_(in_vitro)._Can_reduce_1,2_naphthoquinone_and_9,10_phenanthrenequinone_(in_vitro)._Can_reduce_the_dialdehyde_protein_binding_form_of_aflatoxin_B1_(AFB1)_to_the_non_binding_AFB1_dialcohol._May_be_involved_in_protection_of_liver_against_the_toxic_and_carcinogenic_effects_of_AFB1,_a_potent_hepatocarcinogen._{ECO:0000269|PubMed:17591773,_ECO:0000269|PubMed:9576847}.

Tissue specificity:
TISSUE_SPECIFICITY:_Detected_in_brain,_liver,_small_intestine_and_testis,_and_at_lower_levels_in_heart,_prostate,_skeletal_muscle_and_spleen._Detected_in_kidney_proximal_and_distal_tubules,_endothelial_cells_lining_the_Bowman's_capsules_and_some_cysts._Detected_at_low_levels_in_lung_and_pancreas_(at_protein_level)._Widely_expressed._{ECO:0000269|PubMed:10510318,_ECO:0000269|PubMed:9576847}.
</span></div>    	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom">Disease_description.refGene:
 .

</span></div> 	</td>
	<td ><div class="tooltip">0.0003<span class="tooltiptext tooltip-bottom">gnomAD_genome_ALL	= 0.0003
gnomAD_genome_AFR	= 0.0010
gnomAD_genome_AMR	= 0
gnomAD_genome_ASJ	= 0
gnomAD_genome_EAS	= 0
gnomAD_genome_FIN	= 0
gnomAD_genome_NFE	= 0
gnomAD_genome_OTH	= 0
</span></div> 	</td>
	<td ><div class="tooltip">7.722e-05<span class="tooltiptext tooltip-bottom">gnomAD_exome_ALL	= 7.722e-05
gnomAD_exome_AFR	= 0.0010
gnomAD_exome_AMR	= 0
gnomAD_exome_ASJ	= 0
gnomAD_exome_EAS	= 0
gnomAD_exome_FIN	= 0
gnomAD_exome_NFE	= 8.966e-06
gnomAD_exome_OTH	= 0.0002
</span></div> 	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom">CLNREVSTAT:
 .

CLNDN:
 .

CLNALLELEID:
 .

CLNDISDB:
 .

</span></div> 	</td>
	<td ><div class="tooltip">Uncertain_significance<span class="tooltiptext tooltip-bottom">PM1	= 1
Located in a mutational hot spot and/or critical and well-established functional domain (e.g., active site of an enzyme) without benign variation

</span></div> 	</td>
	<td><div class="tooltip">.</div>	</td>
	<td ><div class="tooltip">exonic<span class="tooltiptext tooltip-bottom">ExonicFunc.refGene:
 nonsynonymous_SNV

AAChange.refGene:
 AKR7A2:NM_001320979:exon2:c.C358T:p.R120W
 AKR7A2:NM_003689:exon2:c.C358T:p.R120W
 AKR7A2:NM_001320979:exon2:c.358C>T:p.R120W
 AKR7A2:NM_003689:exon2:c.358C>T:p.R120W

GeneDetail.refGene:
 .

</span></div> 	</td>
	<td style="background-color:inherit"><div class="tooltip">0/1<span class="tooltiptext tooltip-bottom">CALLER = GATK	 QUALITY = 2819.16

Genotype-B00GMSH	 -	 0/1
DP = 111	 AD = 54,57	 AB = 0.513

Genotype-B00GMSI	 -	 0/1
DP = 118	 AD = 76,42	 AB = 0.355

Genotype-B00GMSJ	 -	 0/0
DP = 147	 AD = 147,0	 AB = 0

</span></div>	</td>
	<td style="background-color:inherit"><div class="tooltip">0/1</div>	</td>
	<td style="background-color:inherit"><div class="tooltip">0/0</div>	</td>
	<td><div class="tooltip">chr1</div>	</td>
	<td><div class="tooltip">19635077</div>	</td>
	<td><div class="tooltip">rs146046785</div>	</td>
	<td><div class="tooltip">G</div>	</td>
	<td><div class="tooltip">A</div>	</td>
	<td><div class="tooltip">PASS</div>	</td>
</tr>
<tr class="ALL OMIMDOM OMIMREC  PATHO1" >
	<td ><div class="tooltip">6<span class="tooltiptext tooltip-bottom">MPA_ranking	= 7
MPA_final_score	= 6.0
MPA_impact	= missense_impact
MPA_adjusted	= 6.0
MPA_available	= 10
MPA_deleterious	= 6

--- SPLICE ---
dbscSNV_ADA_SCORE	= .
dbscSNV_RF_SCORE	= .
spliceai_filtered	= .

--- MISSENSE ---
LRT_pred	= D
SIFT_pred	= T
FATHMM_pred	= D
MetaLR_pred	= D
MetaSVM_pred	= D
PROVEAN_pred	= N
Polyphen2_HDIV_pred	= B
Polyphen2_HVAR_pred	= B
MutationTaster_pred	= D
fathmm-MKL_coding_pred	= D
</span></div>	</td>
	<td><div class="tooltip">6_missense_impact</div>	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom"></span></div> 	</td>
	<td style="background-color:#FF3300"><div class="tooltip">MFN2<span class="tooltiptext tooltip-bottom">LOEUF = 0.282
LOEUF_decile = 1.0

oe_lof = 0.13395
oe_lof_lower = 0.069

Missense_upper = 0.847

Function Description:
FUNCTION:_Mitochondrial_outer_membrane_GTPase_that_mediates_mitochondrial_clustering_and_fusion_(PubMed:11181170,_PubMed:11950885,_PubMed:28114303)._Mitochondria_are_highly_dynamic_organelles,_and_their_morphology_is_determined_by_the_equilibrium_between_mitochondrial_fusion_and_fission_events_(PubMed:28114303)._Overexpression_induces_the_formation_of_mitochondrial_networks_(PubMed:28114303)._Membrane_clustering_requires_GTPase_activity_and_may_involve_a_major_rearrangement_of_the_coiled_coil_domains_(Probable)._Plays_a_central_role_in_mitochondrial_metabolism_and_may_be_associated_with_obesity_and/or_apoptosis_processes_(By_similarity)._Plays_an_important_role_in_the_regulation_of_vascular_smooth_muscle_cell_proliferation_(By_similarity)._Involved_in_the_clearance_of_damaged_mitochondria_via_selective_autophagy_(mitophagy)_(PubMed:23620051)._Is_required_for_PRKN_recruitment_to_dysfunctional_mitochondria_(PubMed:23620051)._Involved_in_the_control_of_unfolded_protein_response_(UPR)_upon_ER_stress_including_activation_of_apoptosis_and_autophagy_during_ER_stress_(By_similarity)._Acts_as_an_upstream_regulator_of_EIF2AK3_and_suppresses_EIF2AK3_activation_under_basal_conditions_(By_similarity)._{ECO:0000250|UniProtKB:Q80U63,_ECO:0000250|UniProtKB:Q8R500,_ECO:0000269|PubMed:11181170,_ECO:0000269|PubMed:11950885,_ECO:0000269|PubMed:23620051,_ECO:0000269|PubMed:26085578,_ECO:0000269|PubMed:28114303,_ECO:0000305}.

Tissue specificity:
TISSUE_SPECIFICITY:_Ubiquitous;_expressed_at_low_level._Highly_expressed_in_heart_and_kidney._{ECO:0000269|PubMed:11950885,_ECO:0000269|PubMed:12759376}.
</span></div>    	</td>
	<td ><div class="tooltip">Hereditary_motor_and_sensory_neuropathy_VIA,_601152__3_,_Autosomal_dominant;_Charcot_Marie_Tooth_disease,_axonal,_type_2A2B,_617087__3_,_Autosomal_recessive;_Charcot_Marie_Tooth_disease,_axonal,_type_2A2A,_609260__3_,_Autosomal_dominant<span class="tooltiptext tooltip-bottom">Disease_description.refGene:
 

DISEASE:_Charcot_Marie_Tooth_disease_2A2B_(CMT2A2B)_[MIM:617087]:_An_axonal_form_of_Charcot_Marie_Tooth_disease,_a_disorder_of_the_peripheral_nervous_system,_characterized_by_progressive_weakness_and_atrophy,_initially_of_the_peroneal_muscles_and_later_of_the_distal_muscles_of_the_arms._Charcot_Marie_Tooth_disease_is_classified_in_two_main_groups_on_the_basis_of_electrophysiologic_properties_and_histopathology:_primary_peripheral_demyelinating_neuropathies_(designated_CMT1_when_they_are_dominantly_inherited)_and_primary_peripheral_axonal_neuropathies_(CMT2)._Neuropathies_of_the_CMT2_group_are_characterized_by_signs_of_axonal_degeneration_in_the_absence_of_obvious_myelin_alterations,_normal_or_slightly_reduced_nerve_conduction_velocities,_and_progressive_distal_muscle_weakness_and_atrophy._CMT2A2B_is_a_severe_form_with_autosomal_recessive_inheritance._{ECO:0000269|PubMed:18458227,_ECO:0000269|PubMed:20350294,_ECO:0000269|PubMed:21715711,_ECO:0000269|PubMed:26085578,_ECO:0000269|PubMed:26955893}._Note=The_disease_is_caused_by_mutations_affecting_the_gene_represented_in_this_entry.;_

DISEASE:_Charcot_Marie_Tooth_disease_2A2A_(CMT2A2A)_[MIM:609260]:_An_axonal_form_of_Charcot_Marie_Tooth_disease,_a_disorder_of_the_peripheral_nervous_system,_characterized_by_progressive_weakness_and_atrophy,_initially_of_the_peroneal_muscles_and_later_of_the_distal_muscles_of_the_arms._Charcot_Marie_Tooth_disease_is_classified_in_two_main_groups_on_the_basis_of_electrophysiologic_properties_and_histopathology:_primary_peripheral_demyelinating_neuropathies_(designated_CMT1_when_they_are_dominantly_inherited)_and_primary_peripheral_axonal_neuropathies_(CMT2)._Neuropathies_of_the_CMT2_group_are_characterized_by_signs_of_axonal_degeneration_in_the_absence_of_obvious_myelin_alterations,_normal_or_slightly_reduced_nerve_conduction_velocities,_and_progressive_distal_muscle_weakness_and_atrophy._{ECO:0000269|PubMed:15064763,_ECO:0000269|PubMed:15549395,_ECO:0000269|PubMed:16762064,_ECO:0000269|PubMed:20350294,_ECO:0000269|PubMed:22206013}._Note=The_disease_is_caused_by_mutations_affecting_the_gene_represented_in_this_entry.;_

DISEASE:_Neuropathy,_hereditary_motor_and_sensory,_6A,_with_optic_atrophy_(HMSN6A)_[MIM:601152]:_An_autosomal_dominant_neurologic_disorder_characterized_by_optic_atrophy_and_peripheral_sensorimotor_neuropathy_manifesting_as_axonal_Charcot_Marie_Tooth_disease._Charcot_Marie_Tooth_disease_is_a_disorder_of_the_peripheral_nervous_system,_characterized_by_progressive_weakness_and_atrophy,_initially_of_the_peroneal_muscles_and_later_of_the_distal_muscles_of_the_arms._It_is_classified_in_two_main_groups_on_the_basis_of_electrophysiologic_properties_and_histopathology:_primary_peripheral_demyelinating_neuropathies_and_primary_peripheral_axonal_neuropathies._Peripheral_axonal_neuropathies_are_characterized_by_signs_of_axonal_regeneration_in_the_absence_of_obvious_myelin_alterations,_and_normal_or_slightly_reduced_nerve_conduction_velocities._{ECO:0000269|PubMed:16437557,_ECO:0000269|PubMed:20350294,_ECO:0000269|PubMed:24604904,_ECO:0000269|PubMed:26085578}._Note=The_disease_is_caused_by_mutations_affecting_the_gene_represented_in_this_entry.

</span></div> 	</td>
	<td ><div class="tooltip">0.0023<span class="tooltiptext tooltip-bottom">gnomAD_genome_ALL	= 0.0023
gnomAD_genome_AFR	= 0.0002
gnomAD_genome_AMR	= 0
gnomAD_genome_ASJ	= 0
gnomAD_genome_EAS	= 0.0018
gnomAD_genome_FIN	= 0.0003
gnomAD_genome_NFE	= 0.0041
gnomAD_genome_OTH	= 0.0031
</span></div> 	</td>
	<td ><div class="tooltip">0.0021<span class="tooltiptext tooltip-bottom">gnomAD_exome_ALL	= 0.0021
gnomAD_exome_AFR	= 0.0003
gnomAD_exome_AMR	= 0.0012
gnomAD_exome_ASJ	= 0.0047
gnomAD_exome_EAS	= 0.0005
gnomAD_exome_FIN	= 0.0010
gnomAD_exome_NFE	= 0.0034
gnomAD_exome_OTH	= 0.0024
</span></div> 	</td>
	<td ><div class="tooltip">Conflicting_interpretations_of_pathogenicity<span class="tooltiptext tooltip-bottom">CLNREVSTAT:
 criteria_provided,_conflicting_interpretations

CLNDN:
 Charcot-Marie-Tooth_disease,_type_2A2A|Charcot-Marie-Tooth_disease|Charcot-Marie-Tooth_disease,_type_2|Hereditary_motor_and_sensory_neuropathy_with_optic_atrophy|not_specified|not_provided

CLNALLELEID:
 196292

CLNDISDB:
 MONDO:MONDO:0012231,MedGen:C4721887,OMIM:609260,Orphanet:ORPHA99947|MONDO:MONDO:0015626,MedGen:C0007959,Orphanet:ORPHA166,SNOMED_CT:50548001|MONDO:MONDO:0018993,MedGen:C0270914,Orphanet:ORPHA64746|MedGen:C0393807,OMIM:601152,SNOMED_CT:128203003|MedGen:CN169374|MedGen:CN517202

</span></div> 	</td>
	<td ><div class="tooltip">Uncertain_significance<span class="tooltiptext tooltip-bottom">PM1	= 1
Located in a mutational hot spot and/or critical and well-established functional domain (e.g., active site of an enzyme) without benign variation

PP3	= 1
Multiple lines of computational evidence support a deleterious effect on the gene or gene product (conservation, evolutionary, splicing impact, etc.) sfit for conservation, GERP++_RS for evolutionary, splicing impact from dbNSFP

BP6	= 1
Reputable source recently reports variant as benign, but the evidence is not available to the laboratory to perform an independent evaluation; Check the ClinVar column to see whether this is "benign".

PM2	= 1
Absent from controls (or at extremely low frequency if recessive) (Table 6) in Exome Sequencing Project, 1000 Genomes Project, or Exome Aggregation Consortium

</span></div> 	</td>
	<td><div class="tooltip">.</div>	</td>
	<td ><div class="tooltip">exonic<span class="tooltiptext tooltip-bottom">ExonicFunc.refGene:
 nonsynonymous_SNV

AAChange.refGene:
 MFN2:NM_001127660:exon8:c.G892A:p.G298R
 MFN2:NM_014874:exon9:c.G892A:p.G298R
 MFN2:NM_001127660:exon8:c.892G>A:p.G298R
 MFN2:NM_014874:exon9:c.892G>A:p.G298R

GeneDetail.refGene:
 .

</span></div> 	</td>
	<td style="background-color:inherit"><div class="tooltip">0/1<span class="tooltiptext tooltip-bottom">CALLER = GATK	 QUALITY = 1911.16

Genotype-B00GMSH	 -	 0/1
DP = 54	 AD = 27,27	 AB = 0.5

Genotype-B00GMSI	 -	 0/1
DP = 66	 AD = 29,37	 AB = 0.560

Genotype-B00GMSJ	 -	 0/0
DP = 66	 AD = 66,0	 AB = 0

</span></div>	</td>
	<td style="background-color:inherit"><div class="tooltip">0/1</div>	</td>
	<td style="background-color:inherit"><div class="tooltip">0/0</div>	</td>
	<td><div class="tooltip">chr1</div>	</td>
	<td><div class="tooltip">12061533</div>	</td>
	<td><div class="tooltip">rs41278630</div>	</td>
	<td><div class="tooltip">G</div>	</td>
	<td><div class="tooltip">A</div>	</td>
	<td><div class="tooltip">PASS</div>	</td>
</tr>
<tr class="ALL" >
	<td ><div class="tooltip">7<span class="tooltiptext tooltip-bottom">MPA_ranking	= 7
MPA_final_score	= 5.0
MPA_impact	= missense_impact
MPA_adjusted	= 5.0
MPA_available	= 10
MPA_deleterious	= 5

--- SPLICE ---
dbscSNV_ADA_SCORE	= .
dbscSNV_RF_SCORE	= .
spliceai_filtered	= .

--- MISSENSE ---
LRT_pred	= D
SIFT_pred	= D
FATHMM_pred	= T
MetaLR_pred	= T
MetaSVM_pred	= T
PROVEAN_pred	= D
Polyphen2_HDIV_pred	= P
Polyphen2_HVAR_pred	= B
MutationTaster_pred	= D
fathmm-MKL_coding_pred	= D
</span></div>	</td>
	<td><div class="tooltip">5_missense_impact</div>	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom"></span></div> 	</td>
	<td style="background-color:#FF9900"><div class="tooltip">THEMIS2<span class="tooltiptext tooltip-bottom">LOEUF = 0.713
LOEUF_decile = 3.0

oe_lof = 0.39502
oe_lof_lower = 0.231

Missense_upper = 0.902

Function Description:
FUNCTION:_May_constitute_a_control_point_in_macrophage_inflammatory_response,_promoting_LPS_induced_TNF_production._{ECO:0000269|PubMed:20644716}.

Tissue specificity:
TISSUE_SPECIFICITY:_Expressed_in_different_endometrial_adenocarcinoma_cell_lines_and_various_other_cell_lines_apart_from_the_prostate_cell_line_LNCaP_and_the_ovarian_cancer_cell_line_BG1._{ECO:0000269|PubMed:9563507}.
</span></div>    	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom">Disease_description.refGene:
 .

</span></div> 	</td>
	<td ><div class="tooltip">0.0066<span class="tooltiptext tooltip-bottom">gnomAD_genome_ALL	= 0.0066
gnomAD_genome_AFR	= 0.0015
gnomAD_genome_AMR	= 0.0024
gnomAD_genome_ASJ	= 0.0166
gnomAD_genome_EAS	= 0
gnomAD_genome_FIN	= 0.0089
gnomAD_genome_NFE	= 0.0100
gnomAD_genome_OTH	= 0.0041
</span></div> 	</td>
	<td ><div class="tooltip">0.0082<span class="tooltiptext tooltip-bottom">gnomAD_exome_ALL	= 0.0082
gnomAD_exome_AFR	= 0.0018
gnomAD_exome_AMR	= 0.0055
gnomAD_exome_ASJ	= 0.0140
gnomAD_exome_EAS	= 0
gnomAD_exome_FIN	= 0.0093
gnomAD_exome_NFE	= 0.0122
gnomAD_exome_OTH	= 0.0086
</span></div> 	</td>
	<td ><div class="tooltip">Benign<span class="tooltiptext tooltip-bottom">CLNREVSTAT:
 criteria_provided,_single_submitter

CLNDN:
 not_provided

CLNALLELEID:
 696674

CLNDISDB:
 MedGen:CN517202

</span></div> 	</td>
	<td ><div class="tooltip">Uncertain_significance<span class="tooltiptext tooltip-bottom">PM1	= 1
Located in a mutational hot spot and/or critical and well-established functional domain (e.g., active site of an enzyme) without benign variation

BS1	= 1
Allele frequency is greater than expected for disorder (see Table 6) > 1% in ESP6500all ExAc? need to check more

</span></div> 	</td>
	<td><div class="tooltip">.</div>	</td>
	<td ><div class="tooltip">exonic<span class="tooltiptext tooltip-bottom">ExonicFunc.refGene:
 nonsynonymous_SNV

AAChange.refGene:
 THEMIS2:NM_001105556:exon3:c.T331C:p.F111L
 THEMIS2:NM_001286113:exon3:c.T331C:p.F111L
 THEMIS2:NM_001286115:exon3:c.T331C:p.F111L
 THEMIS2:NM_004848:exon3:c.T331C:p.F111L
 THEMIS2:NM_001105556:exon3:c.331T>C:p.F111L
 THEMIS2:NM_001286113:exon3:c.331T>C:p.F111L
 THEMIS2:NM_001286115:exon3:c.331T>C:p.F111L
 THEMIS2:NM_004848:exon3:c.331T>C:p.F111L

GeneDetail.refGene:
 .

</span></div> 	</td>
	<td style="background-color:inherit"><div class="tooltip">0/1<span class="tooltiptext tooltip-bottom">CALLER = GATK	 QUALITY = 9941.16

Genotype-B00GMSH	 -	 0/1
DP = 289	 AD = 134,155	 AB = 0.536

Genotype-B00GMSI	 -	 0/0
DP = 350	 AD = 350,0	 AB = 0

Genotype-B00GMSJ	 -	 0/1
DP = 363	 AD = 186,177	 AB = 0.487

</span></div>	</td>
	<td style="background-color:inherit"><div class="tooltip">0/0</div>	</td>
	<td style="background-color:inherit"><div class="tooltip">0/1</div>	</td>
	<td><div class="tooltip">chr1</div>	</td>
	<td><div class="tooltip">28206250</div>	</td>
	<td><div class="tooltip">rs41284294</div>	</td>
	<td><div class="tooltip">T</div>	</td>
	<td><div class="tooltip">C</div>	</td>
	<td><div class="tooltip">PASS</div>	</td>
</tr>
<tr class="ALL" >
	<td ><div class="tooltip">8<span class="tooltiptext tooltip-bottom">MPA_ranking	= 7
MPA_final_score	= 4.444444444444445
MPA_impact	= missense_impact
MPA_adjusted	= 4.444444444444445
MPA_available	= 9
MPA_deleterious	= 4

--- SPLICE ---
dbscSNV_ADA_SCORE	= .
dbscSNV_RF_SCORE	= .
spliceai_filtered	= .

--- MISSENSE ---
LRT_pred	= D
SIFT_pred	= D
FATHMM_pred	= .
MetaLR_pred	= T
MetaSVM_pred	= T
PROVEAN_pred	= N
Polyphen2_HDIV_pred	= P
Polyphen2_HVAR_pred	= B
MutationTaster_pred	= D
fathmm-MKL_coding_pred	= D
</span></div>	</td>
	<td><div class="tooltip">4_missense_impact</div>	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom"></span></div> 	</td>
	<td style="background-color:#FFCC00"><div class="tooltip">KIAA2013<span class="tooltiptext tooltip-bottom">LOEUF = 0.755
LOEUF_decile = 4.0

oe_lof = 0.38251
oe_lof_lower = 0.208

Missense_upper = 0.802

</span></div>    	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom">Disease_description.refGene:
 .

</span></div> 	</td>
	<td ><div class="tooltip">0.0016<span class="tooltiptext tooltip-bottom">gnomAD_genome_ALL	= 0.0016
gnomAD_genome_AFR	= 0.0009
gnomAD_genome_AMR	= 0.0048
gnomAD_genome_ASJ	= 0
gnomAD_genome_EAS	= 0
gnomAD_genome_FIN	= 0
gnomAD_genome_NFE	= 0.0025
gnomAD_genome_OTH	= 0
</span></div> 	</td>
	<td ><div class="tooltip">0.0020<span class="tooltiptext tooltip-bottom">gnomAD_exome_ALL	= 0.0020
gnomAD_exome_AFR	= 0.0007
gnomAD_exome_AMR	= 0.0020
gnomAD_exome_ASJ	= 0.0009
gnomAD_exome_EAS	= 0
gnomAD_exome_FIN	= 0.0003
gnomAD_exome_NFE	= 0.0034
gnomAD_exome_OTH	= 0.0029
</span></div> 	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom">CLNREVSTAT:
 .

CLNDN:
 .

CLNALLELEID:
 .

CLNDISDB:
 .

</span></div> 	</td>
	<td ><div class="tooltip">Uncertain_significance<span class="tooltiptext tooltip-bottom"></span></div> 	</td>
	<td><div class="tooltip">.</div>	</td>
	<td ><div class="tooltip">exonic<span class="tooltiptext tooltip-bottom">ExonicFunc.refGene:
 nonsynonymous_SNV

AAChange.refGene:
 KIAA2013:NM_138346:exon2:c.G1228A:p.E410K
 KIAA2013:NM_138346:exon2:c.1228G>A:p.E410K

GeneDetail.refGene:
 .

</span></div> 	</td>
	<td style="background-color:inherit"><div class="tooltip">0/1<span class="tooltiptext tooltip-bottom">CALLER = GATK	 QUALITY = 1966.16

Genotype-B00GMSH	 -	 0/1
DP = 63	 AD = 35,28	 AB = 0.444

Genotype-B00GMSI	 -	 0/0
DP = 79	 AD = 79,0	 AB = 0

Genotype-B00GMSJ	 -	 0/1
DP = 65	 AD = 32,33	 AB = 0.507

</span></div>	</td>
	<td style="background-color:inherit"><div class="tooltip">0/0</div>	</td>
	<td style="background-color:inherit"><div class="tooltip">0/1</div>	</td>
	<td><div class="tooltip">chr1</div>	</td>
	<td><div class="tooltip">11983352</div>	</td>
	<td><div class="tooltip">rs145773140</div>	</td>
	<td><div class="tooltip">C</div>	</td>
	<td><div class="tooltip">T</div>	</td>
	<td><div class="tooltip">PASS</div>	</td>
</tr>
<tr class="ALL  PATHO1 disease2" >
	<td ><div class="tooltip">9<span class="tooltiptext tooltip-bottom">MPA_ranking	= 7
MPA_final_score	= 3.0
MPA_impact	= missense_impact
MPA_adjusted	= 3.0
MPA_available	= 10
MPA_deleterious	= 3

--- SPLICE ---
dbscSNV_ADA_SCORE	= .
dbscSNV_RF_SCORE	= .
spliceai_filtered	= .

--- MISSENSE ---
LRT_pred	= D
SIFT_pred	= T
FATHMM_pred	= T
MetaLR_pred	= T
MetaSVM_pred	= T
PROVEAN_pred	= N
Polyphen2_HDIV_pred	= B
Polyphen2_HVAR_pred	= B
MutationTaster_pred	= D
fathmm-MKL_coding_pred	= D
</span></div>	</td>
	<td><div class="tooltip">3_missense_impact</div>	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom"></span></div> 	</td>
	<td style="background-color:#FF3300"><div class="tooltip">KAZN<span class="tooltiptext tooltip-bottom">LOEUF = 0.381
LOEUF_decile = 1.0

oe_lof = 0.21123000000000003
oe_lof_lower = 0.123

Missense_upper = 0.773

Function Description:
FUNCTION:_Component_of_the_cornified_envelope_of_keratinocytes._May_be_involved_in_the_interplay_between_adherens_junctions_and_desmosomes._The_function_in_the_nucleus_is_not_known._{ECO:0000269|PubMed:15337775}.

Tissue specificity:
TISSUE_SPECIFICITY:_Isoform_2,_isoform_3_and_isoform_4_are_expressed_in_several_cell_lines_including_keratinocytes_and_bladder_and_epidermoid_carcinoma_(at_protein_level)._Isoform_2,_isoform_3_and_isoform_4_are_expressed_in_hair_follicle_and_interfollicular_epidermis_(at_protein_level)._{ECO:0000269|PubMed:15337775}.
</span></div>    	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom">Disease_description.refGene:
 .

</span></div> 	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom">gnomAD_genome_ALL	= .
gnomAD_genome_AFR	= .
gnomAD_genome_AMR	= .
gnomAD_genome_ASJ	= .
gnomAD_genome_EAS	= .
gnomAD_genome_FIN	= .
gnomAD_genome_NFE	= .
gnomAD_genome_OTH	= .
</span></div> 	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom">gnomAD_exome_ALL	= .
gnomAD_exome_AFR	= .
gnomAD_exome_AMR	= .
gnomAD_exome_ASJ	= .
gnomAD_exome_EAS	= .
gnomAD_exome_FIN	= .
gnomAD_exome_NFE	= .
gnomAD_exome_OTH	= .
</span></div> 	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom">CLNREVSTAT:
 .

CLNDN:
 .

CLNALLELEID:
 .

CLNDISDB:
 .

</span></div> 	</td>
	<td ><div class="tooltip">Uncertain_significance<span class="tooltiptext tooltip-bottom">PM2	= 1
Absent from controls (or at extremely low frequency if recessive) (Table 6) in Exome Sequencing Project, 1000 Genomes Project, or Exome Aggregation Consortium

</span></div> 	</td>
	<td><div class="tooltip">.</div>	</td>
	<td ><div class="tooltip">exonic<span class="tooltiptext tooltip-bottom">ExonicFunc.refGene:
 nonsynonymous_SNV

AAChange.refGene:
 KAZN:NM_001017999:exon6:c.G673C:p.V225L
 KAZN:NM_001018000:exon6:c.G937C:p.V313L
 KAZN:NM_001018001:exon6:c.G673C:p.V225L
 KAZN:NM_015209:exon6:c.G955C:p.V319L
 KAZN:NM_201628:exon6:c.G955C:p.V319L
 KAZN:NM_001017999:exon6:c.673G>C:p.V225L
 KAZN:NM_001018000:exon6:c.937G>C:p.V313L
 KAZN:NM_001018001:exon6:c.673G>C:p.V225L
 KAZN:NM_015209:exon6:c.955G>C:p.V319L
 KAZN:NM_201628:exon6:c.955G>C:p.V319L

GeneDetail.refGene:
 .

</span></div> 	</td>
	<td style="background-color:inherit"><div class="tooltip">0/1<span class="tooltiptext tooltip-bottom">CALLER = GATK	 QUALITY = 4168.16

Genotype-B00GMSH	 -	 0/1
DP = 121	 AD = 61,60	 AB = 0.495

Genotype-B00GMSI	 -	 0/1
DP = 126	 AD = 53,73	 AB = 0.579

Genotype-B00GMSJ	 -	 0/0
DP = 131	 AD = 131,0	 AB = 0

</span></div>	</td>
	<td style="background-color:inherit"><div class="tooltip">0/1</div>	</td>
	<td style="background-color:inherit"><div class="tooltip">0/0</div>	</td>
	<td><div class="tooltip">chr1</div>	</td>
	<td><div class="tooltip">15386706</div>	</td>
	<td><div class="tooltip">.</div>	</td>
	<td><div class="tooltip">G</div>	</td>
	<td><div class="tooltip">C</div>	</td>
	<td><div class="tooltip">PASS</div>	</td>
</tr>
<tr class="ALL" >
	<td ><div class="tooltip">10<span class="tooltiptext tooltip-bottom">MPA_ranking	= 7
MPA_final_score	= 3.0
MPA_impact	= missense_impact
MPA_adjusted	= 3.0
MPA_available	= 10
MPA_deleterious	= 3

--- SPLICE ---
dbscSNV_ADA_SCORE	= .
dbscSNV_RF_SCORE	= .
spliceai_filtered	= .

--- MISSENSE ---
LRT_pred	= D
SIFT_pred	= T
FATHMM_pred	= T
MetaLR_pred	= T
MetaSVM_pred	= T
PROVEAN_pred	= N
Polyphen2_HDIV_pred	= B
Polyphen2_HVAR_pred	= B
MutationTaster_pred	= D
fathmm-MKL_coding_pred	= D
</span></div>	</td>
	<td><div class="tooltip">3_missense_impact</div>	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom"></span></div> 	</td>
	<td style="background-color:#FFFFFF"><div class="tooltip">CRYBG2<span class="tooltiptext tooltip-bottom">.</span></div>    	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom">Disease_description.refGene:
 .

</span></div> 	</td>
	<td ><div class="tooltip">0.0016<span class="tooltiptext tooltip-bottom">gnomAD_genome_ALL	= 0.0016
gnomAD_genome_AFR	= 0.0001
gnomAD_genome_AMR	= 0.0012
gnomAD_genome_ASJ	= 0
gnomAD_genome_EAS	= 0
gnomAD_genome_FIN	= 0.0011
gnomAD_genome_NFE	= 0.0027
gnomAD_genome_OTH	= 0.0031
</span></div> 	</td>
	<td ><div class="tooltip">0.0012<span class="tooltiptext tooltip-bottom">gnomAD_exome_ALL	= 0.0012
gnomAD_exome_AFR	= 0.0002
gnomAD_exome_AMR	= 0.0007
gnomAD_exome_ASJ	= 0.0001
gnomAD_exome_EAS	= 0
gnomAD_exome_FIN	= 0.0014
gnomAD_exome_NFE	= 0.0017
gnomAD_exome_OTH	= 0.0004
</span></div> 	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom">CLNREVSTAT:
 .

CLNDN:
 .

CLNALLELEID:
 .

CLNDISDB:
 .

</span></div> 	</td>
	<td ><div class="tooltip">Uncertain_significance<span class="tooltiptext tooltip-bottom">PM1	= 1
Located in a mutational hot spot and/or critical and well-established functional domain (e.g., active site of an enzyme) without benign variation

</span></div> 	</td>
	<td><div class="tooltip">.</div>	</td>
	<td ><div class="tooltip">exonic<span class="tooltiptext tooltip-bottom">ExonicFunc.refGene:
 nonsynonymous_SNV

AAChange.refGene:
 CRYBG2:NM_001039775:exon18:c.G4669C:p.V1557L
 CRYBG2:NM_001039775:exon18:c.4669G>C:p.V1557L

GeneDetail.refGene:
 .

</span></div> 	</td>
	<td style="background-color:inherit"><div class="tooltip">0/1<span class="tooltiptext tooltip-bottom">CALLER = GATK	 QUALITY = 7346.16

Genotype-B00GMSH	 -	 0/1
DP = 217	 AD = 96,121	 AB = 0.557

Genotype-B00GMSI	 -	 0/0
DP = 237	 AD = 237,0	 AB = 0

Genotype-B00GMSJ	 -	 0/1
DP = 231	 AD = 111,120	 AB = 0.519

</span></div>	</td>
	<td style="background-color:inherit"><div class="tooltip">0/0</div>	</td>
	<td style="background-color:inherit"><div class="tooltip">0/1</div>	</td>
	<td><div class="tooltip">chr1</div>	</td>
	<td><div class="tooltip">26650711</div>	</td>
	<td><div class="tooltip">rs144282899</div>	</td>
	<td><div class="tooltip">C</div>	</td>
	<td><div class="tooltip">G</div>	</td>
	<td><div class="tooltip">PASS</div>	</td>
</tr>
<tr class="ALL  disease2" >
	<td ><div class="tooltip">11<span class="tooltiptext tooltip-bottom">MPA_ranking	= 7
MPA_final_score	= 2.2222222222222223
MPA_impact	= missense_impact
MPA_adjusted	= 2.2222222222222223
MPA_available	= 9
MPA_deleterious	= 2

--- SPLICE ---
dbscSNV_ADA_SCORE	= .
dbscSNV_RF_SCORE	= .
spliceai_filtered	= .

--- MISSENSE ---
LRT_pred	= .
SIFT_pred	= T
FATHMM_pred	= T
MetaLR_pred	= T
MetaSVM_pred	= T
PROVEAN_pred	= N
Polyphen2_HDIV_pred	= P
Polyphen2_HVAR_pred	= B
MutationTaster_pred	= D
fathmm-MKL_coding_pred	= D
</span></div>	</td>
	<td><div class="tooltip">2_missense_impact</div>	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom"></span></div> 	</td>
	<td style="background-color:#FF0000"><div class="tooltip">SPEN<span class="tooltiptext tooltip-bottom">LOEUF = 0.071
LOEUF_decile = 0.0

oe_lof = 0.031010000000000003
oe_lof_lower = 0.014

Missense_upper = 0.854

Function Description:
FUNCTION:_May_serve_as_a_nuclear_matrix_platform_that_organizes_and_integrates_transcriptional_responses._In_osteoblasts,_supports_transcription_activation:_synergizes_with_RUNX2_to_enhance_FGFR2_mediated_activation_of_the_osteocalcin_FGF_responsive_element_(OCFRE)_(By_similarity)._Has_also_been_shown_to_be_an_essential_corepressor_protein,_which_probably_regulates_different_key_pathways_such_as_the_Notch_pathway._Negative_regulator_of_the_Notch_pathway_via_its_interaction_with_RBPSUH,_which_prevents_the_association_between_NOTCH1_and_RBPSUH,_and_therefore_suppresses_the_transactivation_activity_of_Notch_signaling._Blocks_the_differentiation_of_precursor_B_cells_into_marginal_zone_B_cells._Probably_represses_transcription_via_the_recruitment_of_large_complexes_containing_histone_deacetylase_proteins._May_bind_both_to_DNA_and_RNA._{ECO:0000250,_ECO:0000269|PubMed:11331609,_ECO:0000269|PubMed:12374742}.

Tissue specificity:
TISSUE_SPECIFICITY:_Expressed_at_high_level_in_brain,_testis,_spleen_and_thymus._Expressed_at_intermediate_level_in_kidney,_liver,_mammary_gland_and_skin.
</span></div>    	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom">Disease_description.refGene:
 .

</span></div> 	</td>
	<td ><div class="tooltip">6.46e-05<span class="tooltiptext tooltip-bottom">gnomAD_genome_ALL	= 6.46e-05
gnomAD_genome_AFR	= 0
gnomAD_genome_AMR	= 0
gnomAD_genome_ASJ	= 0
gnomAD_genome_EAS	= 0
gnomAD_genome_FIN	= 0
gnomAD_genome_NFE	= 0.0001
gnomAD_genome_OTH	= 0
</span></div> 	</td>
	<td ><div class="tooltip">0.0002<span class="tooltiptext tooltip-bottom">gnomAD_exome_ALL	= 0.0002
gnomAD_exome_AFR	= 0.0002
gnomAD_exome_AMR	= 0.0004
gnomAD_exome_ASJ	= 0
gnomAD_exome_EAS	= 0
gnomAD_exome_FIN	= 0
gnomAD_exome_NFE	= 0.0003
gnomAD_exome_OTH	= 0.0005
</span></div> 	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom">CLNREVSTAT:
 .

CLNDN:
 .

CLNALLELEID:
 .

CLNDISDB:
 .

</span></div> 	</td>
	<td ><div class="tooltip">Uncertain_significance<span class="tooltiptext tooltip-bottom">BS2	= 1
Observed in a healthy adult individual for a recessive (homozygous), dominant (heterozygous), or X-linked (hemizygous) disorder, with full penetrance expected at an early age, check ExAC_ALL

</span></div> 	</td>
	<td><div class="tooltip">.</div>	</td>
	<td ><div class="tooltip">exonic<span class="tooltiptext tooltip-bottom">ExonicFunc.refGene:
 nonsynonymous_SNV

AAChange.refGene:
 SPEN:NM_015001:exon11:c.G2351A:p.R784H
 SPEN:NM_015001:exon11:c.2351G>A:p.R784H

GeneDetail.refGene:
 .

</span></div> 	</td>
	<td style="background-color:inherit"><div class="tooltip">0/1<span class="tooltiptext tooltip-bottom">CALLER = GATK	 QUALITY = 4369.16

Genotype-B00GMSH	 -	 0/1
DP = 126	 AD = 54,72	 AB = 0.571

Genotype-B00GMSI	 -	 0/0
DP = 152	 AD = 152,0	 AB = 0

Genotype-B00GMSJ	 -	 0/1
DP = 144	 AD = 75,69	 AB = 0.479

</span></div>	</td>
	<td style="background-color:inherit"><div class="tooltip">0/0</div>	</td>
	<td style="background-color:inherit"><div class="tooltip">0/1</div>	</td>
	<td><div class="tooltip">chr1</div>	</td>
	<td><div class="tooltip">16255086</div>	</td>
	<td><div class="tooltip">rs181758065</div>	</td>
	<td><div class="tooltip">G</div>	</td>
	<td><div class="tooltip">A</div>	</td>
	<td><div class="tooltip">PASS</div>	</td>
</tr>
<tr class="ALL" >
	<td ><div class="tooltip">12<span class="tooltiptext tooltip-bottom">MPA_ranking	= 8
MPA_final_score	= 2
MPA_impact	= splice_impact
MPA_adjusted	= 0
MPA_available	= 0
MPA_deleterious	= 0

--- SPLICE ---
dbscSNV_ADA_SCORE	= .
dbscSNV_RF_SCORE	= .
spliceai_filtered	= .

--- MISSENSE ---
LRT_pred	= .
SIFT_pred	= .
FATHMM_pred	= .
MetaLR_pred	= .
MetaSVM_pred	= .
PROVEAN_pred	= .
Polyphen2_HDIV_pred	= .
Polyphen2_HVAR_pred	= .
MutationTaster_pred	= .
fathmm-MKL_coding_pred	= .
</span></div>	</td>
	<td><div class="tooltip">splice_impact</div>	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom"></span></div> 	</td>
	<td style="background-color:#FFCC00"><div class="tooltip">FAM131C<span class="tooltiptext tooltip-bottom">LOEUF = 0.892
LOEUF_decile = 4.0

oe_lof = 0.42444
oe_lof_lower = 0.221

Missense_upper = 1.194

</span></div>    	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom">Disease_description.refGene:
 .

</span></div> 	</td>
	<td ><div class="tooltip">0.0010<span class="tooltiptext tooltip-bottom">gnomAD_genome_ALL	= 0.0010
gnomAD_genome_AFR	= 0.0016
gnomAD_genome_AMR	= 0
gnomAD_genome_ASJ	= 0.0036
gnomAD_genome_EAS	= 0
gnomAD_genome_FIN	= 0.0003
gnomAD_genome_NFE	= 0.0009
gnomAD_genome_OTH	= 0.0010
</span></div> 	</td>
	<td ><div class="tooltip">1.287e-05<span class="tooltiptext tooltip-bottom">gnomAD_exome_ALL	= 1.287e-05
gnomAD_exome_AFR	= 6.973e-05
gnomAD_exome_AMR	= 0
gnomAD_exome_ASJ	= 0.0001
gnomAD_exome_EAS	= 0
gnomAD_exome_FIN	= 0
gnomAD_exome_NFE	= 0
gnomAD_exome_OTH	= 0.0002
</span></div> 	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom">CLNREVSTAT:
 .

CLNDN:
 .

CLNALLELEID:
 .

CLNDISDB:
 .

</span></div> 	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom"></span></div> 	</td>
	<td><div class="tooltip">.</div>	</td>
	<td ><div class="tooltip">splicing;intronic<span class="tooltiptext tooltip-bottom">ExonicFunc.refGene:
 .

AAChange.refGene:
 .

GeneDetail.refGene:
 NM_182623:exon6:r.spl

</span></div> 	</td>
	<td style="background-color:inherit"><div class="tooltip">0/1<span class="tooltiptext tooltip-bottom">CALLER = GATK	 QUALITY = 2664.12

Genotype-B00GMSH	 -	 0/1
DP = 194	 AD = 147,47	 AB = 0.242

Genotype-B00GMSI	 -	 0/0
DP = 182	 AD = 182,0	 AB = 0

Genotype-B00GMSJ	 -	 0/1
DP = 229	 AD = 184,45	 AB = 0.196

</span></div>	</td>
	<td style="background-color:inherit"><div class="tooltip">0/0</div>	</td>
	<td style="background-color:pink"><div class="tooltip">0/1</div>	</td>
	<td><div class="tooltip">chr1</div>	</td>
	<td><div class="tooltip">16385954</div>	</td>
	<td><div class="tooltip">.</div>	</td>
	<td><div class="tooltip">GCCCA</div>	</td>
	<td><div class="tooltip">G</div>	</td>
	<td><div class="tooltip">PASS</div>	</td>
</tr>
<tr class="ALL" >
	<td ><div class="tooltip">13<span class="tooltiptext tooltip-bottom">MPA_ranking	= 8
MPA_final_score	= 2
MPA_impact	= splice_impact
MPA_adjusted	= 0
MPA_available	= 0
MPA_deleterious	= 0

--- SPLICE ---
dbscSNV_ADA_SCORE	= .
dbscSNV_RF_SCORE	= .
spliceai_filtered	= .

--- MISSENSE ---
LRT_pred	= .
SIFT_pred	= .
FATHMM_pred	= .
MetaLR_pred	= .
MetaSVM_pred	= .
PROVEAN_pred	= .
Polyphen2_HDIV_pred	= .
Polyphen2_HVAR_pred	= .
MutationTaster_pred	= .
fathmm-MKL_coding_pred	= .
</span></div>	</td>
	<td><div class="tooltip">splice_impact</div>	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom"></span></div> 	</td>
	<td style="background-color:#FFFFFF"><div class="tooltip">MST1L<span class="tooltiptext tooltip-bottom">.</span></div>    	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom">Disease_description.refGene:
 .

</span></div> 	</td>
	<td ><div class="tooltip">0.0078<span class="tooltiptext tooltip-bottom">gnomAD_genome_ALL	= 0.0078
gnomAD_genome_AFR	= 0.0140
gnomAD_genome_AMR	= 0.0062
gnomAD_genome_ASJ	= 0.0095
gnomAD_genome_EAS	= 0.0031
gnomAD_genome_FIN	= 0.0055
gnomAD_genome_NFE	= 0.0052
gnomAD_genome_OTH	= 0.0104
</span></div> 	</td>
	<td ><div class="tooltip">0.2009<span class="tooltiptext tooltip-bottom">gnomAD_exome_ALL	= 0.2009
gnomAD_exome_AFR	= 0.1304
gnomAD_exome_AMR	= 0.2328
gnomAD_exome_ASJ	= 0.3365
gnomAD_exome_EAS	= 0.1425
gnomAD_exome_FIN	= 0.2581
gnomAD_exome_NFE	= 0.1984
gnomAD_exome_OTH	= 0.2234
</span></div> 	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom">CLNREVSTAT:
 .

CLNDN:
 .

CLNALLELEID:
 .

CLNDISDB:
 .

</span></div> 	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom"></span></div> 	</td>
	<td><div class="tooltip">.</div>	</td>
	<td ><div class="tooltip">splicing;intronic<span class="tooltiptext tooltip-bottom">ExonicFunc.refGene:
 .

AAChange.refGene:
 .

GeneDetail.refGene:
 NM_001271733:exon1:c.67+7->A

</span></div> 	</td>
	<td style="background-color:inherit"><div class="tooltip">0/1<span class="tooltiptext tooltip-bottom">CALLER = GATK	 QUALITY = 1299.26

Genotype-B00GMSH	 -	 0/1
DP = 98	 AD = 76,22	 AB = 0.224

Genotype-B00GMSI	 -	 0/1
DP = 178	 AD = 139,39	 AB = 0.219

Genotype-B00GMSJ	 -	 0/0
DP = 115	 AD = 106,9	 AB = 0.078

</span></div>	</td>
	<td style="background-color:inherit"><div class="tooltip">0/1</div>	</td>
	<td style="background-color:inherit"><div class="tooltip">0/0</div>	</td>
	<td><div class="tooltip">chr1</div>	</td>
	<td><div class="tooltip">17090902</div>	</td>
	<td><div class="tooltip">.</div>	</td>
	<td><div class="tooltip">A</div>	</td>
	<td><div class="tooltip">AT</div>	</td>
	<td><div class="tooltip">PASS</div>	</td>
</tr>
<tr class="ALL" >
	<td ><div class="tooltip">14<span class="tooltiptext tooltip-bottom">MPA_ranking	= 9
MPA_final_score	= 1.0
MPA_impact	= missense_impact
MPA_adjusted	= 1.0
MPA_available	= 10
MPA_deleterious	= 1

--- SPLICE ---
dbscSNV_ADA_SCORE	= .
dbscSNV_RF_SCORE	= .
spliceai_filtered	= .

--- MISSENSE ---
LRT_pred	= N
SIFT_pred	= T
FATHMM_pred	= T
MetaLR_pred	= T
MetaSVM_pred	= T
PROVEAN_pred	= N
Polyphen2_HDIV_pred	= B
Polyphen2_HVAR_pred	= B
MutationTaster_pred	= N
fathmm-MKL_coding_pred	= D
</span></div>	</td>
	<td><div class="tooltip">1_missense_impact</div>	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom"></span></div> 	</td>
	<td style="background-color:#7FFF00"><div class="tooltip">GPR153<span class="tooltiptext tooltip-bottom">LOEUF = 1.292
LOEUF_decile = 7.0

oe_lof = 0.7811600000000001
oe_lof_lower = 0.491

Missense_upper = 0.986

Function Description:
FUNCTION:_Orphan_receptor.

</span></div>    	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom">Disease_description.refGene:
 .

</span></div> 	</td>
	<td ><div class="tooltip">0.0088<span class="tooltiptext tooltip-bottom">gnomAD_genome_ALL	= 0.0088
gnomAD_genome_AFR	= 0.0014
gnomAD_genome_AMR	= 0.0012
gnomAD_genome_ASJ	= 0.0033
gnomAD_genome_EAS	= 0
gnomAD_genome_FIN	= 0.0120
gnomAD_genome_NFE	= 0.0138
gnomAD_genome_OTH	= 0.0092
</span></div> 	</td>
	<td ><div class="tooltip">0.0065<span class="tooltiptext tooltip-bottom">gnomAD_exome_ALL	= 0.0065
gnomAD_exome_AFR	= 0.0011
gnomAD_exome_AMR	= 0.0040
gnomAD_exome_ASJ	= 0.0112
gnomAD_exome_EAS	= 5.872e-05
gnomAD_exome_FIN	= 0.0110
gnomAD_exome_NFE	= 0.0094
gnomAD_exome_OTH	= 0.0081
</span></div> 	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom">CLNREVSTAT:
 .

CLNDN:
 .

CLNALLELEID:
 .

CLNDISDB:
 .

</span></div> 	</td>
	<td ><div class="tooltip">Uncertain_significance<span class="tooltiptext tooltip-bottom">PP2	= 1
Missense variant in a gene that has a low rate of benign missense variation and in which missense variants are a common mechanism of disease

BS1	= 1
Allele frequency is greater than expected for disorder (see Table 6) > 1% in ESP6500all ExAc? need to check more

</span></div> 	</td>
	<td><div class="tooltip">.</div>	</td>
	<td ><div class="tooltip">exonic<span class="tooltiptext tooltip-bottom">ExonicFunc.refGene:
 nonsynonymous_SNV

AAChange.refGene:
 GPR153:NM_207370:exon5:c.G1102C:p.G368R
 GPR153:NM_207370:exon5:c.1102G>C:p.G368R

GeneDetail.refGene:
 .

</span></div> 	</td>
	<td style="background-color:inherit"><div class="tooltip">0/1<span class="tooltiptext tooltip-bottom">CALLER = GATK	 QUALITY = 2304.16

Genotype-B00GMSH	 -	 0/1
DP = 88	 AD = 48,40	 AB = 0.454

Genotype-B00GMSI	 -	 0/1
DP = 77	 AD = 29,48	 AB = 0.623

Genotype-B00GMSJ	 -	 0/0
DP = 67	 AD = 67,0	 AB = 0

</span></div>	</td>
	<td style="background-color:inherit"><div class="tooltip">0/1</div>	</td>
	<td style="background-color:inherit"><div class="tooltip">0/0</div>	</td>
	<td><div class="tooltip">chr1</div>	</td>
	<td><div class="tooltip">6310562</div>	</td>
	<td><div class="tooltip">rs142015102</div>	</td>
	<td><div class="tooltip">C</div>	</td>
	<td><div class="tooltip">G</div>	</td>
	<td><div class="tooltip">PASS</div>	</td>
</tr>
<tr class="ALL  disease2" >
	<td ><div class="tooltip">15<span class="tooltiptext tooltip-bottom">MPA_ranking	= 9
MPA_final_score	= 0.0
MPA_impact	= missense_impact
MPA_adjusted	= 0.0
MPA_available	= 9
MPA_deleterious	= 0

--- SPLICE ---
dbscSNV_ADA_SCORE	= .
dbscSNV_RF_SCORE	= .
spliceai_filtered	= .

--- MISSENSE ---
LRT_pred	= .
SIFT_pred	= T
FATHMM_pred	= T
MetaLR_pred	= T
MetaSVM_pred	= T
PROVEAN_pred	= N
Polyphen2_HDIV_pred	= B
Polyphen2_HVAR_pred	= B
MutationTaster_pred	= N
fathmm-MKL_coding_pred	= N
</span></div>	</td>
	<td><div class="tooltip">0_missense_impact</div>	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom"></span></div> 	</td>
	<td style="background-color:#FF0000"><div class="tooltip">SPEN<span class="tooltiptext tooltip-bottom">LOEUF = 0.071
LOEUF_decile = 0.0

oe_lof = 0.031010000000000003
oe_lof_lower = 0.014

Missense_upper = 0.854

Function Description:
FUNCTION:_May_serve_as_a_nuclear_matrix_platform_that_organizes_and_integrates_transcriptional_responses._In_osteoblasts,_supports_transcription_activation:_synergizes_with_RUNX2_to_enhance_FGFR2_mediated_activation_of_the_osteocalcin_FGF_responsive_element_(OCFRE)_(By_similarity)._Has_also_been_shown_to_be_an_essential_corepressor_protein,_which_probably_regulates_different_key_pathways_such_as_the_Notch_pathway._Negative_regulator_of_the_Notch_pathway_via_its_interaction_with_RBPSUH,_which_prevents_the_association_between_NOTCH1_and_RBPSUH,_and_therefore_suppresses_the_transactivation_activity_of_Notch_signaling._Blocks_the_differentiation_of_precursor_B_cells_into_marginal_zone_B_cells._Probably_represses_transcription_via_the_recruitment_of_large_complexes_containing_histone_deacetylase_proteins._May_bind_both_to_DNA_and_RNA._{ECO:0000250,_ECO:0000269|PubMed:11331609,_ECO:0000269|PubMed:12374742}.

Tissue specificity:
TISSUE_SPECIFICITY:_Expressed_at_high_level_in_brain,_testis,_spleen_and_thymus._Expressed_at_intermediate_level_in_kidney,_liver,_mammary_gland_and_skin.
</span></div>    	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom">Disease_description.refGene:
 .

</span></div> 	</td>
	<td ><div class="tooltip">0.0014<span class="tooltiptext tooltip-bottom">gnomAD_genome_ALL	= 0.0014
gnomAD_genome_AFR	= 0.0008
gnomAD_genome_AMR	= 0.0036
gnomAD_genome_ASJ	= 0
gnomAD_genome_EAS	= 0
gnomAD_genome_FIN	= 0.0006
gnomAD_genome_NFE	= 0.0020
gnomAD_genome_OTH	= 0.0020
</span></div> 	</td>
	<td ><div class="tooltip">0.0018<span class="tooltiptext tooltip-bottom">gnomAD_exome_ALL	= 0.0018
gnomAD_exome_AFR	= 0.0006
gnomAD_exome_AMR	= 0.0021
gnomAD_exome_ASJ	= 0.0002
gnomAD_exome_EAS	= 0
gnomAD_exome_FIN	= 8.969e-05
gnomAD_exome_NFE	= 0.0030
gnomAD_exome_OTH	= 0.0042
</span></div> 	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom">CLNREVSTAT:
 .

CLNDN:
 .

CLNALLELEID:
 .

CLNDISDB:
 .

</span></div> 	</td>
	<td ><div class="tooltip">Likely_benign<span class="tooltiptext tooltip-bottom">BP4	= 1
Multiple lines of computational evidence suggest no impact on gene or gene product (conservation, evolutionary,splicing impact, etc.)

BS2	= 1
Observed in a healthy adult individual for a recessive (homozygous), dominant (heterozygous), or X-linked (hemizygous) disorder, with full penetrance expected at an early age, check ExAC_ALL

</span></div> 	</td>
	<td><div class="tooltip">.</div>	</td>
	<td ><div class="tooltip">exonic<span class="tooltiptext tooltip-bottom">ExonicFunc.refGene:
 nonsynonymous_SNV

AAChange.refGene:
 SPEN:NM_015001:exon11:c.G3064A:p.V1022M
 SPEN:NM_015001:exon11:c.3064G>A:p.V1022M

GeneDetail.refGene:
 .

</span></div> 	</td>
	<td style="background-color:inherit"><div class="tooltip">0/1<span class="tooltiptext tooltip-bottom">CALLER = GATK	 QUALITY = 2747.16

Genotype-B00GMSH	 -	 0/1
DP = 73	 AD = 37,36	 AB = 0.493

Genotype-B00GMSI	 -	 0/0
DP = 99	 AD = 99,0	 AB = 0

Genotype-B00GMSJ	 -	 0/1
DP = 109	 AD = 52,57	 AB = 0.522

</span></div>	</td>
	<td style="background-color:inherit"><div class="tooltip">0/0</div>	</td>
	<td style="background-color:inherit"><div class="tooltip">0/1</div>	</td>
	<td><div class="tooltip">chr1</div>	</td>
	<td><div class="tooltip">16255799</div>	</td>
	<td><div class="tooltip">rs115566585</div>	</td>
	<td><div class="tooltip">G</div>	</td>
	<td><div class="tooltip">A</div>	</td>
	<td><div class="tooltip">PASS</div>	</td>
</tr>
<tr class="ALL" >
	<td ><div class="tooltip">16<span class="tooltiptext tooltip-bottom">MPA_ranking	= 9
MPA_final_score	= 0.0
MPA_impact	= missense_impact
MPA_adjusted	= 0.0
MPA_available	= 10
MPA_deleterious	= 0

--- SPLICE ---
dbscSNV_ADA_SCORE	= .
dbscSNV_RF_SCORE	= .
spliceai_filtered	= .

--- MISSENSE ---
LRT_pred	= N
SIFT_pred	= T
FATHMM_pred	= T
MetaLR_pred	= T
MetaSVM_pred	= T
PROVEAN_pred	= N
Polyphen2_HDIV_pred	= P
Polyphen2_HVAR_pred	= P
MutationTaster_pred	= N
fathmm-MKL_coding_pred	= N
</span></div>	</td>
	<td><div class="tooltip">0_missense_impact</div>	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom"></span></div> 	</td>
	<td style="background-color:#7FFF00"><div class="tooltip">CCDC27<span class="tooltiptext tooltip-bottom">LOEUF = 1.433
LOEUF_decile = 7.0

oe_lof = 1.1008
oe_lof_lower = 0.854

Missense_upper = 1.1

</span></div>    	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom">Disease_description.refGene:
 .

</span></div> 	</td>
	<td ><div class="tooltip">6.467e-05<span class="tooltiptext tooltip-bottom">gnomAD_genome_ALL	= 6.467e-05
gnomAD_genome_AFR	= 0
gnomAD_genome_AMR	= 0
gnomAD_genome_ASJ	= 0
gnomAD_genome_EAS	= 0
gnomAD_genome_FIN	= 0
gnomAD_genome_NFE	= 0.0001
gnomAD_genome_OTH	= 0
</span></div> 	</td>
	<td ><div class="tooltip">8.934e-05<span class="tooltiptext tooltip-bottom">gnomAD_exome_ALL	= 8.934e-05
gnomAD_exome_AFR	= 0
gnomAD_exome_AMR	= 5.956e-05
gnomAD_exome_ASJ	= 0
gnomAD_exome_EAS	= 0
gnomAD_exome_FIN	= 0
gnomAD_exome_NFE	= 1.791e-05
gnomAD_exome_OTH	= 0.0002
</span></div> 	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom">CLNREVSTAT:
 .

CLNDN:
 .

CLNALLELEID:
 .

CLNDISDB:
 .

</span></div> 	</td>
	<td ><div class="tooltip">Uncertain_significance<span class="tooltiptext tooltip-bottom"></span></div> 	</td>
	<td><div class="tooltip">.</div>	</td>
	<td ><div class="tooltip">exonic<span class="tooltiptext tooltip-bottom">ExonicFunc.refGene:
 nonsynonymous_SNV

AAChange.refGene:
 CCDC27:NM_152492:exon8:c.C1345G:p.Q449E
 CCDC27:NM_152492:exon8:c.1345C>G:p.Q449E

GeneDetail.refGene:
 .

</span></div> 	</td>
	<td style="background-color:inherit"><div class="tooltip">0/1<span class="tooltiptext tooltip-bottom">CALLER = GATK	 QUALITY = 1726.16

Genotype-B00GMSH	 -	 0/1
DP = 58	 AD = 36,22	 AB = 0.379

Genotype-B00GMSI	 -	 0/0
DP = 84	 AD = 84,0	 AB = 0

Genotype-B00GMSJ	 -	 0/1
DP = 77	 AD = 40,37	 AB = 0.480

</span></div>	</td>
	<td style="background-color:inherit"><div class="tooltip">0/0</div>	</td>
	<td style="background-color:inherit"><div class="tooltip">0/1</div>	</td>
	<td><div class="tooltip">chr1</div>	</td>
	<td><div class="tooltip">3680293</div>	</td>
	<td><div class="tooltip">rs145188556</div>	</td>
	<td><div class="tooltip">C</div>	</td>
	<td><div class="tooltip">G</div>	</td>
	<td><div class="tooltip">PASS</div>	</td>
</tr>
<tr class="ALL" >
	<td ><div class="tooltip">17<span class="tooltiptext tooltip-bottom">MPA_ranking	= 3
MPA_final_score	= 10
MPA_impact	= missense_impact,splice_impact
MPA_adjusted	= 8.0
MPA_available	= 10
MPA_deleterious	= 8

--- SPLICE ---
dbscSNV_ADA_SCORE	= 0.9992
dbscSNV_RF_SCORE	= 0.956
spliceai_filtered	 DS_AG=0.0027
spliceai_filtered	 DS_AL=0.0000
spliceai_filtered	 DS_DG=0.0333
spliceai_filtered	 DS_DL=0.2846

--- MISSENSE ---
LRT_pred	= D
SIFT_pred	= T
FATHMM_pred	= D
MetaLR_pred	= D
MetaSVM_pred	= D
PROVEAN_pred	= N
Polyphen2_HDIV_pred	= D
Polyphen2_HVAR_pred	= D
MutationTaster_pred	= D
fathmm-MKL_coding_pred	= D
</span></div>	</td>
	<td><div class="tooltip">missense_impact,splice_impact</div>	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom"></span></div> 	</td>
	<td style="background-color:#FF9900"><div class="tooltip">COL16A1<span class="tooltiptext tooltip-bottom">LOEUF = 0.627
LOEUF_decile = 3.0

oe_lof = 0.50517
oe_lof_lower = 0.409

Missense_upper = 0.952

Function Description:
FUNCTION:_Involved_in_mediating_cell_attachment_and_inducing_integrin_mediated_cellular_reactions,_such_as_cell_spreading_and_alterations_in_cell_morphology._{ECO:0000269|PubMed:16754661}.

Tissue specificity:
TISSUE_SPECIFICITY:_In_papillary_dermis,_is_a_component_of_specialized_fibrillin_1_containing_microfibrils,_whereas_in_territorial_cartilage_matrix,_it_is_localized_to_a_discrete_population_of_thin,_weakly_banded_collagen_fibrils_in_association_with_other_collagens_(at_protein_level)._In_the_placenta,_where_it_is_found_in_the_amnion,_a_membranous_tissue_lining_the_amniotic_cavity._Within_the_amnion,_it_is_found_in_an_acellular,_relatively_dense_layer_of_a_complex_network_of_reticular_fibers._Also_located_to_a_fibroblast_layer_beneath_this_dense_layer._Exists_in_tissues_in_association_with_other_types_of_collagen._{ECO:0000269|PubMed:12782140}.
</span></div>    	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom">Disease_description.refGene:
 .

</span></div> 	</td>
	<td ><div class="tooltip">0.0033<span class="tooltiptext tooltip-bottom">gnomAD_genome_ALL	= 0.0033
gnomAD_genome_AFR	= 0.0010
gnomAD_genome_AMR	= 0.0048
gnomAD_genome_ASJ	= 0.0298
gnomAD_genome_EAS	= 0
gnomAD_genome_FIN	= 0.0006
gnomAD_genome_NFE	= 0.0051
gnomAD_genome_OTH	= 0.0020
</span></div> 	</td>
	<td ><div class="tooltip">0.0053<span class="tooltiptext tooltip-bottom">gnomAD_exome_ALL	= 0.0053
gnomAD_exome_AFR	= 0.0012
gnomAD_exome_AMR	= 0.0038
gnomAD_exome_ASJ	= 0.0287
gnomAD_exome_EAS	= 0
gnomAD_exome_FIN	= 0.0003
gnomAD_exome_NFE	= 0.0073
gnomAD_exome_OTH	= 0.0062
</span></div> 	</td>
	<td ><div class="tooltip">Likely_benign<span class="tooltiptext tooltip-bottom">CLNREVSTAT:
 criteria_provided,_single_submitter

CLNDN:
 not_provided

CLNALLELEID:
 696680

CLNDISDB:
 MedGen:CN517202

</span></div> 	</td>
	<td ><div class="tooltip">Uncertain_significance<span class="tooltiptext tooltip-bottom">PP3	= 1
Multiple lines of computational evidence support a deleterious effect on the gene or gene product (conservation, evolutionary, splicing impact, etc.) sfit for conservation, GERP++_RS for evolutionary, splicing impact from dbNSFP

BS1	= 1
Allele frequency is greater than expected for disorder (see Table 6) > 1% in ESP6500all ExAc? need to check more

</span></div> 	</td>
	<td><div class="tooltip">.</div>	</td>
	<td ><div class="tooltip">exonic<span class="tooltiptext tooltip-bottom">ExonicFunc.refGene:
 nonsynonymous_SNV

AAChange.refGene:
 COL16A1:NM_001856:exon16:c.A1193G:p.K398R
 COL16A1:NM_001856:exon16:c.1193A>G:p.K398R

GeneDetail.refGene:
 .

</span></div> 	</td>
	<td style="background-color:inherit"><div class="tooltip">0/0<span class="tooltiptext tooltip-bottom">CALLER = GATK	 QUALITY = 847.13

Genotype-B00GMSH	 -	 0/0
DP = 47	 AD = 47,0	 AB = 0

Genotype-B00GMSI	 -	 0/1
DP = 56	 AD = 28,28	 AB = 0.5

Genotype-B00GMSJ	 -	 0/0
DP = 58	 AD = 58,0	 AB = 0

</span></div>	</td>
	<td style="background-color:inherit"><div class="tooltip">0/1</div>	</td>
	<td style="background-color:inherit"><div class="tooltip">0/0</div>	</td>
	<td><div class="tooltip">chr1</div>	</td>
	<td><div class="tooltip">32158076</div>	</td>
	<td><div class="tooltip">rs115606583</div>	</td>
	<td><div class="tooltip">T</div>	</td>
	<td><div class="tooltip">C</div>	</td>
	<td><div class="tooltip">PASS</div>	</td>
</tr>
<tr class="ALL  disease2" >
	<td ><div class="tooltip">18<span class="tooltiptext tooltip-bottom">MPA_ranking	= 5
MPA_final_score	= 10.0
MPA_impact	= missense_impact
MPA_adjusted	= 10.0
MPA_available	= 10
MPA_deleterious	= 10

--- SPLICE ---
dbscSNV_ADA_SCORE	= .
dbscSNV_RF_SCORE	= .
spliceai_filtered	 DS_AG=0.1427
spliceai_filtered	 DS_AL=0.0002
spliceai_filtered	 DS_DG=0.0000
spliceai_filtered	 DS_DL=0.0000

--- MISSENSE ---
LRT_pred	= D
SIFT_pred	= D
FATHMM_pred	= D
MetaLR_pred	= D
MetaSVM_pred	= D
PROVEAN_pred	= D
Polyphen2_HDIV_pred	= D
Polyphen2_HVAR_pred	= D
MutationTaster_pred	= D
fathmm-MKL_coding_pred	= D
</span></div>	</td>
	<td><div class="tooltip">10_missense_impact</div>	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom"></span></div> 	</td>
	<td style="background-color:#FFFFFF"><div class="tooltip">DISP3<span class="tooltiptext tooltip-bottom">.Function Description:
FUNCTION:_Plays_a_role_in_neuronal_proliferation_and_differentiation_(PubMed:25281927)._Plays_a_role_in_the_accumulation_of_cellular_cholesterol_(By_similarity)._Involved_in_intracellular_lipid_droplet_formation_(PubMed:25281927)._May_contribute_to_cholesterol_homeostasis_in_neuronal_cells_(By_similarity)._{ECO:0000250|UniProtKB:B9U3F2,_ECO:0000269|PubMed:25281927}.

Tissue specificity:
TISSUE_SPECIFICITY:_Expressed_in_brain_and_testis_(PubMed:15645143)._{ECO:0000269|PubMed:15645143}.
</span></div>    	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom">Disease_description.refGene:
 .

</span></div> 	</td>
	<td ><div class="tooltip">0.0033<span class="tooltiptext tooltip-bottom">gnomAD_genome_ALL	= 0.0033
gnomAD_genome_AFR	= 0.0008
gnomAD_genome_AMR	= 0.0012
gnomAD_genome_ASJ	= 0.0033
gnomAD_genome_EAS	= 0
gnomAD_genome_FIN	= 0.0020
gnomAD_genome_NFE	= 0.0055
gnomAD_genome_OTH	= 0.0031
</span></div> 	</td>
	<td ><div class="tooltip">0.0029<span class="tooltiptext tooltip-bottom">gnomAD_exome_ALL	= 0.0029
gnomAD_exome_AFR	= 0.0007
gnomAD_exome_AMR	= 0.0025
gnomAD_exome_ASJ	= 0.0007
gnomAD_exome_EAS	= 5.798e-05
gnomAD_exome_FIN	= 0.0012
gnomAD_exome_NFE	= 0.0049
gnomAD_exome_OTH	= 0.0035
</span></div> 	</td>
	<td ><div class="tooltip">Likely_benign<span class="tooltiptext tooltip-bottom">CLNREVSTAT:
 criteria_provided,_single_submitter

CLNDN:
 not_provided

CLNALLELEID:
 706601

CLNDISDB:
 MedGen:CN517202

</span></div> 	</td>
	<td ><div class="tooltip">Uncertain_significance<span class="tooltiptext tooltip-bottom">PM1	= 1
Located in a mutational hot spot and/or critical and well-established functional domain (e.g., active site of an enzyme) without benign variation

PP3	= 1
Multiple lines of computational evidence support a deleterious effect on the gene or gene product (conservation, evolutionary, splicing impact, etc.) sfit for conservation, GERP++_RS for evolutionary, splicing impact from dbNSFP

</span></div> 	</td>
	<td><div class="tooltip">.</div>	</td>
	<td ><div class="tooltip">exonic<span class="tooltiptext tooltip-bottom">ExonicFunc.refGene:
 nonsynonymous_SNV

AAChange.refGene:
 DISP3:NM_020780:exon6:c.G1625A:p.R542H
 DISP3:NM_020780:exon6:c.1625G>A:p.R542H

GeneDetail.refGene:
 .

</span></div> 	</td>
	<td style="background-color:inherit"><div class="tooltip">0/0<span class="tooltiptext tooltip-bottom">CALLER = GATK	 QUALITY = 3292.13

Genotype-B00GMSH	 -	 0/0
DP = 146	 AD = 146,0	 AB = 0

Genotype-B00GMSI	 -	 0/0
DP = 163	 AD = 162,1	 AB = 0.006

Genotype-B00GMSJ	 -	 0/1
DP = 199	 AD = 82,117	 AB = 0.587

</span></div>	</td>
	<td style="background-color:inherit"><div class="tooltip">0/0</div>	</td>
	<td style="background-color:inherit"><div class="tooltip">0/1</div>	</td>
	<td><div class="tooltip">chr1</div>	</td>
	<td><div class="tooltip">11576094</div>	</td>
	<td><div class="tooltip">rs41274532</div>	</td>
	<td><div class="tooltip">G</div>	</td>
	<td><div class="tooltip">A</div>	</td>
	<td><div class="tooltip">PASS</div>	</td>
</tr>
<tr class="ALL OMIMREC" >
	<td ><div class="tooltip">19<span class="tooltiptext tooltip-bottom">MPA_ranking	= 5
MPA_final_score	= 7.0
MPA_impact	= missense_impact
MPA_adjusted	= 7.0
MPA_available	= 10
MPA_deleterious	= 7

--- SPLICE ---
dbscSNV_ADA_SCORE	= .
dbscSNV_RF_SCORE	= .
spliceai_filtered	= .

--- MISSENSE ---
LRT_pred	= D
SIFT_pred	= D
FATHMM_pred	= T
MetaLR_pred	= T
MetaSVM_pred	= T
PROVEAN_pred	= D
Polyphen2_HDIV_pred	= D
Polyphen2_HVAR_pred	= D
MutationTaster_pred	= D
fathmm-MKL_coding_pred	= D
</span></div>	</td>
	<td><div class="tooltip">7_missense_impact</div>	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom"></span></div> 	</td>
	<td style="background-color:#FF9900"><div class="tooltip">CEP104<span class="tooltiptext tooltip-bottom">LOEUF = 0.743
LOEUF_decile = 3.0

oe_lof = 0.54175
oe_lof_lower = 0.4

Missense_upper = 1.018

Function Description:
FUNCTION:_Required_for_ciliogenesis_and_for_structural_integrity_at_the_ciliary_tip._{ECO:0000269|PubMed:23970417}.

</span></div>    	</td>
	<td ><div class="tooltip">Joubert_syndrome_25,_616781__3_,_Autosomal_recessive<span class="tooltiptext tooltip-bottom">Disease_description.refGene:
 

DISEASE:_Joubert_syndrome_25_(JBTS25)_[MIM:616781]:_A_form_of_Joubert_syndrome,_a_disorder_presenting_with_cerebellar_ataxia,_oculomotor_apraxia,_hypotonia,_neonatal_breathing_abnormalities_and_psychomotor_delay._Neuroradiologically,_it_is_characterized_by_cerebellar_vermian_hypoplasia/aplasia,_thickened_and_reoriented_superior_cerebellar_peduncles,_and_an_abnormally_large_interpeduncular_fossa,_giving_the_appearance_of_a_molar_tooth_on_transaxial_slices_(molar_tooth_sign)._Additional_variable_features_include_retinal_dystrophy,_renal_disease,_liver_fibrosis,_and_polydactyly._JBTS25_clinical_manifestations_appear_to_be_confined_to_the_neurologic_system._JBTS25_inheritance_is_autosomal_recessive._{ECO:0000269|PubMed:26477546}._Note=The_disease_is_caused_by_mutations_affecting_the_gene_represented_in_this_entry.

</span></div> 	</td>
	<td ><div class="tooltip">0.0020<span class="tooltiptext tooltip-bottom">gnomAD_genome_ALL	= 0.0020
gnomAD_genome_AFR	= 0.0010
gnomAD_genome_AMR	= 0.0012
gnomAD_genome_ASJ	= 0
gnomAD_genome_EAS	= 0
gnomAD_genome_FIN	= 0.0009
gnomAD_genome_NFE	= 0.0033
gnomAD_genome_OTH	= 0
</span></div> 	</td>
	<td ><div class="tooltip">0.0028<span class="tooltiptext tooltip-bottom">gnomAD_exome_ALL	= 0.0028
gnomAD_exome_AFR	= 0.0012
gnomAD_exome_AMR	= 0.0010
gnomAD_exome_ASJ	= 0.0001
gnomAD_exome_EAS	= 5.798e-05
gnomAD_exome_FIN	= 0.0012
gnomAD_exome_NFE	= 0.0049
gnomAD_exome_OTH	= 0.0015
</span></div> 	</td>
	<td ><div class="tooltip">Benign<span class="tooltiptext tooltip-bottom">CLNREVSTAT:
 criteria_provided,_single_submitter

CLNDN:
 Joubert_syndrome_25

CLNALLELEID:
 690595

CLNDISDB:
 MONDO:MONDO:0014770,MedGen:C4084842,OMIM:616781

</span></div> 	</td>
	<td ><div class="tooltip">Uncertain_significance<span class="tooltiptext tooltip-bottom">PM1	= 1
Located in a mutational hot spot and/or critical and well-established functional domain (e.g., active site of an enzyme) without benign variation

BP1	= 1
Missense variant in a gene for which primarily truncating variants are known to cause disease truncating:  stop_gain / frameshift deletion/  nonframshift deletion
	    We defined Protein truncating variants  (4) (table S1) as single-nucleotide variants (SNVs) predicted to introduce a premature stop codon or to disrupt a splice site, small insertions or deletions (indels) predicted to disrupt a transcript reading frame, and larger deletions 

PM2	= 1
Absent from controls (or at extremely low frequency if recessive) (Table 6) in Exome Sequencing Project, 1000 Genomes Project, or Exome Aggregation Consortium

</span></div> 	</td>
	<td><div class="tooltip">.</div>	</td>
	<td ><div class="tooltip">exonic<span class="tooltiptext tooltip-bottom">ExonicFunc.refGene:
 nonsynonymous_SNV

AAChange.refGene:
 CEP104:NM_014704:exon14:c.C1966T:p.R656C
 CEP104:NM_014704:exon14:c.1966C>T:p.R656C

GeneDetail.refGene:
 .

</span></div> 	</td>
	<td style="background-color:inherit"><div class="tooltip">0/0<span class="tooltiptext tooltip-bottom">CALLER = GATK	 QUALITY = 1456.13

Genotype-B00GMSH	 -	 0/0
DP = 103	 AD = 103,0	 AB = 0

Genotype-B00GMSI	 -	 0/1
DP = 102	 AD = 53,49	 AB = 0.480

Genotype-B00GMSJ	 -	 0/0
DP = 98	 AD = 98,0	 AB = 0

</span></div>	</td>
	<td style="background-color:inherit"><div class="tooltip">0/1</div>	</td>
	<td style="background-color:inherit"><div class="tooltip">0/0</div>	</td>
	<td><div class="tooltip">chr1</div>	</td>
	<td><div class="tooltip">3746432</div>	</td>
	<td><div class="tooltip">rs148465057</div>	</td>
	<td><div class="tooltip">G</div>	</td>
	<td><div class="tooltip">A</div>	</td>
	<td><div class="tooltip">PASS</div>	</td>
</tr>
<tr class="ALL" >
	<td ><div class="tooltip">20<span class="tooltiptext tooltip-bottom">MPA_ranking	= 5
MPA_final_score	= 7.0
MPA_impact	= missense_impact
MPA_adjusted	= 7.0
MPA_available	= 10
MPA_deleterious	= 7

--- SPLICE ---
dbscSNV_ADA_SCORE	= .
dbscSNV_RF_SCORE	= .
spliceai_filtered	= .

--- MISSENSE ---
LRT_pred	= N
SIFT_pred	= D
FATHMM_pred	= D
MetaLR_pred	= D
MetaSVM_pred	= D
PROVEAN_pred	= D
Polyphen2_HDIV_pred	= D
Polyphen2_HVAR_pred	= D
MutationTaster_pred	= N
fathmm-MKL_coding_pred	= N
</span></div>	</td>
	<td><div class="tooltip">7_missense_impact</div>	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom"></span></div> 	</td>
	<td style="background-color:#3FFF00"><div class="tooltip">TAS1R3<span class="tooltiptext tooltip-bottom">LOEUF = 1.563
LOEUF_decile = 8.0

oe_lof = 1.1752
oe_lof_lower = 0.893

Missense_upper = 1.361

Function Description:
FUNCTION:_Putative_taste_receptor._TAS1R1/TAS1R3_responds_to_the_umami_taste_stimulus_(the_taste_of_monosodium_glutamate)._TAS1R2/TAS1R3_recognizes_diverse_natural_and_synthetic_sweeteners._TAS1R3_is_essential_for_the_recognition_and_response_to_the_disaccharide_trehalose_(By_similarity)._Sequence_differences_within_and_between_species_can_significantly_influence_the_selectivity_and_specificity_of_taste_responses._{ECO:0000250,_ECO:0000269|PubMed:11917125,_ECO:0000269|PubMed:12892531}.

</span></div>    	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom">Disease_description.refGene:
 .

</span></div> 	</td>
	<td ><div class="tooltip">0.0002<span class="tooltiptext tooltip-bottom">gnomAD_genome_ALL	= 0.0002
gnomAD_genome_AFR	= 0.0007
gnomAD_genome_AMR	= 0
gnomAD_genome_ASJ	= 0
gnomAD_genome_EAS	= 0
gnomAD_genome_FIN	= 0
gnomAD_genome_NFE	= 6.68e-05
gnomAD_genome_OTH	= 0
</span></div> 	</td>
	<td ><div class="tooltip">8.367e-05<span class="tooltiptext tooltip-bottom">gnomAD_exome_ALL	= 8.367e-05
gnomAD_exome_AFR	= 0.0010
gnomAD_exome_AMR	= 3.227e-05
gnomAD_exome_ASJ	= 0
gnomAD_exome_EAS	= 0
gnomAD_exome_FIN	= 0
gnomAD_exome_NFE	= 3.174e-05
gnomAD_exome_OTH	= 0.0002
</span></div> 	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom">CLNREVSTAT:
 .

CLNDN:
 .

CLNALLELEID:
 .

CLNDISDB:
 .

</span></div> 	</td>
	<td ><div class="tooltip">Uncertain_significance<span class="tooltiptext tooltip-bottom">PM1	= 1
Located in a mutational hot spot and/or critical and well-established functional domain (e.g., active site of an enzyme) without benign variation

</span></div> 	</td>
	<td><div class="tooltip">.</div>	</td>
	<td ><div class="tooltip">exonic<span class="tooltiptext tooltip-bottom">ExonicFunc.refGene:
 nonsynonymous_SNV

AAChange.refGene:
 TAS1R3:NM_152228:exon5:c.C1528T:p.R510W
 TAS1R3:NM_152228:exon5:c.1528C>T:p.R510W

GeneDetail.refGene:
 .

</span></div> 	</td>
	<td style="background-color:inherit"><div class="tooltip">0/0<span class="tooltiptext tooltip-bottom">CALLER = GATK	 QUALITY = 1130.13

Genotype-B00GMSH	 -	 0/0
DP = 94	 AD = 94,0	 AB = 0

Genotype-B00GMSI	 -	 0/1
DP = 81	 AD = 38,43	 AB = 0.530

Genotype-B00GMSJ	 -	 0/0
DP = 102	 AD = 102,0	 AB = 0

</span></div>	</td>
	<td style="background-color:inherit"><div class="tooltip">0/1</div>	</td>
	<td style="background-color:inherit"><div class="tooltip">0/0</div>	</td>
	<td><div class="tooltip">chr1</div>	</td>
	<td><div class="tooltip">1268687</div>	</td>
	<td><div class="tooltip">rs149755069</div>	</td>
	<td><div class="tooltip">C</div>	</td>
	<td><div class="tooltip">T</div>	</td>
	<td><div class="tooltip">PASS</div>	</td>
</tr>
<tr class="ALL" >
	<td ><div class="tooltip">21<span class="tooltiptext tooltip-bottom">MPA_ranking	= 7
MPA_final_score	= 5.0
MPA_impact	= missense_impact
MPA_adjusted	= 5.0
MPA_available	= 4
MPA_deleterious	= 2

--- SPLICE ---
dbscSNV_ADA_SCORE	= .
dbscSNV_RF_SCORE	= .
spliceai_filtered	= .

--- MISSENSE ---
LRT_pred	= .
SIFT_pred	= .
FATHMM_pred	= .
MetaLR_pred	= .
MetaSVM_pred	= .
PROVEAN_pred	= .
Polyphen2_HDIV_pred	= B
Polyphen2_HVAR_pred	= B
MutationTaster_pred	= D
fathmm-MKL_coding_pred	= D
</span></div>	</td>
	<td><div class="tooltip">2_missense_impact</div>	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom"></span></div> 	</td>
	<td style="background-color:#FF9900"><div class="tooltip">CDK11B<span class="tooltiptext tooltip-bottom">LOEUF = 0.678
LOEUF_decile = 3.0

oe_lof = 0.41841999999999996
oe_lof_lower = 0.267

Missense_upper = 0.81

Function Description:
FUNCTION:_Plays_multiple_roles_in_cell_cycle_progression,_cytokinesis_and_apoptosis._Involved_in_pre_mRNA_splicing_in_a_kinase_activity_dependent_manner._Isoform_7_may_act_as_a_negative_regulator_of_normal_cell_cycle_progression._{ECO:0000269|PubMed:12501247,_ECO:0000269|PubMed:12624090,_ECO:0000269|PubMed:18216018,_ECO:0000269|PubMed:2217177}.

Tissue specificity:
TISSUE_SPECIFICITY:_Expressed_ubiquitously._Some_evidence_of_isoform_specific_tissue_distribution._{ECO:0000269|PubMed:8195233,_ECO:0000269|PubMed:9750192}.
</span></div>    	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom">Disease_description.refGene:
 .

</span></div> 	</td>
	<td ><div class="tooltip">0.0011<span class="tooltiptext tooltip-bottom">gnomAD_genome_ALL	= 0.0011
gnomAD_genome_AFR	= 0.0034
gnomAD_genome_AMR	= 0
gnomAD_genome_ASJ	= 0
gnomAD_genome_EAS	= 0.0020
gnomAD_genome_FIN	= 0
gnomAD_genome_NFE	= 0.0003
gnomAD_genome_OTH	= 0
</span></div> 	</td>
	<td ><div class="tooltip">0.0007<span class="tooltiptext tooltip-bottom">gnomAD_exome_ALL	= 0.0007
gnomAD_exome_AFR	= 0.0043
gnomAD_exome_AMR	= 0.0003
gnomAD_exome_ASJ	= 0.0001
gnomAD_exome_EAS	= 0.0009
gnomAD_exome_FIN	= 0.0002
gnomAD_exome_NFE	= 0.0002
gnomAD_exome_OTH	= 0
</span></div> 	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom">CLNREVSTAT:
 .

CLNDN:
 .

CLNALLELEID:
 .

CLNDISDB:
 .

</span></div> 	</td>
	<td ><div class="tooltip">Uncertain_significance<span class="tooltiptext tooltip-bottom">PP3	= 1
Multiple lines of computational evidence support a deleterious effect on the gene or gene product (conservation, evolutionary, splicing impact, etc.) sfit for conservation, GERP++_RS for evolutionary, splicing impact from dbNSFP

BS1	= 1
Allele frequency is greater than expected for disorder (see Table 6) > 1% in ESP6500all ExAc? need to check more

</span></div> 	</td>
	<td><div class="tooltip">.</div>	</td>
	<td ><div class="tooltip">exonic<span class="tooltiptext tooltip-bottom">ExonicFunc.refGene:
 nonsynonymous_SNV

AAChange.refGene:
 CDK11B:NM_033487:exon9:c.G406A:p.D136N
 CDK11B:NM_033490:exon10:c.G523A:p.D175N
 CDK11B:NM_001291345:exon12:c.G1138A:p.D380N
 CDK11B:NM_001787:exon12:c.G1207A:p.D403N
 CDK11B:NM_033486:exon12:c.G1168A:p.D390N
 CDK11B:NM_033489:exon13:c.G1066A:p.D356N
 CDK11B:NM_033487:exon9:c.406G>A:p.D136N
 CDK11B:NM_033490:exon10:c.523G>A:p.D175N
 CDK11B:NM_001291345:exon12:c.1138G>A:p.D380N
 CDK11B:NM_001787:exon12:c.1207G>A:p.D403N
 CDK11B:NM_033486:exon12:c.1168G>A:p.D390N
 CDK11B:NM_033489:exon13:c.1066G>A:p.D356N

GeneDetail.refGene:
 .

</span></div> 	</td>
	<td style="background-color:inherit"><div class="tooltip">0/0<span class="tooltiptext tooltip-bottom">CALLER = GATK	 QUALITY = 240.13

Genotype-B00GMSH	 -	 0/0
DP = 133	 AD = 127,6	 AB = 0.045

Genotype-B00GMSI	 -	 0/0
DP = 139	 AD = 127,12	 AB = 0.086

Genotype-B00GMSJ	 -	 0/1
DP = 140	 AD = 118,22	 AB = 0.157

</span></div>	</td>
	<td style="background-color:inherit"><div class="tooltip">0/0</div>	</td>
	<td style="background-color:pink"><div class="tooltip">0/1</div>	</td>
	<td><div class="tooltip">chr1</div>	</td>
	<td><div class="tooltip">1575715</div>	</td>
	<td><div class="tooltip">.</div>	</td>
	<td><div class="tooltip">C</div>	</td>
	<td><div class="tooltip">T</div>	</td>
	<td><div class="tooltip">PASS</div>	</td>
</tr>
<tr class="ALL" >
	<td ><div class="tooltip">22<span class="tooltiptext tooltip-bottom">MPA_ranking	= 7
MPA_final_score	= 5.0
MPA_impact	= missense_impact
MPA_adjusted	= 5.0
MPA_available	= 10
MPA_deleterious	= 5

--- SPLICE ---
dbscSNV_ADA_SCORE	= .
dbscSNV_RF_SCORE	= .
spliceai_filtered	= .

--- MISSENSE ---
LRT_pred	= N
SIFT_pred	= D
FATHMM_pred	= T
MetaLR_pred	= T
MetaSVM_pred	= T
PROVEAN_pred	= D
Polyphen2_HDIV_pred	= D
Polyphen2_HVAR_pred	= P
MutationTaster_pred	= D
fathmm-MKL_coding_pred	= D
</span></div>	</td>
	<td><div class="tooltip">5_missense_impact</div>	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom"></span></div> 	</td>
	<td style="background-color:#BFFF00"><div class="tooltip">PADI4<span class="tooltiptext tooltip-bottom">LOEUF = 1.136
LOEUF_decile = 6.0

oe_lof = 0.8241200000000001
oe_lof_lower = 0.606

Missense_upper = 1.099

Function Description:
FUNCTION:_Catalyzes_the_citrullination/deimination_of_arginine_residues_of_proteins_such_as_histones,_thereby_playing_a_key_role_in_histone_code_and_regulation_of_stem_cell_maintenance._Citrullinates_histone_H1_at_'Arg_54'_(to_form_H1R54ci),_histone_H3_at_'Arg_2',_'Arg_8',_'Arg_17'_and/or_'Arg_26'_(to_form_H3R2ci,_H3R8ci,_H3R17ci,_H3R26ci,_respectively)_and_histone_H4_at_'Arg_3'_(to_form_H4R3ci)._Acts_as_a_key_regulator_of_stem_cell_maintenance_by_mediating_citrullination_of_histone_H1:_citrullination_of_'Arg_54'_of_histone_H1_(H1R54ci)_results_in_H1_displacement_from_chromatin_and_global_chromatin_decondensation,_thereby_promoting_pluripotency_and_stem_cell_maintenance._Promotes_profound_chromatin_decondensation_during_the_innate_immune_response_to_infection_in_neutrophils_by_mediating_formation_of_H1R54ci._Citrullination_of_histone_H3_prevents_their_methylation_by_CARM1_and_HRMT1L2/PRMT1_and_represses_transcription._Citrullinates_EP300/P300_at_'Arg_2142',_which_favors_its_interaction_with_NCOA2/GRIP1._{ECO:0000269|PubMed:15339660,_ECO:0000269|PubMed:15345777,_ECO:0000269|PubMed:15731352,_ECO:0000269|PubMed:16567635,_ECO:0000269|PubMed:18209087,_ECO:0000269|PubMed:21245532}.

Tissue specificity:
TISSUE_SPECIFICITY:_Expressed_in_eosinophils_and_neutrophils,_not_expressed_in_peripheral_monocytes_or_lymphocytes._{ECO:0000269|PubMed:11435484}.
</span></div>    	</td>
	<td ><div class="tooltip">{Rheumatoid_arthritis,_susceptibility_to},_180300__3_<span class="tooltiptext tooltip-bottom">Disease_description.refGene:
 

DISEASE:_Rheumatoid_arthritis_(RA)_[MIM:180300]:_An_inflammatory_disease_with_autoimmune_features_and_a_complex_genetic_component._It_primarily_affects_the_joints_and_is_characterized_by_inflammatory_changes_in_the_synovial_membranes_and_articular_structures,_widespread_fibrinoid_degeneration_of_the_collagen_fibers_in_mesenchymal_tissues,_and_by_atrophy_and_rarefaction_of_bony_structures._{ECO:0000269|PubMed:12833157}._Note=The_gene_represented_in_this_entry_may_be_involved_in_disease_pathogenesis._The_association_to_rheumatoid_arthritis_was_initially_thought_to_result_from_increased_citrullination_of_target_proteins_(PubMed:12833157)._However,_variants_that_have_been_associated_to_rheumatoid_arthritis_(Ser_55,_Ala_82_and_Ala_112)_do_not_affect_the_catalytic_activity_or_the_citrullination_activity_of_PADI4,_suggesting_that_these_variants_may_affect_the_mRNA_stability_rather_than_the_protein_(PubMed:21245532)._{ECO:0000269|PubMed:12833157,_ECO:0000269|PubMed:21245532}.

</span></div> 	</td>
	<td ><div class="tooltip">0.0058<span class="tooltiptext tooltip-bottom">gnomAD_genome_ALL	= 0.0058
gnomAD_genome_AFR	= 0.0022
gnomAD_genome_AMR	= 0.0095
gnomAD_genome_ASJ	= 0.0132
gnomAD_genome_EAS	= 0
gnomAD_genome_FIN	= 0.0072
gnomAD_genome_NFE	= 0.0075
gnomAD_genome_OTH	= 0.0092
</span></div> 	</td>
	<td ><div class="tooltip">0.0081<span class="tooltiptext tooltip-bottom">gnomAD_exome_ALL	= 0.0081
gnomAD_exome_AFR	= 0.0022
gnomAD_exome_AMR	= 0.0047
gnomAD_exome_ASJ	= 0.0108
gnomAD_exome_EAS	= 0
gnomAD_exome_FIN	= 0.0076
gnomAD_exome_NFE	= 0.0107
gnomAD_exome_OTH	= 0.0123
</span></div> 	</td>
	<td ><div class="tooltip">Benign<span class="tooltiptext tooltip-bottom">CLNREVSTAT:
 criteria_provided,_single_submitter

CLNDN:
 not_provided

CLNALLELEID:
 696276

CLNDISDB:
 MedGen:CN517202

</span></div> 	</td>
	<td ><div class="tooltip">Uncertain_significance<span class="tooltiptext tooltip-bottom">PM1	= 1
Located in a mutational hot spot and/or critical and well-established functional domain (e.g., active site of an enzyme) without benign variation

BS1	= 1
Allele frequency is greater than expected for disorder (see Table 6) > 1% in ESP6500all ExAc? need to check more

</span></div> 	</td>
	<td><div class="tooltip">.</div>	</td>
	<td ><div class="tooltip">exonic<span class="tooltiptext tooltip-bottom">ExonicFunc.refGene:
 nonsynonymous_SNV

AAChange.refGene:
 PADI4:NM_012387:exon8:c.A926G:p.Y309C
 PADI4:NM_012387:exon8:c.926A>G:p.Y309C

GeneDetail.refGene:
 .

</span></div> 	</td>
	<td style="background-color:inherit"><div class="tooltip">0/0<span class="tooltiptext tooltip-bottom">CALLER = GATK	 QUALITY = 703.13

Genotype-B00GMSH	 -	 0/0
DP = 45	 AD = 45,0	 AB = 0

Genotype-B00GMSI	 -	 0/0
DP = 50	 AD = 50,0	 AB = 0

Genotype-B00GMSJ	 -	 0/1
DP = 47	 AD = 23,24	 AB = 0.510

</span></div>	</td>
	<td style="background-color:inherit"><div class="tooltip">0/0</div>	</td>
	<td style="background-color:inherit"><div class="tooltip">0/1</div>	</td>
	<td><div class="tooltip">chr1</div>	</td>
	<td><div class="tooltip">17668888</div>	</td>
	<td><div class="tooltip">rs33981382</div>	</td>
	<td><div class="tooltip">A</div>	</td>
	<td><div class="tooltip">G</div>	</td>
	<td><div class="tooltip">PASS</div>	</td>
</tr>
<tr class="ALL OMIMDOM" >
	<td ><div class="tooltip">23<span class="tooltiptext tooltip-bottom">MPA_ranking	= 7
MPA_final_score	= 3.0
MPA_impact	= missense_impact
MPA_adjusted	= 3.0
MPA_available	= 10
MPA_deleterious	= 3

--- SPLICE ---
dbscSNV_ADA_SCORE	= .
dbscSNV_RF_SCORE	= .
spliceai_filtered	= .

--- MISSENSE ---
LRT_pred	= U
SIFT_pred	= D
FATHMM_pred	= T
MetaLR_pred	= T
MetaSVM_pred	= T
PROVEAN_pred	= N
Polyphen2_HDIV_pred	= P
Polyphen2_HVAR_pred	= B
MutationTaster_pred	= D
fathmm-MKL_coding_pred	= D
</span></div>	</td>
	<td><div class="tooltip">3_missense_impact</div>	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom"></span></div> 	</td>
	<td style="background-color:#FF0000"><div class="tooltip">PRDM16<span class="tooltiptext tooltip-bottom">LOEUF = 0.187
LOEUF_decile = 0.0

oe_lof = 0.081747
oe_lof_lower = 0.039

Missense_upper = 0.925

Function Description:
FUNCTION:_Binds_DNA_and_functions_as_a_transcriptional_regulator_(PubMed:12816872)._Displays_histone_methyltransferase_activity_and_monomethylates_'Lys_9'_of_histone_H3_(H3K9me1)_in_vitro_(By_similarity)._Probably_catalyzes_the_monomethylation_of_free_histone_H3_in_the_cytoplasm_which_is_then_transported_to_the_nucleus_and_incorporated_into_nucleosomes_where_SUV39H_methyltransferases_use_it_as_a_substrate_to_catalyze_histone_H3_'Lys_9'_trimethylation_(By_similarity)._Likely_to_be_one_of_the_primary_histone_methyltransferases_along_with_MECOM/PRDM3_that_direct_cytoplasmic_H3K9me1_methylation_(By_similarity)._Functions_in_the_differentiation_of_brown_adipose_tissue_(BAT)_which_is_specialized_in_dissipating_chemical_energy_in_the_form_of_heat_in_response_to_cold_or_excess_feeding_while_white_adipose_tissue_(WAT)_is_specialized_in_the_storage_of_excess_energy_and_the_control_of_systemic_metabolism_(By_similarity)._Together_with_CEBPB,_regulates_the_differentiation_of_myoblastic_precursors_into_brown_adipose_cells_(By_similarity)._Functions_as_a_repressor_of_TGF_beta_signaling_(PubMed:19049980)._{ECO:0000250|UniProtKB:A2A935,_ECO:0000269|PubMed:12816872,_ECO:0000269|PubMed:19049980}.;_FUNCTION:_[Isoform_4]:_Binds_DNA_and_functions_as_a_transcriptional_regulator_(PubMed:12816872)._Functions_as_a_repressor_of_TGF_beta_signaling_(PubMed:14656887)._May_regulate_granulocyte_differentiation_(PubMed:12816872)._{ECO:0000269|PubMed:12816872,_ECO:0000269|PubMed:14656887}.

Tissue specificity:
TISSUE_SPECIFICITY:_Expressed_in_uterus_and_kidney._Expressed_in_both_cardiomyocytes_and_interstitial_cells._{ECO:0000269|PubMed:11050005,_ECO:0000269|PubMed:12816872,_ECO:0000269|PubMed:23768516}.
</span></div>    	</td>
	<td ><div class="tooltip">Left_ventricular_noncompaction_8,_615373__3_,_Autosomal_dominant;_Cardiomyopathy,_dilated,_1LL,_615373__3_,_Autosomal_dominant<span class="tooltiptext tooltip-bottom">Disease_description.refGene:
 

DISEASE:_Left_ventricular_non_compaction_8_(LVNC8)_[MIM:615373]:_A_form_of_left_ventricular_non_compaction,_a_cardiomyopathy_due_to_myocardial_morphogenesis_arrest_and_characterized_by_a_hypertrophic_left_ventricle,_a_severely_thickened_2_layered_myocardium,_numerous_prominent_trabeculations,_deep_intertrabecular_recesses,_and_poor_systolic_function._Clinical_manifestations_are_variable._Some_affected_individuals_experience_no_symptoms_at_all,_others_develop_heart_failure._In_some_cases,_left_ventricular_non_compaction_is_associated_with_other_congenital_heart_anomalies._LVNC8_is_an_autosomal_dominant_condition._{ECO:0000269|PubMed:23768516}._Note=The_disease_is_caused_by_mutations_affecting_the_gene_represented_in_this_entry.;_

DISEASE:_Cardiomyopathy,_dilated_1LL_(CMD1LL)_[MIM:615373]:_A_disorder_characterized_by_ventricular_dilation_and_impaired_systolic_function,_resulting_in_congestive_heart_failure_and_arrhythmia._Patients_are_at_risk_of_premature_death._{ECO:0000269|PubMed:23768516}._Note=The_disease_is_caused_by_mutations_affecting_the_gene_represented_in_this_entry.;_

DISEASE:_Note=A_chromosomal_aberration_involving_PRDM16_is_found_in_myelodysplastic_syndrome_(MDS)_and_acute_myeloid_leukemia_(AML)._Reciprocal_translocation_t(1;3)(p36;q21)._Isoform_4_is_specifically_expressed_in_adult_T_cell_leukemia._{ECO:0000269|PubMed:11050005,_ECO:0000269|PubMed:12557231}.

</span></div> 	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom">gnomAD_genome_ALL	= .
gnomAD_genome_AFR	= .
gnomAD_genome_AMR	= .
gnomAD_genome_ASJ	= .
gnomAD_genome_EAS	= .
gnomAD_genome_FIN	= .
gnomAD_genome_NFE	= .
gnomAD_genome_OTH	= .
</span></div> 	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom">gnomAD_exome_ALL	= .
gnomAD_exome_AFR	= .
gnomAD_exome_AMR	= .
gnomAD_exome_ASJ	= .
gnomAD_exome_EAS	= .
gnomAD_exome_FIN	= .
gnomAD_exome_NFE	= .
gnomAD_exome_OTH	= .
</span></div> 	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom">CLNREVSTAT:
 .

CLNDN:
 .

CLNALLELEID:
 .

CLNDISDB:
 .

</span></div> 	</td>
	<td ><div class="tooltip">Uncertain_significance<span class="tooltiptext tooltip-bottom">PM1	= 1
Located in a mutational hot spot and/or critical and well-established functional domain (e.g., active site of an enzyme) without benign variation

PM2	= 1
Absent from controls (or at extremely low frequency if recessive) (Table 6) in Exome Sequencing Project, 1000 Genomes Project, or Exome Aggregation Consortium

</span></div> 	</td>
	<td><div class="tooltip">.</div>	</td>
	<td ><div class="tooltip">exonic<span class="tooltiptext tooltip-bottom">ExonicFunc.refGene:
 nonsynonymous_SNV

AAChange.refGene:
 PRDM16:NM_022114:exon2:c.G367A:p.E123K
 PRDM16:NM_199454:exon2:c.G367A:p.E123K
 PRDM16:NM_022114:exon2:c.367G>A:p.E123K
 PRDM16:NM_199454:exon2:c.367G>A:p.E123K

GeneDetail.refGene:
 .

</span></div> 	</td>
	<td style="background-color:inherit"><div class="tooltip">0/0<span class="tooltiptext tooltip-bottom">CALLER = GATK	 QUALITY = 2442.13

Genotype-B00GMSH	 -	 0/0
DP = 138	 AD = 138,0	 AB = 0

Genotype-B00GMSI	 -	 0/0
DP = 154	 AD = 154,0	 AB = 0

Genotype-B00GMSJ	 -	 0/1
DP = 149	 AD = 69,80	 AB = 0.536

</span></div>	</td>
	<td style="background-color:inherit"><div class="tooltip">0/0</div>	</td>
	<td style="background-color:inherit"><div class="tooltip">0/1</div>	</td>
	<td><div class="tooltip">chr1</div>	</td>
	<td><div class="tooltip">3103018</div>	</td>
	<td><div class="tooltip">.</div>	</td>
	<td><div class="tooltip">G</div>	</td>
	<td><div class="tooltip">A</div>	</td>
	<td><div class="tooltip">PASS</div>	</td>
</tr>
<tr class="ALL" >
	<td ><div class="tooltip">24<span class="tooltiptext tooltip-bottom">MPA_ranking	= 7
MPA_final_score	= 3.0
MPA_impact	= missense_impact
MPA_adjusted	= 3.0
MPA_available	= 10
MPA_deleterious	= 3

--- SPLICE ---
dbscSNV_ADA_SCORE	= .
dbscSNV_RF_SCORE	= .
spliceai_filtered	= .

--- MISSENSE ---
LRT_pred	= D
SIFT_pred	= T
FATHMM_pred	= T
MetaLR_pred	= T
MetaSVM_pred	= T
PROVEAN_pred	= N
Polyphen2_HDIV_pred	= B
Polyphen2_HVAR_pred	= B
MutationTaster_pred	= D
fathmm-MKL_coding_pred	= D
</span></div>	</td>
	<td><div class="tooltip">3_missense_impact</div>	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom"></span></div> 	</td>
	<td style="background-color:#FF9900"><div class="tooltip">EPHA8<span class="tooltiptext tooltip-bottom">LOEUF = 0.712
LOEUF_decile = 3.0

oe_lof = 0.506
oe_lof_lower = 0.365

Missense_upper = 0.987

Function Description:
FUNCTION:_Receptor_tyrosine_kinase_which_binds_promiscuously_GPI_anchored_ephrin_A_family_ligands_residing_on_adjacent_cells,_leading_to_contact_dependent_bidirectional_signaling_into_neighboring_cells._The_signaling_pathway_downstream_of_the_receptor_is_referred_to_as_forward_signaling_while_the_signaling_pathway_downstream_of_the_ephrin_ligand_is_referred_to_as_reverse_signaling._The_GPI_anchored_ephrin_A_EFNA2,_EFNA3,_and_EFNA5_are_able_to_activate_EPHA8_through_phosphorylation._With_EFNA5_may_regulate_integrin_mediated_cell_adhesion_and_migration_on_fibronectin_substrate_but_also_neurite_outgrowth._During_development_of_the_nervous_system_plays_also_a_role_in_axon_guidance._Downstream_effectors_of_the_EPHA8_signaling_pathway_include_FYN_which_promotes_cell_adhesion_upon_activation_by_EPHA8_and_the_MAP_kinases_in_the_stimulation_of_neurite_outgrowth_(By_similarity)._{ECO:0000250}.;FUNCTION:_Receptor_tyrosine_kinase_which_binds_promiscuously_GPI_anchored_ephrin_A_family_ligands_residing_on_adjacent_cells,_leading_to_contact_dependent_bidirectional_signaling_into_neighboring_cells._The_signaling_pathway_downstream_of_the_receptor_is_referred_to_as_forward_signaling_while_the_signaling_pathway_downstream_of_the_ephrin_ligand_is_referred_to_as_reverse_signaling._The_GPI_anchored_ephrin_A_EFNA2,_EFNA3,_and_EFNA5_are_able_to_activate_EPHA8_through_phosphorylation._With_EFNA5_may_regulate_integrin_mediated_cell_adhesion_and_migration_on_fibronectin_substrate_but_also_neurite_outgrowth._During_development_of_the_nervous_system_plays_also_a_role_in_axon_guidance._Downstream_effectors_of_the_EPHA8_signaling_pathway_include_FYN_which_promotes_cell_adhesion_upon_activation_by_EPHA8_and_the_MAP_kinases_in_the_stimulation_of_neurite_outgrowth_(By_similarity)._{ECO:0000250}.

</span></div>    	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom">Disease_description.refGene:
 .

</span></div> 	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom">gnomAD_genome_ALL	= .
gnomAD_genome_AFR	= .
gnomAD_genome_AMR	= .
gnomAD_genome_ASJ	= .
gnomAD_genome_EAS	= .
gnomAD_genome_FIN	= .
gnomAD_genome_NFE	= .
gnomAD_genome_OTH	= .
</span></div> 	</td>
	<td ><div class="tooltip">1.625e-05<span class="tooltiptext tooltip-bottom">gnomAD_exome_ALL	= 1.625e-05
gnomAD_exome_AFR	= 0
gnomAD_exome_AMR	= 0
gnomAD_exome_ASJ	= 0
gnomAD_exome_EAS	= 0
gnomAD_exome_FIN	= 0
gnomAD_exome_NFE	= 3.583e-05
gnomAD_exome_OTH	= 0
</span></div> 	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom">CLNREVSTAT:
 .

CLNDN:
 .

CLNALLELEID:
 .

CLNDISDB:
 .

</span></div> 	</td>
	<td ><div class="tooltip">Uncertain_significance<span class="tooltiptext tooltip-bottom">PM1	= 1
Located in a mutational hot spot and/or critical and well-established functional domain (e.g., active site of an enzyme) without benign variation

PM2	= 1
Absent from controls (or at extremely low frequency if recessive) (Table 6) in Exome Sequencing Project, 1000 Genomes Project, or Exome Aggregation Consortium

</span></div> 	</td>
	<td><div class="tooltip">.</div>	</td>
	<td ><div class="tooltip">exonic<span class="tooltiptext tooltip-bottom">ExonicFunc.refGene:
 nonsynonymous_SNV

AAChange.refGene:
 EPHA8:NM_020526:exon10:c.A1879G:p.T627A
 EPHA8:NM_020526:exon10:c.1879A>G:p.T627A

GeneDetail.refGene:
 .

</span></div> 	</td>
	<td style="background-color:inherit"><div class="tooltip">0/0<span class="tooltiptext tooltip-bottom">CALLER = GATK	 QUALITY = 1761.13

Genotype-B00GMSH	 -	 0/0
DP = 98	 AD = 98,0	 AB = 0

Genotype-B00GMSI	 -	 0/1
DP = 143	 AD = 75,68	 AB = 0.475

Genotype-B00GMSJ	 -	 0/0
DP = 148	 AD = 148,0	 AB = 0

</span></div>	</td>
	<td style="background-color:inherit"><div class="tooltip">0/1</div>	</td>
	<td style="background-color:inherit"><div class="tooltip">0/0</div>	</td>
	<td><div class="tooltip">chr1</div>	</td>
	<td><div class="tooltip">22923918</div>	</td>
	<td><div class="tooltip">.</div>	</td>
	<td><div class="tooltip">A</div>	</td>
	<td><div class="tooltip">G</div>	</td>
	<td><div class="tooltip">PASS</div>	</td>
</tr>
<tr class="ALL OMIMDOM OMIMREC  PATHO1" >
	<td ><div class="tooltip">25<span class="tooltiptext tooltip-bottom">MPA_ranking	= 7
MPA_final_score	= 3.0
MPA_impact	= missense_impact
MPA_adjusted	= 3.0
MPA_available	= 10
MPA_deleterious	= 3

--- SPLICE ---
dbscSNV_ADA_SCORE	= .
dbscSNV_RF_SCORE	= .
spliceai_filtered	= .

--- MISSENSE ---
LRT_pred	= N
SIFT_pred	= D
FATHMM_pred	= T
MetaLR_pred	= T
MetaSVM_pred	= T
PROVEAN_pred	= N
Polyphen2_HDIV_pred	= P
Polyphen2_HVAR_pred	= P
MutationTaster_pred	= D
fathmm-MKL_coding_pred	= D
</span></div>	</td>
	<td><div class="tooltip">3_missense_impact</div>	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom"></span></div> 	</td>
	<td style="background-color:#FF9900"><div class="tooltip">SDC3<span class="tooltiptext tooltip-bottom">LOEUF = 0.664
LOEUF_decile = 3.0

oe_lof = 0.21097
oe_lof_lower = 0.085

Missense_upper = 1.076

Function Description:
FUNCTION:_Cell_surface_proteoglycan_that_may_bear_heparan_sulfate_(By_similarity)._May_have_a_role_in_the_organization_of_cell_shape_by_affecting_the_actin_cytoskeleton,_possibly_by_transferring_signals_from_the_cell_surface_in_a_sugar_dependent_mechanism._{ECO:0000250,_ECO:0000269|PubMed:11527150}.

Tissue specificity:
TISSUE_SPECIFICITY:_Expressed_in_the_nervous_system,_the_adrenal_gland,_and_the_spleen._{ECO:0000269|PubMed:11527150}.
</span></div>    	</td>
	<td ><div class="tooltip">{Obesity,_association_with},_601665__3_,_Multifactorial,_Autosomal_recessive,_Autosomal_dominant<span class="tooltiptext tooltip-bottom">Disease_description.refGene:
 .

</span></div> 	</td>
	<td ><div class="tooltip">0.0097<span class="tooltiptext tooltip-bottom">gnomAD_genome_ALL	= 0.0097
gnomAD_genome_AFR	= 0.0021
gnomAD_genome_AMR	= 0.0012
gnomAD_genome_ASJ	= 0.0033
gnomAD_genome_EAS	= 0.0006
gnomAD_genome_FIN	= 0.0052
gnomAD_genome_NFE	= 0.0167
gnomAD_genome_OTH	= 0.0123
</span></div> 	</td>
	<td ><div class="tooltip">0.0090<span class="tooltiptext tooltip-bottom">gnomAD_exome_ALL	= 0.0090
gnomAD_exome_AFR	= 0.0024
gnomAD_exome_AMR	= 0.0036
gnomAD_exome_ASJ	= 0.0051
gnomAD_exome_EAS	= 0
gnomAD_exome_FIN	= 0.0064
gnomAD_exome_NFE	= 0.0138
gnomAD_exome_OTH	= 0.0077
</span></div> 	</td>
	<td ><div class="tooltip">Benign<span class="tooltiptext tooltip-bottom">CLNREVSTAT:
 criteria_provided,_single_submitter

CLNDN:
 not_provided

CLNALLELEID:
 707336

CLNDISDB:
 MedGen:CN517202

</span></div> 	</td>
	<td ><div class="tooltip">Uncertain_significance<span class="tooltiptext tooltip-bottom">BS1	= 1
Allele frequency is greater than expected for disorder (see Table 6) > 1% in ESP6500all ExAc? need to check more

</span></div> 	</td>
	<td><div class="tooltip">.</div>	</td>
	<td ><div class="tooltip">exonic<span class="tooltiptext tooltip-bottom">ExonicFunc.refGene:
 nonsynonymous_SNV

AAChange.refGene:
 SDC3:NM_014654:exon3:c.G286A:p.A96T
 SDC3:NM_014654:exon3:c.286G>A:p.A96T

GeneDetail.refGene:
 .

</span></div> 	</td>
	<td style="background-color:inherit"><div class="tooltip">0/0<span class="tooltiptext tooltip-bottom">CALLER = GATK	 QUALITY = 1220.13

Genotype-B00GMSH	 -	 0/0
DP = 87	 AD = 87,0	 AB = 0

Genotype-B00GMSI	 -	 0/1
DP = 92	 AD = 50,42	 AB = 0.456

Genotype-B00GMSJ	 -	 0/0
DP = 94	 AD = 94,0	 AB = 0

</span></div>	</td>
	<td style="background-color:inherit"><div class="tooltip">0/1</div>	</td>
	<td style="background-color:inherit"><div class="tooltip">0/0</div>	</td>
	<td><div class="tooltip">chr1</div>	</td>
	<td><div class="tooltip">31349983</div>	</td>
	<td><div class="tooltip">rs41269523</div>	</td>
	<td><div class="tooltip">C</div>	</td>
	<td><div class="tooltip">T</div>	</td>
	<td><div class="tooltip">PASS</div>	</td>
</tr>
<tr class="ALL" >
	<td ><div class="tooltip">26<span class="tooltiptext tooltip-bottom">MPA_ranking	= 8
MPA_final_score	= 2
MPA_impact	= splice_impact
MPA_adjusted	= 0
MPA_available	= 0
MPA_deleterious	= 0

--- SPLICE ---
dbscSNV_ADA_SCORE	= .
dbscSNV_RF_SCORE	= .
spliceai_filtered	= .

--- MISSENSE ---
LRT_pred	= .
SIFT_pred	= .
FATHMM_pred	= .
MetaLR_pred	= .
MetaSVM_pred	= .
PROVEAN_pred	= .
Polyphen2_HDIV_pred	= .
Polyphen2_HVAR_pred	= .
MutationTaster_pred	= .
fathmm-MKL_coding_pred	= .
</span></div>	</td>
	<td><div class="tooltip">splice_impact</div>	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom"></span></div> 	</td>
	<td style="background-color:#FF0000"><div class="tooltip">EIF4G3<span class="tooltiptext tooltip-bottom">LOEUF = 0.112
LOEUF_decile = 0.0

oe_lof = 0.04915
oe_lof_lower = 0.023

Missense_upper = 0.763

Function Description:
FUNCTION:_Probable_component_of_the_protein_complex_eIF4F,_which_is_involved_in_the_recognition_of_the_mRNA_cap,_ATP_dependent_unwinding_of_5'_terminal_secondary_structure_and_recruitment_of_mRNA_to_the_ribosome._Thought_to_be_a_functional_homolog_of_EIF4G1._{ECO:0000269|PubMed:9418880}.

</span></div>    	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom">Disease_description.refGene:
 .

</span></div> 	</td>
	<td ><div class="tooltip">0.0035<span class="tooltiptext tooltip-bottom">gnomAD_genome_ALL	= 0.0035
gnomAD_genome_AFR	= 0.0011
gnomAD_genome_AMR	= 0.0024
gnomAD_genome_ASJ	= 0.0132
gnomAD_genome_EAS	= 0
gnomAD_genome_FIN	= 0.0011
gnomAD_genome_NFE	= 0.0059
gnomAD_genome_OTH	= 0.0010
</span></div> 	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom">gnomAD_exome_ALL	= .
gnomAD_exome_AFR	= .
gnomAD_exome_AMR	= .
gnomAD_exome_ASJ	= .
gnomAD_exome_EAS	= .
gnomAD_exome_FIN	= .
gnomAD_exome_NFE	= .
gnomAD_exome_OTH	= .
</span></div> 	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom">CLNREVSTAT:
 .

CLNDN:
 .

CLNALLELEID:
 .

CLNDISDB:
 .

</span></div> 	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom"></span></div> 	</td>
	<td><div class="tooltip">.</div>	</td>
	<td ><div class="tooltip">splicing;intronic<span class="tooltiptext tooltip-bottom">ExonicFunc.refGene:
 .

AAChange.refGene:
 .

GeneDetail.refGene:
 NM_001198801:exon3:r.spl
 NM_001198802:exon3:r.spl
 NM_001198803:exon3:r.spl

</span></div> 	</td>
	<td style="background-color:inherit"><div class="tooltip">0/0<span class="tooltiptext tooltip-bottom">CALLER = GATK	 QUALITY = 2682.09

Genotype-B00GMSH	 -	 0/0
DP = 170	 AD = 170,0	 AB = 0

Genotype-B00GMSI	 -	 0/0
DP = 189	 AD = 189,0	 AB = 0

Genotype-B00GMSJ	 -	 0/1
DP = 164	 AD = 91,73	 AB = 0.445

</span></div>	</td>
	<td style="background-color:inherit"><div class="tooltip">0/0</div>	</td>
	<td style="background-color:inherit"><div class="tooltip">0/1</div>	</td>
	<td><div class="tooltip">chr1</div>	</td>
	<td><div class="tooltip">21415549</div>	</td>
	<td><div class="tooltip">.</div>	</td>
	<td><div class="tooltip">TCAAA</div>	</td>
	<td><div class="tooltip">T</div>	</td>
	<td><div class="tooltip">PASS</div>	</td>
</tr>
<tr class="ALL" >
	<td ><div class="tooltip">27<span class="tooltiptext tooltip-bottom">MPA_ranking	= 8
MPA_final_score	= 2
MPA_impact	= splice_impact
MPA_adjusted	= 0
MPA_available	= 0
MPA_deleterious	= 0

--- SPLICE ---
dbscSNV_ADA_SCORE	= .
dbscSNV_RF_SCORE	= .
spliceai_filtered	= .

--- MISSENSE ---
LRT_pred	= .
SIFT_pred	= .
FATHMM_pred	= .
MetaLR_pred	= .
MetaSVM_pred	= .
PROVEAN_pred	= .
Polyphen2_HDIV_pred	= .
Polyphen2_HVAR_pred	= .
MutationTaster_pred	= .
fathmm-MKL_coding_pred	= .
</span></div>	</td>
	<td><div class="tooltip">splice_impact</div>	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom"></span></div> 	</td>
	<td style="background-color:#7FFF00"><div class="tooltip">CCDC27<span class="tooltiptext tooltip-bottom">LOEUF = 1.433
LOEUF_decile = 7.0

oe_lof = 1.1008
oe_lof_lower = 0.854

Missense_upper = 1.1

</span></div>    	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom">Disease_description.refGene:
 .

</span></div> 	</td>
	<td ><div class="tooltip">0.0058<span class="tooltiptext tooltip-bottom">gnomAD_genome_ALL	= 0.0058
gnomAD_genome_AFR	= 0.0026
gnomAD_genome_AMR	= 0.0239
gnomAD_genome_ASJ	= 0.0199
gnomAD_genome_EAS	= 0.0049
gnomAD_genome_FIN	= 0.0009
gnomAD_genome_NFE	= 0.0076
gnomAD_genome_OTH	= 0.0071
</span></div> 	</td>
	<td ><div class="tooltip">0.0139<span class="tooltiptext tooltip-bottom">gnomAD_exome_ALL	= 0.0139
gnomAD_exome_AFR	= 0.0028
gnomAD_exome_AMR	= 0.0248
gnomAD_exome_ASJ	= 0.0217
gnomAD_exome_EAS	= 0.0031
gnomAD_exome_FIN	= 0.0015
gnomAD_exome_NFE	= 0.0103
gnomAD_exome_OTH	= 0.0105
</span></div> 	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom">CLNREVSTAT:
 .

CLNDN:
 .

CLNALLELEID:
 .

CLNDISDB:
 .

</span></div> 	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom"></span></div> 	</td>
	<td><div class="tooltip">.</div>	</td>
	<td ><div class="tooltip">splicing;intronic<span class="tooltiptext tooltip-bottom">ExonicFunc.refGene:
 .

AAChange.refGene:
 .

GeneDetail.refGene:
 NM_152492:exon9:r.spl

</span></div> 	</td>
	<td style="background-color:inherit"><div class="tooltip">0/0<span class="tooltiptext tooltip-bottom">CALLER = GATK	 QUALITY = 2856.09

Genotype-B00GMSH	 -	 0/0
DP = 129	 AD = 129,0	 AB = 0

Genotype-B00GMSI	 -	 0/1
DP = 149	 AD = 73,76	 AB = 0.510

Genotype-B00GMSJ	 -	 0/0
DP = 156	 AD = 156,0	 AB = 0

</span></div>	</td>
	<td style="background-color:inherit"><div class="tooltip">0/1</div>	</td>
	<td style="background-color:inherit"><div class="tooltip">0/0</div>	</td>
	<td><div class="tooltip">chr1</div>	</td>
	<td><div class="tooltip">3683083</div>	</td>
	<td><div class="tooltip">rs374337995</div>	</td>
	<td><div class="tooltip">TTC</div>	</td>
	<td><div class="tooltip">T</div>	</td>
	<td><div class="tooltip">PASS</div>	</td>
</tr>
<tr class="ALL  PATHO1 disease2" >
	<td ><div class="tooltip">28<span class="tooltiptext tooltip-bottom">MPA_ranking	= 8
MPA_final_score	= 2
MPA_impact	= splice_impact
MPA_adjusted	= 0
MPA_available	= 0
MPA_deleterious	= 0

--- SPLICE ---
dbscSNV_ADA_SCORE	= .
dbscSNV_RF_SCORE	= .
spliceai_filtered	= .

--- MISSENSE ---
LRT_pred	= .
SIFT_pred	= .
FATHMM_pred	= .
MetaLR_pred	= .
MetaSVM_pred	= .
PROVEAN_pred	= .
Polyphen2_HDIV_pred	= .
Polyphen2_HVAR_pred	= .
MutationTaster_pred	= .
fathmm-MKL_coding_pred	= .
</span></div>	</td>
	<td><div class="tooltip">splice_impact</div>	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom"></span></div> 	</td>
	<td style="background-color:#FFFFFF"><div class="tooltip">TMEM51-AS1;KAZN<span class="tooltiptext tooltip-bottom">.</span></div>    	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom">Disease_description.refGene:
 .

</span></div> 	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom">gnomAD_genome_ALL	= .
gnomAD_genome_AFR	= .
gnomAD_genome_AMR	= .
gnomAD_genome_ASJ	= .
gnomAD_genome_EAS	= .
gnomAD_genome_FIN	= .
gnomAD_genome_NFE	= .
gnomAD_genome_OTH	= .
</span></div> 	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom">gnomAD_exome_ALL	= .
gnomAD_exome_AFR	= .
gnomAD_exome_AMR	= .
gnomAD_exome_ASJ	= .
gnomAD_exome_EAS	= .
gnomAD_exome_FIN	= .
gnomAD_exome_NFE	= .
gnomAD_exome_OTH	= .
</span></div> 	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom">CLNREVSTAT:
 .

CLNDN:
 .

CLNALLELEID:
 .

CLNDISDB:
 .

</span></div> 	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom"></span></div> 	</td>
	<td><div class="tooltip">.</div>	</td>
	<td ><div class="tooltip">ncRNA_exonic;splicing<span class="tooltiptext tooltip-bottom">ExonicFunc.refGene:
 .

AAChange.refGene:
 .

GeneDetail.refGene:
 NM_201628:exon14:c.2163+46->AGGTCTTTTTTTTCTTCTCCCAGCGGC

</span></div> 	</td>
	<td style="background-color:inherit"><div class="tooltip">0/1<span class="tooltiptext tooltip-bottom">CALLER = GATK	 QUALITY = 2432.86

Genotype-B00GMSH	 -	 0/1
DP = 32	 AD = 14,18	 AB = 0.562

Genotype-B00GMSI	 -	 0/0
DP = 43	 AD = 43,0	 AB = 0

Genotype-B00GMSJ	 -	 1/1
DP = 36	 AD = 0,36	 AB = 1

</span></div>	</td>
	<td style="background-color:inherit"><div class="tooltip">0/0</div>	</td>
	<td style="background-color:inherit"><div class="tooltip">1/1</div>	</td>
	<td><div class="tooltip">chr1</div>	</td>
	<td><div class="tooltip">15439083</div>	</td>
	<td><div class="tooltip">rs140963536</div>	</td>
	<td><div class="tooltip">A</div>	</td>
	<td><div class="tooltip">AAGGTCTTTTTTTTCTTCTCCCAGCGGC</div>	</td>
	<td><div class="tooltip">PASS</div>	</td>
</tr>
<tr class="ALL OMIMREC" >
	<td ><div class="tooltip">29<span class="tooltiptext tooltip-bottom">MPA_ranking	= 9
MPA_final_score	= 2.0
MPA_impact	= missense_impact
MPA_adjusted	= 2.0
MPA_available	= 10
MPA_deleterious	= 2

--- SPLICE ---
dbscSNV_ADA_SCORE	= .
dbscSNV_RF_SCORE	= .
spliceai_filtered	= .

--- MISSENSE ---
LRT_pred	= N
SIFT_pred	= T
FATHMM_pred	= T
MetaLR_pred	= T
MetaSVM_pred	= T
PROVEAN_pred	= N
Polyphen2_HDIV_pred	= P
Polyphen2_HVAR_pred	= B
MutationTaster_pred	= D
fathmm-MKL_coding_pred	= D
</span></div>	</td>
	<td><div class="tooltip">2_missense_impact</div>	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom"></span></div> 	</td>
	<td style="background-color:#FF6600"><div class="tooltip">AGRN<span class="tooltiptext tooltip-bottom">LOEUF = 0.435
LOEUF_decile = 2.0

oe_lof = 0.31914000000000003
oe_lof_lower = 0.237

Missense_upper = 1.029

Function Description:
FUNCTION:_[Isoform_1]:_heparan_sulfate_basal_lamina_glycoprotein_that_plays_a_central_role_in_the_formation_and_the_maintenance_of_the_neuromuscular_junction_(NMJ)_and_directs_key_events_in_postsynaptic_differentiation._Component_of_the_AGRN_LRP4_receptor_complex_that_induces_the_phosphorylation_and_activation_of_MUSK._The_activation_of_MUSK_in_myotubes_induces_the_formation_of_NMJ_by_regulating_different_processes_including_the_transcription_of_specific_genes_and_the_clustering_of_AChR_in_the_postsynaptic_membrane._Calcium_ions_are_required_for_maximal_AChR_clustering._AGRN_function_in_neurons_is_highly_regulated_by_alternative_splicing,_glycan_binding_and_proteolytic_processing._Modulates_calcium_ion_homeostasis_in_neurons,_specifically_by_inducing_an_increase_in_cytoplasmic_calcium_ions._Functions_differentially_in_the_central_nervous_system_(CNS)_by_inhibiting_the_alpha(3)_subtype_of_Naplus/Kplus_ATPase_and_evoking_depolarization_at_CNS_synapses._This_secreted_isoform_forms_a_bridge,_after_release_from_motor_neurons,_to_basal_lamina_through_binding_laminin_via_the_NtA_domain.;_FUNCTION:_[Isoform_2]:_transmembrane_form_that_is_the_predominate_form_in_neurons_of_the_brain,_induces_dendritic_filopodia_and_synapse_formation_in_mature_hippocampal_neurons_in_large_part_due_to_the_attached_glycosaminoglycan_chains_and_the_action_of_Rho_family_GTPases.;_FUNCTION:_Isoform_1,_isoform_4_and_isoform_5:_neuron_specific_(zplus)_isoforms_that_contain_C_terminal_insertions_of_8_19_AA_are_potent_activators_of_AChR_clustering._Isoform_5,_agrin_(zplus8),_containing_the_8_AA_insert,_forms_a_receptor_complex_in_myotubules_containing_the_neuronal_AGRN,_the_muscle_specific_kinase_MUSK_and_LRP4,_a_member_of_the_LDL_receptor_family._The_splicing_factors,_NOVA1_and_NOVA2,_regulate_AGRN_splicing_and_production_of_the_'z'_isoforms.;_FUNCTION:_Isoform_3_and_isoform_6:_lack_any_'z'_insert,_are_muscle_specific_and_may_be_involved_in_endothelial_cell_differentiation.;_FUNCTION:_[Agrin_N_terminal_110_kDa_subunit]:_is_involved_in_regulation_of_neurite_outgrowth_probably_due_to_the_presence_of_the_glycosaminoglcan_(GAG)_side_chains_of_heparan_and_chondroitin_sulfate_attached_to_the_Ser/Thr__and_Gly/Ser_rich_regions._Also_involved_in_modulation_of_growth_factor_signaling_(By_similarity)._{ECO:0000250,_ECO:0000269|PubMed:19631309,_ECO:0000269|PubMed:21969364}.;_FUNCTION:_[Agrin_C_terminal_22_kDa_fragment]:_this_released_fragment_is_important_for_agrin_signaling_and_to_exert_a_maximal_dendritic_filopodia_inducing_effect._All_'z'_splice_variants_(zplus)_of_this_fragment_also_show_an_increase_in_the_number_of_filopodia.

Tissue specificity:
TISSUE_SPECIFICITY:_Expressed_in_basement_membranes_of_lung_and_kidney._Muscle__and_neuron_specific_isoforms_are_found._Isoforms_(yplus)_with_the_4_AA_insert_and_(zplus8)_isoforms_with_the_8_AA_insert_are_all_neuron_specific._Isoforms_(zplus11)_are_found_in_both_neuronal_and_non_neuronal_tissues._{ECO:0000269|PubMed:16487930,_ECO:0000269|PubMed:20551380,_ECO:0000269|PubMed:9652404}.
</span></div>    	</td>
	<td ><div class="tooltip">Myasthenic_syndrome,_congenital,_8,_with_pre__and_postsynaptic_defects,_615120__3_,_Autosomal_recessive<span class="tooltiptext tooltip-bottom">Disease_description.refGene:
 

DISEASE:_Myasthenic_syndrome,_congenital,_8_(CMS8)_[MIM:615120]:_A_form_of_congenital_myasthenic_syndrome,_a_group_of_disorders_characterized_by_failure_of_neuromuscular_transmission,_including_pre_synaptic,_synaptic,_and_post_synaptic_disorders_that_are_not_of_autoimmune_origin._Clinical_features_are_easy_fatigability_and_muscle_weakness._CMS8_is_an_autosomal_recessive_disease_characterized_by_prominent_defects_of_both_the_pre__and_postsynaptic_regions._Affected_individuals_have_onset_of_muscle_weakness_in_early_childhood;_the_severity_of_the_weakness_and_muscles_affected_is_variable._{ECO:0000269|PubMed:19631309,_ECO:0000269|PubMed:22205389,_ECO:0000269|PubMed:24951643}._Note=The_disease_is_caused_by_mutations_affecting_the_gene_represented_in_this_entry.

</span></div> 	</td>
	<td ><div class="tooltip">9.752e-05<span class="tooltiptext tooltip-bottom">gnomAD_genome_ALL	= 9.752e-05
gnomAD_genome_AFR	= 0
gnomAD_genome_AMR	= 0
gnomAD_genome_ASJ	= 0
gnomAD_genome_EAS	= 0
gnomAD_genome_FIN	= 0
gnomAD_genome_NFE	= 0.0002
gnomAD_genome_OTH	= 0
</span></div> 	</td>
	<td ><div class="tooltip">0.0002<span class="tooltiptext tooltip-bottom">gnomAD_exome_ALL	= 0.0002
gnomAD_exome_AFR	= 6.942e-05
gnomAD_exome_AMR	= 0.0005
gnomAD_exome_ASJ	= 0
gnomAD_exome_EAS	= 0.0001
gnomAD_exome_FIN	= 0
gnomAD_exome_NFE	= 0.0001
gnomAD_exome_OTH	= 0.0011
</span></div> 	</td>
	<td ><div class="tooltip">Uncertain_significance<span class="tooltiptext tooltip-bottom">CLNREVSTAT:
 criteria_provided,_multiple_submitters,_no_conflicts

CLNDN:
 Myasthenic_syndrome,_congenital,_8|not_provided

CLNALLELEID:
 514887

CLNDISDB:
 MONDO:MONDO:0014052,MedGen:C3808739,OMIM:615120|MedGen:CN517202

</span></div> 	</td>
	<td ><div class="tooltip">Uncertain_significance<span class="tooltiptext tooltip-bottom">PM1	= 1
Located in a mutational hot spot and/or critical and well-established functional domain (e.g., active site of an enzyme) without benign variation

PM2	= 1
Absent from controls (or at extremely low frequency if recessive) (Table 6) in Exome Sequencing Project, 1000 Genomes Project, or Exome Aggregation Consortium

</span></div> 	</td>
	<td><div class="tooltip">.</div>	</td>
	<td ><div class="tooltip">exonic<span class="tooltiptext tooltip-bottom">ExonicFunc.refGene:
 nonsynonymous_SNV

AAChange.refGene:
 AGRN:NM_198576:exon10:c.G1865A:p.R622Q
 AGRN:NM_198576:exon10:c.1865G>A:p.R622Q

GeneDetail.refGene:
 .

</span></div> 	</td>
	<td style="background-color:inherit"><div class="tooltip">0/0<span class="tooltiptext tooltip-bottom">CALLER = GATK	 QUALITY = 1416.13

Genotype-B00GMSH	 -	 0/0
DP = 108	 AD = 108,0	 AB = 0

Genotype-B00GMSI	 -	 0/0
DP = 108	 AD = 108,0	 AB = 0

Genotype-B00GMSJ	 -	 0/1
DP = 96	 AD = 47,49	 AB = 0.510

</span></div>	</td>
	<td style="background-color:inherit"><div class="tooltip">0/0</div>	</td>
	<td style="background-color:inherit"><div class="tooltip">0/1</div>	</td>
	<td><div class="tooltip">chr1</div>	</td>
	<td><div class="tooltip">979269</div>	</td>
	<td><div class="tooltip">rs140789461</div>	</td>
	<td><div class="tooltip">G</div>	</td>
	<td><div class="tooltip">A</div>	</td>
	<td><div class="tooltip">PASS</div>	</td>
</tr>
<tr class="ALL" >
	<td ><div class="tooltip">30<span class="tooltiptext tooltip-bottom">MPA_ranking	= 9
MPA_final_score	= 2.0
MPA_impact	= missense_impact
MPA_adjusted	= 2.0
MPA_available	= 10
MPA_deleterious	= 2

--- SPLICE ---
dbscSNV_ADA_SCORE	= .
dbscSNV_RF_SCORE	= .
spliceai_filtered	= .

--- MISSENSE ---
LRT_pred	= N
SIFT_pred	= D
FATHMM_pred	= T
MetaLR_pred	= T
MetaSVM_pred	= T
PROVEAN_pred	= N
Polyphen2_HDIV_pred	= P
Polyphen2_HVAR_pred	= B
MutationTaster_pred	= N
fathmm-MKL_coding_pred	= D
</span></div>	</td>
	<td><div class="tooltip">2_missense_impact</div>	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom"></span></div> 	</td>
	<td style="background-color:#7FFF00"><div class="tooltip">HTR6<span class="tooltiptext tooltip-bottom">LOEUF = 1.275
LOEUF_decile = 7.0

oe_lof = 0.67952
oe_lof_lower = 0.385

Missense_upper = 1.042

Function Description:
FUNCTION:_This_is_one_of_the_several_different_receptors_for_5_hydroxytryptamine_(serotonin),_a_biogenic_hormone_that_functions_as_a_neurotransmitter,_a_hormone,_and_a_mitogen._The_activity_of_this_receptor_is_mediated_by_G_proteins_that_stimulate_adenylate_cyclase._It_has_a_high_affinity_for_tricyclic_psychotropic_drugs_(By_similarity)._Controls_pyramidal_neurons_migration_during_corticogenesis,_through_the_regulation_of_CDK5_activity_(By_similarity)._Is_an_activator_of_TOR_signaling_(PubMed:23027611)._{ECO:0000250|UniProtKB:P31388,_ECO:0000250|UniProtKB:Q9R1C8,_ECO:0000269|PubMed:23027611}.

Tissue specificity:
TISSUE_SPECIFICITY:_Expressed_in_several_human_brain_regions,_most_prominently_in_the_caudate_nucleus.
</span></div>    	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom">Disease_description.refGene:
 .

</span></div> 	</td>
	<td ><div class="tooltip">0.0015<span class="tooltiptext tooltip-bottom">gnomAD_genome_ALL	= 0.0015
gnomAD_genome_AFR	= 0.0001
gnomAD_genome_AMR	= 0
gnomAD_genome_ASJ	= 0
gnomAD_genome_EAS	= 0.0006
gnomAD_genome_FIN	= 0.0014
gnomAD_genome_NFE	= 0.0025
gnomAD_genome_OTH	= 0.0020
</span></div> 	</td>
	<td ><div class="tooltip">0.0019<span class="tooltiptext tooltip-bottom">gnomAD_exome_ALL	= 0.0019
gnomAD_exome_AFR	= 0.0002
gnomAD_exome_AMR	= 0.0010
gnomAD_exome_ASJ	= 0.0008
gnomAD_exome_EAS	= 0
gnomAD_exome_FIN	= 0.0009
gnomAD_exome_NFE	= 0.0030
gnomAD_exome_OTH	= 0.0010
</span></div> 	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom">CLNREVSTAT:
 .

CLNDN:
 .

CLNALLELEID:
 .

CLNDISDB:
 .

</span></div> 	</td>
	<td ><div class="tooltip">Uncertain_significance<span class="tooltiptext tooltip-bottom">BS1	= 1
Allele frequency is greater than expected for disorder (see Table 6) > 1% in ESP6500all ExAc? need to check more

</span></div> 	</td>
	<td><div class="tooltip">.</div>	</td>
	<td ><div class="tooltip">exonic<span class="tooltiptext tooltip-bottom">ExonicFunc.refGene:
 nonsynonymous_SNV

AAChange.refGene:
 HTR6:NM_000871:exon3:c.C1123A:p.P375T
 HTR6:NM_000871:exon3:c.1123C>A:p.P375T

GeneDetail.refGene:
 .

</span></div> 	</td>
	<td style="background-color:inherit"><div class="tooltip">0/0<span class="tooltiptext tooltip-bottom">CALLER = GATK	 QUALITY = 1157.13

Genotype-B00GMSH	 -	 0/0
DP = 64	 AD = 64,0	 AB = 0

Genotype-B00GMSI	 -	 0/1
DP = 70	 AD = 32,38	 AB = 0.542

Genotype-B00GMSJ	 -	 0/0
DP = 80	 AD = 80,0	 AB = 0

</span></div>	</td>
	<td style="background-color:inherit"><div class="tooltip">0/1</div>	</td>
	<td style="background-color:inherit"><div class="tooltip">0/0</div>	</td>
	<td><div class="tooltip">chr1</div>	</td>
	<td><div class="tooltip">20005661</div>	</td>
	<td><div class="tooltip">rs8192533</div>	</td>
	<td><div class="tooltip">C</div>	</td>
	<td><div class="tooltip">A</div>	</td>
	<td><div class="tooltip">PASS</div>	</td>
</tr>
<tr class="ALL" >
	<td ><div class="tooltip">31<span class="tooltiptext tooltip-bottom">MPA_ranking	= 9
MPA_final_score	= 2.0
MPA_impact	= missense_impact
MPA_adjusted	= 2.0
MPA_available	= 10
MPA_deleterious	= 2

--- SPLICE ---
dbscSNV_ADA_SCORE	= .
dbscSNV_RF_SCORE	= .
spliceai_filtered	= .

--- MISSENSE ---
LRT_pred	= D
SIFT_pred	= T
FATHMM_pred	= T
MetaLR_pred	= T
MetaSVM_pred	= T
PROVEAN_pred	= N
Polyphen2_HDIV_pred	= B
Polyphen2_HVAR_pred	= B
MutationTaster_pred	= N
fathmm-MKL_coding_pred	= D
</span></div>	</td>
	<td><div class="tooltip">2_missense_impact</div>	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom"></span></div> 	</td>
	<td style="background-color:#7FFF00"><div class="tooltip">TRIM63<span class="tooltiptext tooltip-bottom">LOEUF = 1.469
LOEUF_decile = 7.0

oe_lof = 1.0021
oe_lof_lower = 0.698

Missense_upper = 1.16

Function Description:
FUNCTION:_E3_ubiquitin_ligase._Mediates_the_ubiquitination_and_subsequent_proteasomal_degradation_of_CKM,_GMEB1_and_HIBADH._Regulates_the_proteasomal_degradation_of_muscle_proteins_under_amino_acid_starvation,_where_muscle_protein_is_catabolized_to_provide_other_organs_with_amino_acids._Inhibits_de_novo_skeletal_muscle_protein_synthesis_under_amino_acid_starvation._Regulates_proteasomal_degradation_of_cardiac_troponin_I/TNNI3_and_probably_of_other_sarcomeric_associated_proteins._May_play_a_role_in_striated_muscle_atrophy_and_hypertrophy_by_regulating_an_anti_hypertrophic_PKC_mediated_signaling_pathway._May_regulate_the_organization_of_myofibrils_through_TTN_in_muscle_cells._{ECO:0000269|PubMed:11927605,_ECO:0000269|PubMed:18222470}.

Tissue specificity:
TISSUE_SPECIFICITY:_Muscle_specific._Selectively_expressed_in_heart_and_skeletal_muscle._Also_expressed_in_the_iris._{ECO:0000269|PubMed:11243782,_ECO:0000269|PubMed:11283016,_ECO:0000269|PubMed:11679633,_ECO:0000269|PubMed:12107412}.
</span></div>    	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom">Disease_description.refGene:
 .

</span></div> 	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom">gnomAD_genome_ALL	= .
gnomAD_genome_AFR	= .
gnomAD_genome_AMR	= .
gnomAD_genome_ASJ	= .
gnomAD_genome_EAS	= .
gnomAD_genome_FIN	= .
gnomAD_genome_NFE	= .
gnomAD_genome_OTH	= .
</span></div> 	</td>
	<td ><div class="tooltip">4.092e-06<span class="tooltiptext tooltip-bottom">gnomAD_exome_ALL	= 4.092e-06
gnomAD_exome_AFR	= 0
gnomAD_exome_AMR	= 0
gnomAD_exome_ASJ	= 0
gnomAD_exome_EAS	= 0
gnomAD_exome_FIN	= 0
gnomAD_exome_NFE	= 9.084e-06
gnomAD_exome_OTH	= 0
</span></div> 	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom">CLNREVSTAT:
 .

CLNDN:
 .

CLNALLELEID:
 .

CLNDISDB:
 .

</span></div> 	</td>
	<td ><div class="tooltip">Uncertain_significance<span class="tooltiptext tooltip-bottom"></span></div> 	</td>
	<td><div class="tooltip">.</div>	</td>
	<td ><div class="tooltip">exonic<span class="tooltiptext tooltip-bottom">ExonicFunc.refGene:
 nonsynonymous_SNV

AAChange.refGene:
 TRIM63:NM_032588:exon5:c.C609A:p.H203Q
 TRIM63:NM_032588:exon5:c.609C>A:p.H203Q

GeneDetail.refGene:
 .

</span></div> 	</td>
	<td style="background-color:inherit"><div class="tooltip">0/0<span class="tooltiptext tooltip-bottom">CALLER = GATK	 QUALITY = 2122.13

Genotype-B00GMSH	 -	 0/0
DP = 165	 AD = 165,0	 AB = 0

Genotype-B00GMSI	 -	 0/0
DP = 171	 AD = 171,0	 AB = 0

Genotype-B00GMSJ	 -	 0/1
DP = 165	 AD = 88,77	 AB = 0.466

</span></div>	</td>
	<td style="background-color:inherit"><div class="tooltip">0/0</div>	</td>
	<td style="background-color:inherit"><div class="tooltip">0/1</div>	</td>
	<td><div class="tooltip">chr1</div>	</td>
	<td><div class="tooltip">26385103</div>	</td>
	<td><div class="tooltip">.</div>	</td>
	<td><div class="tooltip">G</div>	</td>
	<td><div class="tooltip">T</div>	</td>
	<td><div class="tooltip">PASS</div>	</td>
</tr>
<tr class="ALL" >
	<td ><div class="tooltip">32<span class="tooltiptext tooltip-bottom">MPA_ranking	= 9
MPA_final_score	= 1.0
MPA_impact	= missense_impact
MPA_adjusted	= 1.0
MPA_available	= 10
MPA_deleterious	= 1

--- SPLICE ---
dbscSNV_ADA_SCORE	= .
dbscSNV_RF_SCORE	= .
spliceai_filtered	= .

--- MISSENSE ---
LRT_pred	= N
SIFT_pred	= T
FATHMM_pred	= D
MetaLR_pred	= T
MetaSVM_pred	= T
PROVEAN_pred	= N
Polyphen2_HDIV_pred	= B
Polyphen2_HVAR_pred	= B
MutationTaster_pred	= N
fathmm-MKL_coding_pred	= N
</span></div>	</td>
	<td><div class="tooltip">1_missense_impact</div>	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom"></span></div> 	</td>
	<td style="background-color:#FFCC00"><div class="tooltip">NIPAL3<span class="tooltiptext tooltip-bottom">LOEUF = 0.844
LOEUF_decile = 4.0

oe_lof = 0.4975
oe_lof_lower = 0.306

Missense_upper = 0.984

</span></div>    	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom">Disease_description.refGene:
 .

</span></div> 	</td>
	<td ><div class="tooltip">0.0029<span class="tooltiptext tooltip-bottom">gnomAD_genome_ALL	= 0.0029
gnomAD_genome_AFR	= 0.0013
gnomAD_genome_AMR	= 0
gnomAD_genome_ASJ	= 0
gnomAD_genome_EAS	= 0.0086
gnomAD_genome_FIN	= 0.0011
gnomAD_genome_NFE	= 0.0039
gnomAD_genome_OTH	= 0.0020
</span></div> 	</td>
	<td ><div class="tooltip">0.0028<span class="tooltiptext tooltip-bottom">gnomAD_exome_ALL	= 0.0028
gnomAD_exome_AFR	= 0.0005
gnomAD_exome_AMR	= 0.0004
gnomAD_exome_ASJ	= 0.0010
gnomAD_exome_EAS	= 0.0103
gnomAD_exome_FIN	= 0.0005
gnomAD_exome_NFE	= 0.0038
gnomAD_exome_OTH	= 0.0013
</span></div> 	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom">CLNREVSTAT:
 .

CLNDN:
 .

CLNALLELEID:
 .

CLNDISDB:
 .

</span></div> 	</td>
	<td ><div class="tooltip">Likely_benign<span class="tooltiptext tooltip-bottom">BP4	= 1
Multiple lines of computational evidence suggest no impact on gene or gene product (conservation, evolutionary,splicing impact, etc.)

BS1	= 1
Allele frequency is greater than expected for disorder (see Table 6) > 1% in ESP6500all ExAc? need to check more

</span></div> 	</td>
	<td><div class="tooltip">.</div>	</td>
	<td ><div class="tooltip">exonic<span class="tooltiptext tooltip-bottom">ExonicFunc.refGene:
 nonsynonymous_SNV

AAChange.refGene:
 NIPAL3:NM_001322854:exon2:c.G19A:p.A7T
 NIPAL3:NM_001322857:exon2:c.G19A:p.A7T
 NIPAL3:NM_001322865:exon2:c.G19A:p.A7T
 NIPAL3:NM_001322866:exon2:c.G19A:p.A7T
 NIPAL3:NM_001322868:exon2:c.G19A:p.A7T
 NIPAL3:NM_020448:exon2:c.G19A:p.A7T
 NIPAL3:NM_001322855:exon3:c.G19A:p.A7T
 NIPAL3:NM_001322856:exon3:c.G19A:p.A7T
 NIPAL3:NM_001330409:exon3:c.G19A:p.A7T
 NIPAL3:NM_001322854:exon2:c.19G>A:p.A7T
 NIPAL3:NM_001322857:exon2:c.19G>A:p.A7T
 NIPAL3:NM_001322865:exon2:c.19G>A:p.A7T
 NIPAL3:NM_001322866:exon2:c.19G>A:p.A7T
 NIPAL3:NM_001322868:exon2:c.19G>A:p.A7T
 NIPAL3:NM_020448:exon2:c.19G>A:p.A7T
 NIPAL3:NM_001322855:exon3:c.19G>A:p.A7T
 NIPAL3:NM_001322856:exon3:c.19G>A:p.A7T
 NIPAL3:NM_001330409:exon3:c.19G>A:p.A7T

GeneDetail.refGene:
 .

</span></div> 	</td>
	<td style="background-color:inherit"><div class="tooltip">0/0<span class="tooltiptext tooltip-bottom">CALLER = GATK	 QUALITY = 1577.13

Genotype-B00GMSH	 -	 0/0
DP = 88	 AD = 88,0	 AB = 0

Genotype-B00GMSI	 -	 0/1
DP = 90	 AD = 42,48	 AB = 0.533

Genotype-B00GMSJ	 -	 0/0
DP = 94	 AD = 94,0	 AB = 0

</span></div>	</td>
	<td style="background-color:inherit"><div class="tooltip">0/1</div>	</td>
	<td style="background-color:inherit"><div class="tooltip">0/0</div>	</td>
	<td><div class="tooltip">chr1</div>	</td>
	<td><div class="tooltip">24746056</div>	</td>
	<td><div class="tooltip">rs148391167</div>	</td>
	<td><div class="tooltip">G</div>	</td>
	<td><div class="tooltip">A</div>	</td>
	<td><div class="tooltip">PASS</div>	</td>
</tr>
<tr class="ALL" >
	<td ><div class="tooltip">33<span class="tooltiptext tooltip-bottom">MPA_ranking	= 9
MPA_final_score	= 0.0
MPA_impact	= missense_impact
MPA_adjusted	= 0.0
MPA_available	= 10
MPA_deleterious	= 0

--- SPLICE ---
dbscSNV_ADA_SCORE	= .
dbscSNV_RF_SCORE	= .
spliceai_filtered	= .

--- MISSENSE ---
LRT_pred	= N
SIFT_pred	= T
FATHMM_pred	= T
MetaLR_pred	= T
MetaSVM_pred	= T
PROVEAN_pred	= N
Polyphen2_HDIV_pred	= B
Polyphen2_HVAR_pred	= B
MutationTaster_pred	= N
fathmm-MKL_coding_pred	= N
</span></div>	</td>
	<td><div class="tooltip">0_missense_impact</div>	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom"></span></div> 	</td>
	<td style="background-color:#BFFF00"><div class="tooltip">PADI4<span class="tooltiptext tooltip-bottom">LOEUF = 1.136
LOEUF_decile = 6.0

oe_lof = 0.8241200000000001
oe_lof_lower = 0.606

Missense_upper = 1.099

Function Description:
FUNCTION:_Catalyzes_the_citrullination/deimination_of_arginine_residues_of_proteins_such_as_histones,_thereby_playing_a_key_role_in_histone_code_and_regulation_of_stem_cell_maintenance._Citrullinates_histone_H1_at_'Arg_54'_(to_form_H1R54ci),_histone_H3_at_'Arg_2',_'Arg_8',_'Arg_17'_and/or_'Arg_26'_(to_form_H3R2ci,_H3R8ci,_H3R17ci,_H3R26ci,_respectively)_and_histone_H4_at_'Arg_3'_(to_form_H4R3ci)._Acts_as_a_key_regulator_of_stem_cell_maintenance_by_mediating_citrullination_of_histone_H1:_citrullination_of_'Arg_54'_of_histone_H1_(H1R54ci)_results_in_H1_displacement_from_chromatin_and_global_chromatin_decondensation,_thereby_promoting_pluripotency_and_stem_cell_maintenance._Promotes_profound_chromatin_decondensation_during_the_innate_immune_response_to_infection_in_neutrophils_by_mediating_formation_of_H1R54ci._Citrullination_of_histone_H3_prevents_their_methylation_by_CARM1_and_HRMT1L2/PRMT1_and_represses_transcription._Citrullinates_EP300/P300_at_'Arg_2142',_which_favors_its_interaction_with_NCOA2/GRIP1._{ECO:0000269|PubMed:15339660,_ECO:0000269|PubMed:15345777,_ECO:0000269|PubMed:15731352,_ECO:0000269|PubMed:16567635,_ECO:0000269|PubMed:18209087,_ECO:0000269|PubMed:21245532}.

Tissue specificity:
TISSUE_SPECIFICITY:_Expressed_in_eosinophils_and_neutrophils,_not_expressed_in_peripheral_monocytes_or_lymphocytes._{ECO:0000269|PubMed:11435484}.
</span></div>    	</td>
	<td ><div class="tooltip">{Rheumatoid_arthritis,_susceptibility_to},_180300__3_<span class="tooltiptext tooltip-bottom">Disease_description.refGene:
 

DISEASE:_Rheumatoid_arthritis_(RA)_[MIM:180300]:_An_inflammatory_disease_with_autoimmune_features_and_a_complex_genetic_component._It_primarily_affects_the_joints_and_is_characterized_by_inflammatory_changes_in_the_synovial_membranes_and_articular_structures,_widespread_fibrinoid_degeneration_of_the_collagen_fibers_in_mesenchymal_tissues,_and_by_atrophy_and_rarefaction_of_bony_structures._{ECO:0000269|PubMed:12833157}._Note=The_gene_represented_in_this_entry_may_be_involved_in_disease_pathogenesis._The_association_to_rheumatoid_arthritis_was_initially_thought_to_result_from_increased_citrullination_of_target_proteins_(PubMed:12833157)._However,_variants_that_have_been_associated_to_rheumatoid_arthritis_(Ser_55,_Ala_82_and_Ala_112)_do_not_affect_the_catalytic_activity_or_the_citrullination_activity_of_PADI4,_suggesting_that_these_variants_may_affect_the_mRNA_stability_rather_than_the_protein_(PubMed:21245532)._{ECO:0000269|PubMed:12833157,_ECO:0000269|PubMed:21245532}.

</span></div> 	</td>
	<td ><div class="tooltip">0.0095<span class="tooltiptext tooltip-bottom">gnomAD_genome_ALL	= 0.0095
gnomAD_genome_AFR	= 0.0026
gnomAD_genome_AMR	= 0.0156
gnomAD_genome_ASJ	= 0.0033
gnomAD_genome_EAS	= 0
gnomAD_genome_FIN	= 0.0097
gnomAD_genome_NFE	= 0.0144
gnomAD_genome_OTH	= 0.0071
</span></div> 	</td>
	<td ><div class="tooltip">0.0107<span class="tooltiptext tooltip-bottom">gnomAD_exome_ALL	= 0.0107
gnomAD_exome_AFR	= 0.0038
gnomAD_exome_AMR	= 0.0102
gnomAD_exome_ASJ	= 0.0012
gnomAD_exome_EAS	= 0.0002
gnomAD_exome_FIN	= 0.0071
gnomAD_exome_NFE	= 0.0174
gnomAD_exome_OTH	= 0.0137
</span></div> 	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom">CLNREVSTAT:
 .

CLNDN:
 .

CLNALLELEID:
 .

CLNDISDB:
 .

</span></div> 	</td>
	<td ><div class="tooltip">Likely_benign<span class="tooltiptext tooltip-bottom">PM1	= 1
Located in a mutational hot spot and/or critical and well-established functional domain (e.g., active site of an enzyme) without benign variation

BP4	= 1
Multiple lines of computational evidence suggest no impact on gene or gene product (conservation, evolutionary,splicing impact, etc.)

BS1	= 1
Allele frequency is greater than expected for disorder (see Table 6) > 1% in ESP6500all ExAc? need to check more

</span></div> 	</td>
	<td><div class="tooltip">.</div>	</td>
	<td ><div class="tooltip">exonic<span class="tooltiptext tooltip-bottom">ExonicFunc.refGene:
 nonsynonymous_SNV

AAChange.refGene:
 PADI4:NM_012387:exon3:c.C304A:p.P102T
 PADI4:NM_012387:exon3:c.304C>A:p.P102T

GeneDetail.refGene:
 .

</span></div> 	</td>
	<td style="background-color:inherit"><div class="tooltip">0/0<span class="tooltiptext tooltip-bottom">CALLER = GATK	 QUALITY = 1097.13

Genotype-B00GMSH	 -	 0/0
DP = 61	 AD = 61,0	 AB = 0

Genotype-B00GMSI	 -	 0/1
DP = 83	 AD = 45,38	 AB = 0.457

Genotype-B00GMSJ	 -	 0/0
DP = 83	 AD = 83,0	 AB = 0

</span></div>	</td>
	<td style="background-color:inherit"><div class="tooltip">0/1</div>	</td>
	<td style="background-color:inherit"><div class="tooltip">0/0</div>	</td>
	<td><div class="tooltip">chr1</div>	</td>
	<td><div class="tooltip">17660468</div>	</td>
	<td><div class="tooltip">rs34309058</div>	</td>
	<td><div class="tooltip">C</div>	</td>
	<td><div class="tooltip">A</div>	</td>
	<td><div class="tooltip">PASS</div>	</td>
</tr>
<tr class="ALL" >
	<td ><div class="tooltip">34<span class="tooltiptext tooltip-bottom">MPA_ranking	= 9
MPA_final_score	= 0.0
MPA_impact	= missense_impact
MPA_adjusted	= 0.0
MPA_available	= 10
MPA_deleterious	= 0

--- SPLICE ---
dbscSNV_ADA_SCORE	= .
dbscSNV_RF_SCORE	= .
spliceai_filtered	= .

--- MISSENSE ---
LRT_pred	= N
SIFT_pred	= T
FATHMM_pred	= T
MetaLR_pred	= T
MetaSVM_pred	= T
PROVEAN_pred	= N
Polyphen2_HDIV_pred	= B
Polyphen2_HVAR_pred	= B
MutationTaster_pred	= N
fathmm-MKL_coding_pred	= N
</span></div>	</td>
	<td><div class="tooltip">0_missense_impact</div>	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom"></span></div> 	</td>
	<td style="background-color:#FFFF00"><div class="tooltip">FBXO2<span class="tooltiptext tooltip-bottom">LOEUF = 0.997
LOEUF_decile = 5.0

oe_lof = 0.55272
oe_lof_lower = 0.323

Missense_upper = 1.116

Function Description:
FUNCTION:_Substrate_recognition_component_of_a_SCF_(SKP1_CUL1_F_box_protein)_E3_ubiquitin_protein_ligase_complex_that_mediates_the_ubiquitination_and_subsequent_proteasomal_degradation_of_target_proteins._Involved_in_the_endoplasmic_reticulum_associated_degradation_pathway_(ERAD)_for_misfolded_lumenal_proteins_by_recognizing_and_binding_sugar_chains_on_unfolded_glycoproteins_that_are_retrotranslocated_into_the_cytosol_and_promoting_their_ubiquitination_and_subsequent_degradation._Prevents_formation_of_cytosolic_aggregates_of_unfolded_glycoproteins_that_have_been_retrotranslocated_into_the_cytosol._Able_to_recognize_and_bind_denatured_glycoproteins,_preferentially_those_of_the_high_mannose_type_(By_similarity)._{ECO:0000250}.;FUNCTION:_Substrate_recognition_component_of_a_SCF_(SKP1_CUL1_F_box_protein)_E3_ubiquitin_protein_ligase_complex_that_mediates_the_ubiquitination_and_subsequent_proteasomal_degradation_of_target_proteins._Involved_in_the_endoplasmic_reticulum_associated_degradation_pathway_(ERAD)_for_misfolded_lumenal_proteins_by_recognizing_and_binding_sugar_chains_on_unfolded_glycoproteins_that_are_retrotranslocated_into_the_cytosol_and_promoting_their_ubiquitination_and_subsequent_degradation._Prevents_formation_of_cytosolic_aggregates_of_unfolded_glycoproteins_that_have_been_retrotranslocated_into_the_cytosol._Able_to_recognize_and_bind_denatured_glycoproteins,_preferentially_those_of_the_high_mannose_type_(By_similarity)._{ECO:0000250}.

</span></div>    	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom">Disease_description.refGene:
 .

</span></div> 	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom">gnomAD_genome_ALL	= .
gnomAD_genome_AFR	= .
gnomAD_genome_AMR	= .
gnomAD_genome_ASJ	= .
gnomAD_genome_EAS	= .
gnomAD_genome_FIN	= .
gnomAD_genome_NFE	= .
gnomAD_genome_OTH	= .
</span></div> 	</td>
	<td ><div class="tooltip">4.414e-06<span class="tooltiptext tooltip-bottom">gnomAD_exome_ALL	= 4.414e-06
gnomAD_exome_AFR	= 0
gnomAD_exome_AMR	= 0
gnomAD_exome_ASJ	= 0
gnomAD_exome_EAS	= 0
gnomAD_exome_FIN	= 0
gnomAD_exome_NFE	= 9.807e-06
gnomAD_exome_OTH	= 0
</span></div> 	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom">CLNREVSTAT:
 .

CLNDN:
 .

CLNALLELEID:
 .

CLNDISDB:
 .

</span></div> 	</td>
	<td ><div class="tooltip">Uncertain_significance<span class="tooltiptext tooltip-bottom">PM1	= 1
Located in a mutational hot spot and/or critical and well-established functional domain (e.g., active site of an enzyme) without benign variation

BP4	= 1
Multiple lines of computational evidence suggest no impact on gene or gene product (conservation, evolutionary,splicing impact, etc.)

</span></div> 	</td>
	<td><div class="tooltip">.</div>	</td>
	<td ><div class="tooltip">exonic<span class="tooltiptext tooltip-bottom">ExonicFunc.refGene:
 nonsynonymous_SNV

AAChange.refGene:
 FBXO2:NM_012168:exon2:c.C334G:p.Q112E
 FBXO2:NM_012168:exon2:c.334C>G:p.Q112E

GeneDetail.refGene:
 .

</span></div> 	</td>
	<td style="background-color:inherit"><div class="tooltip">0/0<span class="tooltiptext tooltip-bottom">CALLER = GATK	 QUALITY = 1092.13

Genotype-B00GMSH	 -	 0/0
DP = 124	 AD = 123,1	 AB = 0.008

Genotype-B00GMSI	 -	 0/0
DP = 120	 AD = 120,0	 AB = 0

Genotype-B00GMSJ	 -	 0/1
DP = 112	 AD = 70,42	 AB = 0.375

</span></div>	</td>
	<td style="background-color:inherit"><div class="tooltip">0/0</div>	</td>
	<td style="background-color:inherit"><div class="tooltip">0/1</div>	</td>
	<td><div class="tooltip">chr1</div>	</td>
	<td><div class="tooltip">11710580</div>	</td>
	<td><div class="tooltip">.</div>	</td>
	<td><div class="tooltip">G</div>	</td>
	<td><div class="tooltip">C</div>	</td>
	<td><div class="tooltip">PASS</div>	</td>
</tr>
<tr class="ALL" >
	<td ><div class="tooltip">35<span class="tooltiptext tooltip-bottom">MPA_ranking	= 9
MPA_final_score	= 0.0
MPA_impact	= missense_impact
MPA_adjusted	= 0.0
MPA_available	= 10
MPA_deleterious	= 0

--- SPLICE ---
dbscSNV_ADA_SCORE	= .
dbscSNV_RF_SCORE	= .
spliceai_filtered	= .

--- MISSENSE ---
LRT_pred	= N
SIFT_pred	= T
FATHMM_pred	= T
MetaLR_pred	= T
MetaSVM_pred	= T
PROVEAN_pred	= N
Polyphen2_HDIV_pred	= B
Polyphen2_HVAR_pred	= B
MutationTaster_pred	= N
fathmm-MKL_coding_pred	= N
</span></div>	</td>
	<td><div class="tooltip">0_missense_impact</div>	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom"></span></div> 	</td>
	<td style="background-color:#BFFF00"><div class="tooltip">PLEKHN1<span class="tooltiptext tooltip-bottom">LOEUF = 1.124
LOEUF_decile = 6.0

oe_lof = 0.80467
oe_lof_lower = 0.585

Missense_upper = 1.358

Function Description:
FUNCTION:_Controls_the_stability_of_the_leptin_mRNA_harboring_an_AU_rich_element_(ARE)_in_its_3'_UTR,_in_cooperation_with_the_RNA_stabilizer_ELAVL1_(PubMed:29180010)._Decreases_the_stability_of_the_leptin_mRNA_by_antagonizing_the_function_of_ELAVL1_by_inducing_its_atypical_recruitment_from_the_nucleus_to_the_cytosol_(By_similarity)._Binds_to_cardiolipin_(CL),_phosphatidic_acid_(PA),_phosphatidylinositol_4_phosphate_(PtdIns(4)P)_and_phosphatidylserine_(PS)_(PubMed:18191643)._Promotes_apoptosis_by_enhancing_BAX_BAK_hetero_oligomerization_via_interaction_with_BID_in_colon_cancer_cells_(PubMed:29531808)_(By_similarity)._{ECO:0000250|UniProtKB:Q8C886,_ECO:0000269|PubMed:18191643,_ECO:0000269|PubMed:29180010,_ECO:0000269|PubMed:29531808}.

Tissue specificity:
TISSUE_SPECIFICITY:_Ubiquitous_(PubMed:18191643)._Epressed_in_several_cancer_cell_lines_of_differing_origin_(PubMed:29531808)._{ECO:0000269|PubMed:18191643,_ECO:0000269|PubMed:29531808}.
</span></div>    	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom">Disease_description.refGene:
 .

</span></div> 	</td>
	<td ><div class="tooltip">0.0003<span class="tooltiptext tooltip-bottom">gnomAD_genome_ALL	= 0.0003
gnomAD_genome_AFR	= 0
gnomAD_genome_AMR	= 0.0012
gnomAD_genome_ASJ	= 0.0038
gnomAD_genome_EAS	= 0
gnomAD_genome_FIN	= 0
gnomAD_genome_NFE	= 0.0004
gnomAD_genome_OTH	= 0
</span></div> 	</td>
	<td ><div class="tooltip">0.0036<span class="tooltiptext tooltip-bottom">gnomAD_exome_ALL	= 0.0036
gnomAD_exome_AFR	= 0.0024
gnomAD_exome_AMR	= 0.0007
gnomAD_exome_ASJ	= 0.0012
gnomAD_exome_EAS	= 0.0021
gnomAD_exome_FIN	= 0.0073
gnomAD_exome_NFE	= 0.0062
gnomAD_exome_OTH	= 0.0023
</span></div> 	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom">CLNREVSTAT:
 .

CLNDN:
 .

CLNALLELEID:
 .

CLNDISDB:
 .

</span></div> 	</td>
	<td ><div class="tooltip">Uncertain_significance<span class="tooltiptext tooltip-bottom">PM1	= 1
Located in a mutational hot spot and/or critical and well-established functional domain (e.g., active site of an enzyme) without benign variation

BP4	= 1
Multiple lines of computational evidence suggest no impact on gene or gene product (conservation, evolutionary,splicing impact, etc.)

</span></div> 	</td>
	<td><div class="tooltip">.</div>	</td>
	<td ><div class="tooltip">exonic<span class="tooltiptext tooltip-bottom">ExonicFunc.refGene:
 nonsynonymous_SNV

AAChange.refGene:
 PLEKHN1:NM_001160184:exon10:c.A992G:p.E331G
 PLEKHN1:NM_032129:exon10:c.A956G:p.E319G
 PLEKHN1:NM_001160184:exon10:c.992A>G:p.E331G
 PLEKHN1:NM_032129:exon10:c.956A>G:p.E319G

GeneDetail.refGene:
 .

</span></div> 	</td>
	<td style="background-color:inherit"><div class="tooltip">0/0<span class="tooltiptext tooltip-bottom">CALLER = GATK	 QUALITY = 151.16

Genotype-B00GMSH	 -	 0/0
DP = 44	 AD = 38,6	 AB = 0.136

Genotype-B00GMSI	 -	 0/1
DP = 30	 AD = 21,9	 AB = 0.3

Genotype-B00GMSJ	 -	 0/1
DP = 35	 AD = 24,11	 AB = 0.314

</span></div>	</td>
	<td style="background-color:inherit"><div class="tooltip">0/1</div>	</td>
	<td style="background-color:inherit"><div class="tooltip">0/1</div>	</td>
	<td><div class="tooltip">chr1</div>	</td>
	<td><div class="tooltip">907758</div>	</td>
	<td><div class="tooltip">.</div>	</td>
	<td><div class="tooltip">A</div>	</td>
	<td><div class="tooltip">G</div>	</td>
	<td><div class="tooltip">PASS</div>	</td>
</tr>
<tr class="ALL" >
	<td ><div class="tooltip">36<span class="tooltiptext tooltip-bottom">MPA_ranking	= 9
MPA_final_score	= 0.0
MPA_impact	= missense_impact
MPA_adjusted	= 0.0
MPA_available	= 2
MPA_deleterious	= 0

--- SPLICE ---
dbscSNV_ADA_SCORE	= .
dbscSNV_RF_SCORE	= .
spliceai_filtered	= .

--- MISSENSE ---
LRT_pred	= .
SIFT_pred	= .
FATHMM_pred	= .
MetaLR_pred	= .
MetaSVM_pred	= .
PROVEAN_pred	= .
Polyphen2_HDIV_pred	= .
Polyphen2_HVAR_pred	= .
MutationTaster_pred	= N
fathmm-MKL_coding_pred	= N
</span></div>	</td>
	<td><div class="tooltip">0_missense_impact</div>	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom"></span></div> 	</td>
	<td style="background-color:#FFFFFF"><div class="tooltip">CRYBG2<span class="tooltiptext tooltip-bottom">.</span></div>    	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom">Disease_description.refGene:
 .

</span></div> 	</td>
	<td ><div class="tooltip">0.0028<span class="tooltiptext tooltip-bottom">gnomAD_genome_ALL	= 0.0028
gnomAD_genome_AFR	= 0.0009
gnomAD_genome_AMR	= 0.0036
gnomAD_genome_ASJ	= 0
gnomAD_genome_EAS	= 0
gnomAD_genome_FIN	= 0.0009
gnomAD_genome_NFE	= 0.0048
gnomAD_genome_OTH	= 0
</span></div> 	</td>
	<td ><div class="tooltip">0.0026<span class="tooltiptext tooltip-bottom">gnomAD_exome_ALL	= 0.0026
gnomAD_exome_AFR	= 0.0003
gnomAD_exome_AMR	= 0.0028
gnomAD_exome_ASJ	= 0.0016
gnomAD_exome_EAS	= 0
gnomAD_exome_FIN	= 9.28e-05
gnomAD_exome_NFE	= 0.0042
gnomAD_exome_OTH	= 0.0030
</span></div> 	</td>
	<td ><div class="tooltip">.<span class="tooltiptext tooltip-bottom">CLNREVSTAT:
 .

CLNDN:
 .

CLNALLELEID:
 .

CLNDISDB:
 .

</span></div> 	</td>
	<td ><div class="tooltip">Uncertain_significance<span class="tooltiptext tooltip-bottom">BS1	= 1
Allele frequency is greater than expected for disorder (see Table 6) > 1% in ESP6500all ExAc? need to check more

</span></div> 	</td>
	<td><div class="tooltip">.</div>	</td>
	<td ><div class="tooltip">exonic<span class="tooltiptext tooltip-bottom">ExonicFunc.refGene:
 nonsynonymous_SNV

AAChange.refGene:
 CRYBG2:NM_001039775:exon2:c.A2102G:p.D701G
 CRYBG2:NM_001039775:exon2:c.2102A>G:p.D701G

GeneDetail.refGene:
 .

</span></div> 	</td>
	<td style="background-color:inherit"><div class="tooltip">0/0<span class="tooltiptext tooltip-bottom">CALLER = GATK	 QUALITY = 1296.13

Genotype-B00GMSH	 -	 0/0
DP = 93	 AD = 93,0	 AB = 0

Genotype-B00GMSI	 -	 0/1
DP = 100	 AD = 56,44	 AB = 0.44

Genotype-B00GMSJ	 -	 0/0
DP = 93	 AD = 93,0	 AB = 0

</span></div>	</td>
	<td style="background-color:inherit"><div class="tooltip">0/1</div>	</td>
	<td style="background-color:inherit"><div class="tooltip">0/0</div>	</td>
	<td><div class="tooltip">chr1</div>	</td>
	<td><div class="tooltip">26671047</div>	</td>
	<td><div class="tooltip">rs199647245</div>	</td>
	<td><div class="tooltip">T</div>	</td>
	<td><div class="tooltip">C</div>	</td>
	<td><div class="tooltip">PASS</div>	</td>
</tr>
</tbody>
</table>
</div>


<div class="tab">
<input type="button" class="tablinks" value="ALL_0.01" onclick="filter('ALL')">
<input type="button" class="tablinks" value="disease2 (6)" onclick="filter('disease2')" >
<input type="button" class="tablinks" value="OMIM_REC (5)" onclick="filter('OMIMREC')" >
<input type="button" class="tablinks" value="PATHO1 (4)" onclick="filter('PATHO1')" >
<input type="button" class="tablinks" value="OMIM_DOM (3)" onclick="filter('OMIMDOM')" >
<input type="button" class="tablinks" value="METADATA (1)" onclick="filter('META')" >
<input type="button" class="tablinks" value="HTZ_compo (0)" onclick="filter('HTZcompo')" >
<input type="button" class="tablinks" value="SNVmumVsCNVdad (0)" onclick="filter('SNPmCNVp')" >
<input type="button" class="tablinks" value="SNVdadVsCNVmum (0)" onclick="filter('SNPpCNVm')" >
<input type="button" class="tablinks" value="DENOVO (0)" onclick="filter('DENOVO')" >
<input type="button" class="tablinks" value="AR (0)" onclick="filter('AUTOREC')" >
<input type="button" class="tablinks" value="DS_ACMG (0)" onclick="filter('ACMG')" >
</div>
</body>
</html>